{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import tokenizer\n",
    "import pickle\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import glob\n",
    "import os\n",
    "import string\n",
    "from nltk.tokenize import sent_tokenize\n",
    "from nltk.tokenize import word_tokenize\n",
    "from nltk.corpus import stopwords\n",
    "from nltk.stem.porter import PorterStemmer\n",
    "from nltk.util import ngrams\n",
    "import json\n",
    "import sparse_matrix_functions\n",
    "import scipy.sparse as scs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "path = '/Users/ale/Dropbox (Yadlin Family)/galvanize/capstone/*.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open(\"gene_dictionary_final.pickle\", \"rb\") as dict_gene:\n",
    "        gene_dict = pickle.load(dict_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_keys = set(gene_dict.keys())\n",
    "gene_values = set(gene_dict.values())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'#erbb2#'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_dict['her2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 604,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open(\"drug_dictionary_final.pickle\", \"rb\") as dict_drug:\n",
    "        drug_dict = pickle.load(dict_drug)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 605,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "drug_keys = set(drug_dict.keys())\n",
    "drug_values = set(drug_dict.values())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 606,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'#trastuzumab#'"
      ]
     },
     "execution_count": 606,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_dict['trastuzumab']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open(\"greek_alphabet.pickle\", \"rb\") as dict_greek:\n",
    "        greek_dict = pickle.load(dict_greek)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import data_frame_creator"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test=data_frame_creator.create_data_frame(path, short_list=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>files</th>\n",
       "      <th>sentences</th>\n",
       "      <th>#abcg2#</th>\n",
       "      <th>#acadl#</th>\n",
       "      <th>#ackr3#</th>\n",
       "      <th>#acpp#</th>\n",
       "      <th>#ado#</th>\n",
       "      <th>#afuresertib#</th>\n",
       "      <th>#agfg1#</th>\n",
       "      <th>#aicda#</th>\n",
       "      <th>...</th>\n",
       "      <th>#voxtalisib#</th>\n",
       "      <th>#wdtc1#</th>\n",
       "      <th>#wnt5a#</th>\n",
       "      <th>#xs#</th>\n",
       "      <th>#zbp1#</th>\n",
       "      <th>#zbtb8os#</th>\n",
       "      <th>#zfyve9#</th>\n",
       "      <th>#znf746#</th>\n",
       "      <th>resist</th>\n",
       "      <th>sensit</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>guidelines.txt</td>\n",
       "      <td>Something that moves progressivley First thing...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>test.txt</td>\n",
       "      <td>13  Deregulation of the EGFR PI3K PTEN Akt mTO...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>test.txt</td>\n",
       "      <td>ABSTRACT The EGFR PI3K PTEN Akt mTORC1 GSK-3 p...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>test.txt</td>\n",
       "      <td>The expression of this pathway is frequently a...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>test.txt</td>\n",
       "      <td>In some breast cancer cases, mutations at cert...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 408 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            files                                          sentences  #abcg2#  \\\n",
       "0  guidelines.txt  Something that moves progressivley First thing...        0   \n",
       "3        test.txt  13  Deregulation of the EGFR PI3K PTEN Akt mTO...        0   \n",
       "4        test.txt  ABSTRACT The EGFR PI3K PTEN Akt mTORC1 GSK-3 p...        0   \n",
       "5        test.txt  The expression of this pathway is frequently a...        0   \n",
       "6        test.txt  In some breast cancer cases, mutations at cert...        0   \n",
       "\n",
       "   #acadl#  #ackr3#  #acpp#  #ado#  #afuresertib#  #agfg1#  #aicda#   ...    \\\n",
       "0        0        0       0      0              0        0        0   ...     \n",
       "3        0        0       0      0              0        0        0   ...     \n",
       "4        0        0       0      0              0        0        0   ...     \n",
       "5        0        0       0      0              0        0        0   ...     \n",
       "6        0        0       0      0              0        0        0   ...     \n",
       "\n",
       "   #voxtalisib#  #wdtc1#  #wnt5a#  #xs#  #zbp1#  #zbtb8os#  #zfyve9#  \\\n",
       "0             0        0        0     0       0          0         0   \n",
       "3             0        0        0     0       0          0         0   \n",
       "4             0        0        0     0       0          0         0   \n",
       "5             0        0        0     0       0          0         0   \n",
       "6             0        0        0     0       0          0         0   \n",
       "\n",
       "   #znf746#  resist  sensit  \n",
       "0         0       0       0  \n",
       "3         0       1       0  \n",
       "4         0       1       0  \n",
       "5         0       0       0  \n",
       "6         0       0       0  \n",
       "\n",
       "[5 rows x 408 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1428, 408)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "doc_names, orig_sentences = data_frame_creator.create_corpus()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#orig_sentences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data has been processed and saved\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[<2893x267 sparse matrix of type '<class 'numpy.int64'>'\n",
       " \twith 12901 stored elements in Compressed Sparse Row format>,\n",
       " ['#akt1#',\n",
       "  '#ar#',\n",
       "  '#brca1#',\n",
       "  '#brca2#',\n",
       "  '#carbendazim#',\n",
       "  '#celp#',\n",
       "  '#cic#',\n",
       "  '#doxorubicin#',\n",
       "  '#egfr#',\n",
       "  '#erbb2#',\n",
       "  '#esr1#',\n",
       "  '#everolimus#',\n",
       "  '#lapatinib#',\n",
       "  '#letrozole#',\n",
       "  '#mice#',\n",
       "  '#mtor#',\n",
       "  '#nf1#',\n",
       "  '#paclitaxel#',\n",
       "  '#pertuzumab#',\n",
       "  '#pik3ca#',\n",
       "  '#pten#',\n",
       "  '#sgcg#',\n",
       "  '#sirt1#',\n",
       "  '#tamoxifen#',\n",
       "  '#tc2n#',\n",
       "  '#tp53#',\n",
       "  '#trastuzumab#',\n",
       "  'aberr',\n",
       "  'abil',\n",
       "  'activ',\n",
       "  'addit',\n",
       "  'advanc',\n",
       "  'affect',\n",
       "  'age',\n",
       "  'alpha',\n",
       "  'also',\n",
       "  'alter',\n",
       "  'amplif',\n",
       "  'analysi',\n",
       "  'angulo',\n",
       "  'anti',\n",
       "  'antibodi',\n",
       "  'apoptosi',\n",
       "  'approach',\n",
       "  'associ',\n",
       "  'aurora',\n",
       "  'author',\n",
       "  'base',\n",
       "  'beta',\n",
       "  'bind',\n",
       "  'biol',\n",
       "  'biolog',\n",
       "  'black',\n",
       "  'breast',\n",
       "  'cancer',\n",
       "  'carcinoma',\n",
       "  'case',\n",
       "  'cell',\n",
       "  'certain',\n",
       "  'chang',\n",
       "  'chemotherapi',\n",
       "  'cic',\n",
       "  'clin',\n",
       "  'clinic',\n",
       "  'combin',\n",
       "  'compar',\n",
       "  'complex',\n",
       "  'construct',\n",
       "  'contain',\n",
       "  'contrast',\n",
       "  'control',\n",
       "  'correl',\n",
       "  'cycl',\n",
       "  'cyclin',\n",
       "  'decreas',\n",
       "  'depend',\n",
       "  'detect',\n",
       "  'determin',\n",
       "  'develop',\n",
       "  'differ',\n",
       "  'diseas',\n",
       "  'display',\n",
       "  'dna',\n",
       "  'doi',\n",
       "  'domain',\n",
       "  'drug',\n",
       "  'effect',\n",
       "  'egfr',\n",
       "  'egfr3',\n",
       "  'egfr4',\n",
       "  'egfrviii',\n",
       "  'either',\n",
       "  'elev',\n",
       "  'emt',\n",
       "  'endocrin',\n",
       "  'endometri',\n",
       "  'enhanc',\n",
       "  'epitheli',\n",
       "  'eralpha',\n",
       "  'erk',\n",
       "  'estrogen',\n",
       "  'evalu',\n",
       "  'examin',\n",
       "  'express',\n",
       "  'factor',\n",
       "  'famili',\n",
       "  'figur',\n",
       "  'first',\n",
       "  'follow',\n",
       "  'format',\n",
       "  'free',\n",
       "  'frequent',\n",
       "  'function',\n",
       "  'gene',\n",
       "  'genet',\n",
       "  'group',\n",
       "  'growth',\n",
       "  'gsk-3',\n",
       "  'gsk-3beta',\n",
       "  'high',\n",
       "  'hormon',\n",
       "  'howev',\n",
       "  'human',\n",
       "  'identifi',\n",
       "  'immunoprecipit',\n",
       "  'implic',\n",
       "  'import',\n",
       "  'inactiv',\n",
       "  'includ',\n",
       "  'increas',\n",
       "  'independ',\n",
       "  'indic',\n",
       "  'induc',\n",
       "  'inhibit',\n",
       "  'inhibitor',\n",
       "  'interact',\n",
       "  'invas',\n",
       "  'investig',\n",
       "  'involv',\n",
       "  'key',\n",
       "  'kinas',\n",
       "  'knockdown',\n",
       "  'level',\n",
       "  'ligand',\n",
       "  'like',\n",
       "  'line',\n",
       "  'link',\n",
       "  'liu',\n",
       "  'local',\n",
       "  'loss',\n",
       "  'lumin',\n",
       "  'lung',\n",
       "  'malign',\n",
       "  'mammari',\n",
       "  'mani',\n",
       "  'martelli',\n",
       "  'may',\n",
       "  'mccubrey',\n",
       "  'mcf-7',\n",
       "  'mechan',\n",
       "  'med',\n",
       "  'mediat',\n",
       "  'mek',\n",
       "  'member',\n",
       "  'metastasi',\n",
       "  'metastat',\n",
       "  'metformin',\n",
       "  'methyl',\n",
       "  'mill',\n",
       "  'molecul',\n",
       "  'molecular',\n",
       "  'mrna',\n",
       "  'mtorc1',\n",
       "  'mutant',\n",
       "  'mutat',\n",
       "  'neg',\n",
       "  'normal',\n",
       "  'novel',\n",
       "  'observ',\n",
       "  'occur',\n",
       "  'oncogen',\n",
       "  'oncol',\n",
       "  'one',\n",
       "  'outcom',\n",
       "  'oval',\n",
       "  'ovarian',\n",
       "  'overexpress',\n",
       "  'part',\n",
       "  'pathway',\n",
       "  'patient',\n",
       "  'pcr',\n",
       "  'phase',\n",
       "  'phenotyp',\n",
       "  'phosphoryl',\n",
       "  'play',\n",
       "  'plu',\n",
       "  'popul',\n",
       "  'posit',\n",
       "  'potenti',\n",
       "  'presenc',\n",
       "  'present',\n",
       "  'prevent',\n",
       "  'primari',\n",
       "  'profil',\n",
       "  'prognosi',\n",
       "  'progress',\n",
       "  'prolifer',\n",
       "  'promot',\n",
       "  'protein',\n",
       "  'provid',\n",
       "  'publish',\n",
       "  'raf',\n",
       "  'recent',\n",
       "  'receptor',\n",
       "  'region',\n",
       "  'regul',\n",
       "  'relat',\n",
       "  'repair',\n",
       "  'report',\n",
       "  'requir',\n",
       "  'research',\n",
       "  'resist',\n",
       "  'respons',\n",
       "  'result',\n",
       "  'review',\n",
       "  'risk',\n",
       "  'role',\n",
       "  'sampl',\n",
       "  'sensit',\n",
       "  'show',\n",
       "  'shown',\n",
       "  'signal',\n",
       "  'site',\n",
       "  'small',\n",
       "  'specif',\n",
       "  'sporad',\n",
       "  'stage',\n",
       "  'statu',\n",
       "  'stem',\n",
       "  'studi',\n",
       "  'suggest',\n",
       "  'supplementari',\n",
       "  'suppress',\n",
       "  'suppressor',\n",
       "  'surviv',\n",
       "  'target',\n",
       "  'therapeut',\n",
       "  'therapi',\n",
       "  'thu',\n",
       "  'tissu',\n",
       "  'tnbc',\n",
       "  'transcript',\n",
       "  'transfect',\n",
       "  'transform',\n",
       "  'translat',\n",
       "  'treat',\n",
       "  'treatment',\n",
       "  'trial',\n",
       "  'tripl',\n",
       "  'tumor',\n",
       "  'tumorigenesi',\n",
       "  'two',\n",
       "  'use',\n",
       "  'variou',\n",
       "  'well',\n",
       "  'without',\n",
       "  'women']]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_frame_creator.process_data(orig_sentences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "vocab_matrix=data_frame_creator.open_pickle('vocab_matrix.pickle')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<2893x406 sparse matrix of type '<class 'numpy.int64'>'\n",
       "\twith 2941 stored elements in Compressed Sparse Row format>"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vocab_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'.pickle'"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'vocab_matrix.pickle'[-7:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "vocab_columns=data_frame_creator.open_pickle('vocabulary.pickle')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "406"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(vocab_columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "col_idx=sparse_matrix_functions.column_indexing(vocab_columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'#abcg2#': 0,\n",
       " '#acadl#': 1,\n",
       " '#ackr3#': 2,\n",
       " '#acpp#': 3,\n",
       " '#ado#': 4,\n",
       " '#afuresertib#': 5,\n",
       " '#agfg1#': 6,\n",
       " '#aicda#': 7,\n",
       " '#airn#': 8,\n",
       " '#akt inhibitor gsk2141795#': 9,\n",
       " '#akt inhibitor mk2206#': 10,\n",
       " '#akt1#': 11,\n",
       " '#alb#': 12,\n",
       " '#alisertib#': 13,\n",
       " '#alpelisib#': 14,\n",
       " '#alpi#': 15,\n",
       " '#alyref#': 16,\n",
       " '#amh#': 17,\n",
       " '#anastrozole#': 18,\n",
       " '#anpep#': 19,\n",
       " '#ar#': 20,\n",
       " '#arc#': 21,\n",
       " '#areg#': 22,\n",
       " '#arhgef7#': 23,\n",
       " '#armc9#': 24,\n",
       " '#atg12#': 25,\n",
       " '#atm#': 26,\n",
       " '#atr#': 27,\n",
       " '#bank1#': 28,\n",
       " '#bax#': 29,\n",
       " '#bbc3#': 30,\n",
       " '#bcl2#': 31,\n",
       " '#bcs1l#': 32,\n",
       " '#bevacizumab#': 33,\n",
       " '#bid#': 34,\n",
       " '#binimetinib#': 35,\n",
       " '#birc5#': 36,\n",
       " '#bivm#': 37,\n",
       " '#bmi1#': 38,\n",
       " '#bmp7#': 39,\n",
       " '#bpifa4p#': 40,\n",
       " '#braf#': 41,\n",
       " '#brca1#': 42,\n",
       " '#brca2#': 43,\n",
       " '#brd2#': 44,\n",
       " '#brip1#': 45,\n",
       " '#broxuridine#': 46,\n",
       " '#buparlisib#': 47,\n",
       " '#c19orf66#': 48,\n",
       " '#c1qbp#': 49,\n",
       " '#ca1#': 50,\n",
       " '#capecitabine#': 51,\n",
       " '#carbendazim#': 52,\n",
       " '#carboplatin#': 53,\n",
       " '#cat#': 54,\n",
       " '#ccnd1#': 55,\n",
       " '#ccng2#': 56,\n",
       " '#cct#': 57,\n",
       " '#cd24#': 58,\n",
       " '#cd44#': 59,\n",
       " '#cdh1#': 60,\n",
       " '#cdh17#': 61,\n",
       " '#cdk2#': 62,\n",
       " '#cdk4#': 63,\n",
       " '#cdkn1a#': 64,\n",
       " '#cdkn1b#': 65,\n",
       " '#cdkn2a#': 66,\n",
       " '#cebpd#': 67,\n",
       " '#celp#': 68,\n",
       " '#cetuximab#': 69,\n",
       " '#chek1#': 70,\n",
       " '#chek2#': 71,\n",
       " '#cic#': 72,\n",
       " '#cish#': 73,\n",
       " '#cisplatin#': 74,\n",
       " '#cixutumumab#': 75,\n",
       " '#clofarabine#': 76,\n",
       " '#cnr1#': 77,\n",
       " '#cox8a#': 78,\n",
       " '#cp#': 79,\n",
       " '#cs#': 80,\n",
       " '#csn1s1#': 81,\n",
       " '#ctbp1#': 82,\n",
       " '#ctnnb1#': 83,\n",
       " '#cxcr4#': 84,\n",
       " '#cyclophosphamide#': 85,\n",
       " '#cycs#': 86,\n",
       " '#cyp19a1#': 87,\n",
       " '#cyp4z1#': 88,\n",
       " '#cyp4z2p#': 89,\n",
       " '#dacarbazine#': 90,\n",
       " '#dalir#': 91,\n",
       " '#dalotuzumab#': 92,\n",
       " '#daunorubicin#': 93,\n",
       " '#dcstamp#': 94,\n",
       " '#des#': 95,\n",
       " '#dexamethasone#': 96,\n",
       " '#dgcr2#': 97,\n",
       " '#dlx4#': 98,\n",
       " '#dna2#': 99,\n",
       " '#docetaxel#': 100,\n",
       " '#doxorubicin hydrochloride#': 101,\n",
       " '#doxorubicin#': 102,\n",
       " '#dpm1#': 103,\n",
       " '#dpm2#': 104,\n",
       " '#duoxa1#': 105,\n",
       " '#dusp4#': 106,\n",
       " '#e2f1#': 107,\n",
       " '#eaf2#': 108,\n",
       " '#eec1#': 109,\n",
       " '#eef2#': 110,\n",
       " '#eef2k#': 111,\n",
       " '#egf#': 112,\n",
       " '#egfr#': 113,\n",
       " '#egr3#': 114,\n",
       " '#eif4e#': 115,\n",
       " '#eif4g1#': 116,\n",
       " '#ep300#': 117,\n",
       " '#epirubicin#': 118,\n",
       " '#erbb2#': 119,\n",
       " '#erbb3#': 120,\n",
       " '#erbb4#': 121,\n",
       " '#erlotinib#': 122,\n",
       " '#esr1#': 123,\n",
       " '#esr2#': 124,\n",
       " '#esrra#': 125,\n",
       " '#esrrb#': 126,\n",
       " '#everolimus#': 127,\n",
       " '#exemestane#': 128,\n",
       " '#exo1#': 129,\n",
       " '#exog#': 130,\n",
       " '#f11r#': 131,\n",
       " '#f12#': 132,\n",
       " '#fam129b#': 133,\n",
       " '#fanca#': 134,\n",
       " '#fancc#': 135,\n",
       " '#fancd2#': 136,\n",
       " '#fbn1#': 137,\n",
       " '#fbxw7#': 138,\n",
       " '#fh#': 139,\n",
       " '#fhit#': 140,\n",
       " '#fluorouracil#': 141,\n",
       " '#folitixorin#': 142,\n",
       " '#foxc1#': 143,\n",
       " '#foxg1#': 144,\n",
       " '#foxm1#': 145,\n",
       " '#foxo3#': 146,\n",
       " '#fry#': 147,\n",
       " '#ftcdnl1#': 148,\n",
       " '#fulvestrant#': 149,\n",
       " '#gaa#': 150,\n",
       " '#gata3#': 151,\n",
       " '#gemcitabine#': 152,\n",
       " '#gk#': 153,\n",
       " '#glys1#': 154,\n",
       " '#gnas-as1#': 155,\n",
       " '#gpbar1#': 156,\n",
       " '#gpr149#': 157,\n",
       " '#gpr180#': 158,\n",
       " '#hdac1#': 159,\n",
       " '#hdac3#': 160,\n",
       " '#hdac9#': 161,\n",
       " '#hic1#': 162,\n",
       " '#hif1a#': 163,\n",
       " '#hpp1#': 164,\n",
       " '#hr#': 165,\n",
       " '#hras#': 166,\n",
       " '#hrg#': 167,\n",
       " '#hsp90aa1#': 168,\n",
       " '#hydroxyurea#': 169,\n",
       " '#ifi27#': 170,\n",
       " '#igf1r#': 171,\n",
       " '#igll3p#': 172,\n",
       " '#il36g#': 173,\n",
       " '#impact#': 174,\n",
       " '#infliximab#': 175,\n",
       " '#ins#': 176,\n",
       " '#iproplatin#': 177,\n",
       " '#irf1#': 178,\n",
       " '#irs1#': 179,\n",
       " '#iv#': 180,\n",
       " '#jak2#': 181,\n",
       " '#kalrn#': 182,\n",
       " '#kcnt1#': 183,\n",
       " '#kiaa1551#': 184,\n",
       " '#kit#': 185,\n",
       " '#kl#': 186,\n",
       " '#kras#': 187,\n",
       " '#krit1#': 188,\n",
       " '#krt19#': 189,\n",
       " '#krt7#': 190,\n",
       " '#lapatinib ditosylate#': 191,\n",
       " '#lapatinib#': 192,\n",
       " '#large1#': 193,\n",
       " '#lcn2#': 194,\n",
       " '#lekr1#': 195,\n",
       " '#lenalidomide#': 196,\n",
       " '#letrozole#': 197,\n",
       " '#leucovorin calcium#': 198,\n",
       " '#limch1#': 199,\n",
       " '#lin28a#': 200,\n",
       " '#lipa#': 201,\n",
       " '#litaf#': 202,\n",
       " '#ltf#': 203,\n",
       " '#magee1#': 204,\n",
       " '#mak#': 205,\n",
       " '#map2k1#': 206,\n",
       " '#map2k4#': 207,\n",
       " '#mapk3#': 208,\n",
       " '#mb#': 209,\n",
       " '#mbnl1#': 210,\n",
       " '#mcm8#': 211,\n",
       " '#mdm2#': 212,\n",
       " '#mek inhibitor ci-1040#': 213,\n",
       " '#melk#': 214,\n",
       " '#menogaril#': 215,\n",
       " '#methotrexate#': 216,\n",
       " '#mfap1#': 217,\n",
       " '#mib1#': 218,\n",
       " '#mice#': 219,\n",
       " '#mir145#': 220,\n",
       " '#mitoxantrone#': 221,\n",
       " '#mlh1#': 222,\n",
       " '#mmp9#': 223,\n",
       " '#mro#': 224,\n",
       " '#ms#': 225,\n",
       " '#msh2#': 226,\n",
       " '#msh6#': 227,\n",
       " '#msmp#': 228,\n",
       " '#msx2#': 229,\n",
       " '#mtor kinase inhibitor cc-223#': 230,\n",
       " '#mtor#': 231,\n",
       " '#muc7#': 232,\n",
       " '#myb#': 233,\n",
       " '#myc#': 234,\n",
       " '#myod1#': 235,\n",
       " '#nde1#': 236,\n",
       " '#neratinib#': 237,\n",
       " '#nf1#': 238,\n",
       " '#nfkbiz#': 239,\n",
       " '#niacinamide#': 240,\n",
       " '#nm#': 241,\n",
       " '#nodal#': 242,\n",
       " '#nova2#': 243,\n",
       " '#nqo1#': 244,\n",
       " '#nr4a2#': 245,\n",
       " '#nras#': 246,\n",
       " '#nrg1#': 247,\n",
       " '#nrip3#': 248,\n",
       " '#nsmaf#': 249,\n",
       " '#ovaal#': 250,\n",
       " '#oxaliplatin#': 251,\n",
       " '#paclitaxel albumin-stabilized nanoparticle formulation#': 252,\n",
       " '#paclitaxel#': 253,\n",
       " '#pafah1b1#': 254,\n",
       " '#paics#': 255,\n",
       " '#palb2#': 256,\n",
       " '#palld#': 257,\n",
       " '#parp1#': 258,\n",
       " '#patz1#': 259,\n",
       " '#pdcd4#': 260,\n",
       " '#pdgfra#': 261,\n",
       " '#pdgfrb#': 262,\n",
       " '#pdk1#': 263,\n",
       " '#pdlim5#': 264,\n",
       " '#pdxp#': 265,\n",
       " '#per1#': 266,\n",
       " '#perifosine#': 267,\n",
       " '#pertuzumab#': 268,\n",
       " '#pgp#': 269,\n",
       " '#pgr#': 270,\n",
       " '#phlpp2#': 271,\n",
       " '#phospho2#': 272,\n",
       " '#pi3#': 273,\n",
       " '#pi3k family kinase inhibitor xl147#': 274,\n",
       " '#pi3k-beta inhibitor gsk2636771#': 275,\n",
       " '#pi3k/mtor inhibitor bez235#': 276,\n",
       " '#pik3ca#': 277,\n",
       " '#pik3r1#': 278,\n",
       " '#pik3r2#': 279,\n",
       " '#pin1#': 280,\n",
       " '#pinx1#': 281,\n",
       " '#plcg1#': 282,\n",
       " '#plxna2#': 283,\n",
       " '#pmaip1#': 284,\n",
       " '#pms1#': 285,\n",
       " '#pms2#': 286,\n",
       " '#ppara#': 287,\n",
       " '#ppp6c#': 288,\n",
       " '#prdm6#': 289,\n",
       " '#prdx2#': 290,\n",
       " '#prkaa2#': 291,\n",
       " '#prl#': 292,\n",
       " '#proc#': 293,\n",
       " '#procarbazine hydrochloride#': 294,\n",
       " '#prss30p#': 295,\n",
       " '#psmd4#': 296,\n",
       " '#psmd5#': 297,\n",
       " '#pten#': 298,\n",
       " '#ptpn5#': 299,\n",
       " '#rab11fip1#': 300,\n",
       " '#rad51#': 301,\n",
       " '#rad51c#': 302,\n",
       " '#raf1#': 303,\n",
       " '#rapgef5#': 304,\n",
       " '#rasa1#': 305,\n",
       " '#rasal2#': 306,\n",
       " '#rb1#': 307,\n",
       " '#reck#': 308,\n",
       " '#recombinant interferon#': 309,\n",
       " '#recombinant interleukin-6#': 310,\n",
       " '#rel#': 311,\n",
       " '#ren#': 312,\n",
       " '#rerg#': 313,\n",
       " '#resistant starch#': 314,\n",
       " '#ribociclib#': 315,\n",
       " '#ridaforolimus#': 316,\n",
       " '#ripk2#': 317,\n",
       " '#rnase1#': 318,\n",
       " '#rnf2#': 319,\n",
       " '#ros1#': 320,\n",
       " '#rps3a#': 321,\n",
       " '#rps6#': 322,\n",
       " '#rps6kb1#': 323,\n",
       " '#rrad#': 324,\n",
       " '#runx1t1#': 325,\n",
       " '#saa4#': 326,\n",
       " '#sart3#': 327,\n",
       " '#sdhc#': 328,\n",
       " '#sds#': 329,\n",
       " '#sec14l2#': 330,\n",
       " '#selumetinib#': 331,\n",
       " '#setd2#': 332,\n",
       " '#sfrp2#': 333,\n",
       " '#sgcg#': 334,\n",
       " '#sgk3#': 335,\n",
       " '#si#': 336,\n",
       " '#sirt1#': 337,\n",
       " '#slc40a1#': 338,\n",
       " '#slc7a1#': 339,\n",
       " '#sltm#': 340,\n",
       " '#smad3#': 341,\n",
       " '#smad5#': 342,\n",
       " '#smim10l2a#': 343,\n",
       " '#snai1#': 344,\n",
       " '#snai2#': 345,\n",
       " '#son#': 346,\n",
       " '#sorafenib tosylate#': 347,\n",
       " '#sox2#': 348,\n",
       " '#sox7#': 349,\n",
       " '#spen#': 350,\n",
       " '#src#': 351,\n",
       " '#srebf1#': 352,\n",
       " '#stat1#': 353,\n",
       " '#stat3#': 354,\n",
       " '#stk11#': 355,\n",
       " '#supt16h#': 356,\n",
       " '#t#': 357,\n",
       " '#tamoxifen citrate#': 358,\n",
       " '#tamoxifen#': 359,\n",
       " '#tc2n#': 360,\n",
       " '#tcf7l2#': 361,\n",
       " '#telab1#': 362,\n",
       " '#temsirolimus#': 363,\n",
       " '#tenm1#': 364,\n",
       " '#tf#': 365,\n",
       " '#tfap2a#': 366,\n",
       " '#tfap2b#': 367,\n",
       " '#tg#': 368,\n",
       " '#tgfb2#': 369,\n",
       " '#th#': 370,\n",
       " '#thalidomide#': 371,\n",
       " '#tiparp#': 372,\n",
       " '#tnfrsf11a#': 373,\n",
       " '#tnfrsf11b#': 374,\n",
       " '#tnfsf13b#': 375,\n",
       " '#tnfsf14#': 376,\n",
       " '#tnip1#': 377,\n",
       " '#tnip3#': 378,\n",
       " '#tox#': 379,\n",
       " '#tox3#': 380,\n",
       " '#tp53#': 381,\n",
       " '#tra-tgc7-1#': 382,\n",
       " '#trametinib#': 383,\n",
       " '#trans sodium crocetinate#': 384,\n",
       " '#trastuzumab emtansine#': 385,\n",
       " '#trastuzumab#': 386,\n",
       " '#triciribine phosphate#': 387,\n",
       " '#twist1#': 388,\n",
       " '#uap1l1#': 389,\n",
       " '#ugt3a1#': 390,\n",
       " '#ulk1#': 391,\n",
       " '#vegfa#': 392,\n",
       " '#vgf#': 393,\n",
       " '#vinorelbine tartrate oral#': 394,\n",
       " '#vinorelbine#': 395,\n",
       " '#voxtalisib#': 396,\n",
       " '#wdtc1#': 397,\n",
       " '#wnt5a#': 398,\n",
       " '#xs#': 399,\n",
       " '#zbp1#': 400,\n",
       " '#zbtb8os#': 401,\n",
       " '#zfyve9#': 402,\n",
       " '#znf746#': 403,\n",
       " 'resist': 404,\n",
       " 'sensit': 405}"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col_idx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data has been processed and saved\n"
     ]
    }
   ],
   "source": [
    "df =data_frame_creator.from_papers_to_panda(path=path,\n",
    "                      doc_reader=tokenizer.tokenize_many_docs,\n",
    "                      tokenizer=tokenizer.tokenize,\n",
    "                      short_list=True,\n",
    "                      min_df=0.0000001,\n",
    "                      mat_out='vocab_matrix',\n",
    "                      vocab_out='vocabulary')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1428"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "non_empty_rows = sparse_matrix_functions.non_zero_rows(vocab_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sentences = [orig_sentences[i] for i in non_empty_rows]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Something that moves progressivley First thing on your resume Make it as good as possible Highlight your skills  Show complex skills in simple way Clearly defined purposed Has workflow  Translate very complex research onto very simple application Communication is very important Make sure you can get the data promptly  Project should be safe enough to work on the time frame',\n",
       " '13  Deregulation of the EGFR PI3K PTEN Akt mTORC1 pathway in breast cancer: possibilities for therapeutic intervention Nicole M. Davis1, Melissa Sokolosky1, Kristin Stadelman1, Stephen L. Abrams1, Massimo Libra2, Saverio Candido2, Ferdinando Nicoletti2, Jerry Polesel3, Roberta Maestro4, Antonino DAssoro5, Lyudmyla Drobot6, Dariusz Rakus7, Agnieszka Gizak7, Piotr Laidler8, Joanna Duliska-Litewka8, Joerg Basecke9, Sanja Mijatovic10, Danijela Maksimovic-Ivanic10, Giuseppe Montalto11,12, Melchiorre Cervello12, Timothy L. Fitzgerald13, Zoya N. Demidenko14, Alberto M. Martelli15, Lucio Cocco15, Linda S. Steelman1 and James A. McCubrey1 Department of Microbiology and Immunology, Brody School of Medicine at East Carolina University Greenville, NC 27858 USA 2 3 4 5 6 7 8 9 1  Department of Bio-Medical Sciences, University of Catania, Catania, Italy Unit of Epidemiology and Biostatistics, Centro di Riferimento Oncologico, IRCCS, Aviano, Italy Experimental Oncology 1, CRO IRCCS, National Cancer Institute, Aviano, Pordenone, Italy Department of Biochemistry and Molecular Biology, Mayo Clinic College of Medicine, Rochester, MN, USA Palladin Institute of Biochemistry, National Academy of Sciences of Ukraine, Kyiv, Ukraine Department of Animal Molecular Physiology, Institute of Experimental Biology, Wroclaw University, Wroclaw, Poland Chair of Medical Biochemistry, Jagiellonian University Medical College, Kraków, Poland Department of Medicine University of Göttingen, Göttingen, Germany Department of Immunology, Institute for Biological Research Sinisa Stankovic University of Belgrade, Belgrade, Serbia Biomedical Department of Internal Medicine and Specialties, University of Palermo, Palermo, Italy Consiglio Nazionale delle Ricerche,Istituto di Biomedicina e Immunologia Molecolare Alberto Monroy, Palermo, Italy Department of Surgery, Brody School of Medicine at East Carolina University, Greenville, NC Department of Cell Stress Biology, Roswell Park Cancer Institute, Buffalo, NY, USA Dipartimento di Scienze Biomediche e Neuromotorie, Università di Bologna, Bologna, Italy  10 11 12 13 14 15  Correspondence to: James A. McCubrey, email: mccubreyj@ecu.edu Keywords: Targeted Therapy, Therapy Resistance, Mutations, PI3K, mTOR, rapamycin Received: June 04, 2014 Accepted: July 11, 2014 Published: July 12, 2014 This is an open-access article distributed under the terms of the Creative Commons Attribution License, which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.',\n",
       " 'ABSTRACT The EGFR PI3K PTEN Akt mTORC1 GSK-3 pathway plays prominent roles in malignant transformation, prevention of apoptosis, drug resistance and metastasis.',\n",
       " 'The expression of this pathway is frequently altered in breast cancer due to mutations at or aberrant expression of: HER2, ERalpha, BRCA1, BRCA2, EGFR1, PIK3CA, PTEN, TP53, RB as well as other oncogenes and tumor suppressor genes.',\n",
       " 'In some breast cancer cases, mutations at certain components of this pathway (e.g., PIK3CA) are associated with a better prognosis than breast cancers lacking these mutations.',\n",
       " 'The expression of this pathway and upstream HER2 has been associated with breast cancer initiating cells (CICs) and in some cases resistance to treatment.',\n",
       " 'The anti-diabetes drug metformin can suppress the growth of breast CICs and herceptin-resistant HER2+ cells.',\n",
       " 'This review will discuss the importance of the EGFR PI3K PTEN Akt  mTORC1 GSK-3 pathway primarily in breast cancer but will also include relevant examples from other cancer types.',\n",
       " 'The targeting of this pathway will be discussed as well as clinical trials with novel small molecule inhibitors.',\n",
       " 'The targeting of the hormone receptor, HER2 and EGFR1 in breast cancer will be reviewed in association with suppression of the EGFR PI3K PTEN Akt mTORC1 GSK-3 pathway.',\n",
       " 'www.impactjournals.com oncotarget  4603  Oncotarget  \\x0cINTRODUCTION Breast cancer is a leading cause of cancer-related death in women.',\n",
       " 'This disease is diagnosed in nearly 1.4 million women and is responsible for more than 450,000 death every year [1].',\n",
       " 'According to the WHO release, there has been a 20% increase in the number of reported worldwide breast cancer patients which resulted in 522,000 deaths since 2008.',\n",
       " 'According to the US National Cancer Institute, approximately 232,000 occur resulting in about 40,000 deaths in the USA each year.',\n",
       " 'Breast cancer is not gender specific [2,3].',\n",
       " 'The frequency of breast cancer in men is approximately 100-fold lower than in women as approximately 2,200 males will be diagnosed with breast cancer each year in the USA.',\n",
       " 'Breast Cancer Genetics.',\n",
       " 'A prominent risk factor for the onset of breast cancer is age; however factors linked to lifestyle and diet also play important roles in breast cancer.',\n",
       " 'Mutations at or deregulation of certain genes (BRCA1, BRCA2, HER2, PIK3CA) and others play important roles in breast cancer [4-18].',\n",
       " 'Mutations or aberrant or deregulated expression of TP53, MDM2 and RB also can play roles in the therapeutic responses of breast cancer [19-26].',\n",
       " 'Restoration of functional TP53 activity can increase the sensitivity of some TP53 mutant cells to certain anticancer drugs [27].',\n",
       " 'BRCA Genes and Other Genes Involved in DNA Repair Are Implicated in Breast Cancer.',\n",
       " 'Breast cancer occurrence is attributed to both genetic and environmental factors.',\n",
       " 'Some breast cancers are due to hereditary mutations, namely those involving BRCA1 and BRCA2.',\n",
       " 'BRCA1 encodes breast cancer type 1 susceptibility protein which is involved in DNA repair and is considered a caretaker gene.',\n",
       " 'The BRCA1 protein interacts with RNA polymerase II and also with histone deacetylase complexes [28].',\n",
       " 'BRCA1 plays key roles in transcription, repair of breaks in double stranded DNA as well as ubiquitination.',\n",
       " 'The BRCA1 protein also combines with other proteins which detect DNA damage and other cell signals and forms a multi-subunit protein complex known as the BRCA1-associated genome surveillance complex (BASC) [29].',\n",
       " 'Components of this complex may be mutated in certain cancers.',\n",
       " 'BRCA2 is also involved in the repair of DNA double strand breaks [30].',\n",
       " 'BRCA2 binds single stranded DNA.',\n",
       " 'BRAC2 interacts with the RAD51 recombinase to stimulate strand invasion which is a critical step in homologous recombination.',\n",
       " 'For RAD51 to bind the DNA double-strand breaks, a complex of BRCA1 partner and localizer of BRCA2 (PALB2) BRCA2 is required [31].',\n",
       " 'The risk of developing breast or ovarian cancer www.impactjournals.com oncotarget 4604  in individuals with certain cancer-associated BRCA1  BRCA2 alleles is 60-80% for breast cancer and 2040% for ovarian cancer.',\n",
       " 'These individuals also develop cancer at an earlier age.',\n",
       " 'In addition, other genes involved in DNA repair and signaling are implicated in breast cancer including: Fanconi anemia (FA) genes (FANCD2, FANCA and FANCC), mismatch repair genes (MutL homolog 1 [MLH1], MutS protein homolog 2 [MSH2], PMS1 protein homolog 1 [PMS1], mutS homolog 6 [MSH6]), mismatch repair endonuclease PMS2 [PMS2] and DNA repair genes (Ataxia telangiectasia mutated [ATM], Ataxia telangiectasia and Rad3 related [ATR] and serine threonine-protein kinases Chk 2 (CHK1 2), and the tumor suppressor genes (TP53, Serine threonine kinase 11 [STK11] also known as liver kinase B1[LKB1], phosphatase and tensin homolog [PTEN]) and protein phosphatase 6 (PP6) [32-44].',\n",
       " 'In an important study with triple negative breast cancer (TNBC) patients, the frequency of BRCA1 and BRCA2 mutations and survival was examined [45].',\n",
       " 'DNA was isolated from tumor samples as well as normal tissues from 77 TNBC patients and the genetic sequence of the BRCA1 2 exons and flanking regions determined.',\n",
       " '19.5% of the TNBC patients had BRCA mutations, 15.6% were mutant at BRCA1, and 3.9% were mutant at BRCA2.',\n",
       " 'It turns out that the patients with BRCA mutations were younger than the patients with WT BRCA genes.',\n",
       " 'In this study which followed the patients for up to 214 months, there were 42.9% recurrences and 45.5% deaths.',\n",
       " 'Interestingly, the five-year recurrencefree survival estimates were associated with the genetic status of the BRCA genes.',\n",
       " 'As the five-year recurrence-free survival rates were 51.7% for patients with WT BRCA genes whereas they were 86.2% for patients with BRCA mutations.',\n",
       " 'BRCA1 and BRCA2 are also mutated in patients with ovarian cancer [46].',\n",
       " 'BRCA1 2 mutations are present in approximately 11 to 15% of unselected ovarian cancer patients.',\n",
       " 'BRCA1 mutations were positively associated with TP53 mutations.',\n",
       " 'The presence of BRCA1 2 mutations after platinum chemotherapy were associated with improved progression free survival.',\n",
       " 'Hereditary and Sporadic Breast and Ovarian Cancer.',\n",
       " 'Many spontaneous breast cancers are associated with environmental exposures to carcinogens [4761].',\n",
       " 'These include: air pollution [52], exposure to polychlorinated biphenyl congeners [53].',\n",
       " 'Deregulation of BRCA1 expression has been implicated in sporadic breast cancer.',\n",
       " 'The trinucleotideOncotarget  \\x0crepeat-containing 9 (TNRC9) gene has been shown to downregulate BRCA1 expression which results in breast cancer aggressiveness.',\n",
       " 'TNRC9 is amplified in certain breast cancer patients and is associated with a poor prognosis [62].',\n",
       " 'This group also determined that ectopic expression of TNRC9 affected breast cancer cell survival.',\n",
       " 'TNRC9 and BRCA1 protein expression were inversely correlated in large data sets of breast and ovarian cancer samples.',\n",
       " 'Interesting this group determined that TNRC9 bound to both the BRCA1 promoter and the cAMPresponsive element-binding protein (CREB) complex.',\n",
       " 'CREB is a regulator of BRCA1 transcription.',\n",
       " 'Finally TNRC9 expression suppressed BRCA1 expression by altering the methylation status of the BRCA1 promoter region.',\n",
       " 'BRCA mutations have also been detected in familial and sporadic ovarian cancer patients.',\n",
       " 'Germline mutations in BRCA1 or BRCA2 are present in approximately 18% of hereditary ovarian cancers.',\n",
       " 'These mutations confer an estimated risk from 15 to 50% in the ovarian cancer patients [63].',\n",
       " 'In this study, the prevalence of BRCA1 mutations in 106 familial Greek ovarian cancer patients who had a strong family history of ovarian cancer or metachronous breast cancer.',\n",
       " 'Metachronous breast cancer refers to a breast cancer patient which has two different breast cancers which occur at two different times, the two cancers can occur in the same breast.',\n",
       " 'In addition, the prevalence of BRCA1 mutations were examined in 592 sporadic Greek ovarian cancer patients.',\n",
       " 'In Greece, it had been previously determined that there were 6 types of BRCA1 mutations that accounted for 63% of all the mutations in the BRCA1 and BRCA2 genes.',\n",
       " 'Deleterious BRCA1 mutations were observed in 40.6% of familial ovarian cancer cases and 4.6% of sporadic ovarian cases.',\n",
       " 'This study determined that 71.2% of the BRCA1 carriers presented a high-grade serous phenotype.',\n",
       " 'These studies document the importance of identifying BRCA mutations in breast and or ovarian cancer families.',\n",
       " 'The authors have stated that all serous ovarian cancer patients should consider genetic testing.',\n",
       " 'Hereditary breast cancer often results from disruption of the normal functions of BRCA1 and BRCA2.',\n",
       " 'In contrast BRCA1 BRCA2 are not necessarily mutated in sporadic breast cancer, but there may be mutations in TP53 and epigenetic alterations which change the expression of other genes.',\n",
       " 'These changes result in breast cancer cells which may be wild WT at BRCA1 and BRCA2 but have defects in DNA repair.',\n",
       " 'In addition certain sporadic breast cancers may have hypermethylation of the BRCA1, estrogen receptor (ER), progesterone receptor (PR) and other genes which prevents or lowers their expression.',\n",
       " 'These changes may results in the breast cancer cells having a mutant BRCA phenotype which is referred to as BRCAness.',\n",
       " 'These breast cancers arise at early age and are aggressive [64].',\n",
       " 'In one study, the frequencies of methylation in the www.impactjournals.com oncotarget 4605  BRCA1 promoter region were examined in 96 sporadic invasive breast carcinomas and 43 sporadic ovarian carcinomas.',\n",
       " 'Methylation of the BRCA promoter region was detected in 11% and 5% of the breast and ovarian carcinomas respectively.',\n",
       " 'Methylation of the BRCA1 promoter was linked with lack of ER and PR expression in these tumors [65].',\n",
       " 'Alterations of Genes Involved in DNA Repair in Breast Cancer.',\n",
       " 'Sporadic breast cancers accounts for approximately 70% of breast cancers, while familiar breast cancers account for the remaining 30% which are due to the present of mutations in breast cancer families (familiar breast cancers) [66].',\n",
       " 'Familial breast cancer families have a higher incidence of breast cancers.',\n",
       " 'There exist different susceptibility genes which are high-, moderate-, and lowpenetrance susceptibility genes.',\n",
       " 'High-penetrance genes include: BRCA1, BRCA2, PTEN and TP53, which in many cases are responsible at least in part for the familiar breast cancer.',\n",
       " 'Other genes have been associated with moderate penetrance and risk.',\n",
       " 'These include genes involved in DNA repair, such as CHEK2, ATM, BRIP1 (FANCJ), PALB2 (FANCN) and RAD51C (FANCO).',\n",
       " 'In addition, genes involved in low penetrance and risk are being identified.',\n",
       " 'The presence of mutations at these different types of breast cancer susceptibility genes may be examined in breast cancer screening in the future.',\n",
       " 'The genetic structures of the TP53, BRCA1, ATM, and PIK3CA genes were examined in 145 Bulgarian patients with sporadic breast cancer.',\n",
       " 'The expression of HER2 was examined by immunohistochemistry and chromogenic hybridization in situ (CISH) [67].',\n",
       " 'In this study, mutations were observed: at TP53 in 22.07% of the patients, at BRCA1 in 0.69% of the patients, at ATM in 7.59% of the patients, and in 31.25% of the patients at PIK3CA.',\n",
       " 'Overexpression of HER2 was observed in 21.21% of the patients.',\n",
       " 'Mutations at TP53 were associated with both tumor size and grade of malignancy.',\n",
       " 'Mutations at ATM were associated with grade of malignancy.',\n",
       " 'Mutations at PIK3CA were associated with PR+ tumors.',\n",
       " 'HER2 overexpression was correlated with the age of the patient when diagnosed with breast cancer, tumor size and ER+.',\n",
       " 'This important study documented that TP53 mutations were an indicator for poorer outcome.',\n",
       " 'Importantly, in this study the presence of two genetic mutations did not correlate with either a more aggressive carcinoma or a poorer overall survival.',\n",
       " 'Chk2 is an important kinase which is activated in response to DNA damage.',\n",
       " 'Chk2 is involved both in TP53 and cell-cycle pathways.',\n",
       " 'In some Li-Fraumeni syndrome families, which have germline mutations at TP53, mutations at CHK2 have also be observed.',\n",
       " 'The presence of loss of heterozygosity (LOH) at the CHK2 gene was Oncotarget  \\x0cexamined in 139 breast cancer tumors [68].',\n",
       " '139 breast tumors were screened for loss of heterozygosity (LOH) at chromosome 22q (where the CHK2 gene is located), using seven microsatellite markers, LOH was detected in 53% of the breast tumors.',\n",
       " 'Further studies examined the mutational status of the CHK2 gene and a germ line variant (T59K) in the first coding exon was detected.',\n",
       " 'Upon screening 1172 cancer patients with different types of cancer for the T59K sequence variant, it was detected in four breast cancer patients.',\n",
       " 'This study concluded that CHK2 mutations were rare in breast cancer but the CHK2 gene product may perform a tumor suppressor function.',\n",
       " 'were treated with chemotherapy regmins without an anthracycline, that had the A2A2 genotype of the PIN3 (+16 bp) polymorphism, had a poorer overall survival than other genotypes.',\n",
       " 'Although some controversity exists on use of anthracyclines in breast cancer patients, those with the A2A2 genotype of PIN3 (+16bp) polymorphism may benefit.',\n",
       " 'These important studies document the importance of genetics in personalized medicine.',\n",
       " 'Epigenetic Modification of the ER genes in Breast Cancer.',\n",
       " 'Methylation of the promoter region of the ER and other genes has been associated with their decreased expression.',\n",
       " 'The methylation status of the ER-alpha promoter region was examined in 138 sporadic breast cancers.',\n",
       " 'The ER-alpha promoter region was observed to be methylated in 60.1% (83 138) tumors, including 57 of 69 of the tumors which did not express ER-alpha.',\n",
       " 'This study determined that the probability of ER-alpha promoter methylation was increased in those cases that were ERalpha- and PR- [72].',\n",
       " 'In a study of 100 sporadic primary breast cancers of which 51 were ER-alpha- and 49 ER-alpha+, ER methylation was observed in 98% of ER- and 65% of ER+ tumor samples.',\n",
       " 'ER- promoter region methylation was also associated with lack of PR expression and double receptor negative expression status of the breast cancer specimens [73].',\n",
       " 'The methylation of the ER-beta promoter region was examined in 178 sporadic breast cancer patients.',\n",
       " 'ER-beta promoter methylation was observed in 44.9% of breast tumor samples.',\n",
       " 'In contrast ER-beta promoter hypermethylation was detected in only 14.3% of patients with benign breast hyperplasia.',\n",
       " '58% of the ER-betatumors exhibited ER-beta promoter region methylation whereas 36.7% of the ER-beta+-positive cases exhibited methylation at the ER-beta promoter region.',\n",
       " 'As the levels of ER-beta promoter methylation increased- the levels of ER-beta protein detected decreased in the tumor samples.',\n",
       " 'A strong correlation between ER-alpha promoter methylation and ER-beta promoter methylation was observed [74].',\n",
       " 'Effects of c-Myc Overexpression in BRCA1Deficient Breast Cancer Patients.',\n",
       " 'Elevated c-Myc expression leads to a poor prognosis in sporadic breast cancer patients that are BRCA1-deficient [69].',\n",
       " 'In this important study, the presence of BRCA1 gene expression was examined in 374 sporadic breast cancer patients.',\n",
       " 'BRCA1 expression was lacking in 60.4% of the breast cancer patients.',\n",
       " 'Patients that lost BRCA1 expression often had more advanced breast cancer and were tumor node metastasis stage III positive, lymph node positive and overexpressed c-Myc.',\n",
       " 'The presence of hypermethylation of the BRCA1 promoter region was examined.',\n",
       " 'BRCA1 promoter hypermethylation was observed in 16.4% of the breast cancer patients examined and was associated with BRCA1-, ER- , c-Myc overexpression, and the triple-negative phenotype.',\n",
       " 'Thus loss of functional BRCA1 expression combined with increased expression of c-Myc was associated with a poor prognosis.',\n",
       " 'Alteration of TP53 in Breast Cancer.',\n",
       " 'Dysregulation of the TP53 gene also occurs in sporadic breast cancer [70].',\n",
       " 'The genetic structure of the TP53 gene was examined in 136 unselected sporadic breast cancer patients tumors.',\n",
       " 'Approximately 40% of the tumor samples had TP53 mutations.',\n",
       " 'Moreover, this group postulated that these mutations were due to exposure of the breast cancer patients to environmental mutagens based on the frequency of G-T transversions and the incidence of guanosine mutations on the non-transcribed DNA strand of the TP53 gene.',\n",
       " 'Polymorphisms of the TP53 gene have been observed to have prognostic and predictive values in cancer therapy.',\n",
       " 'The presence of two TP53 gene polymorphisms, Arg72Pro and PIN3 (+16 bp) was observed in ninety-four women with sporadic breast cancer who were followed for a mean of 67.9 months after therapy [71].',\n",
       " 'This study determined that the different genotypes of the Arg72Pro and PIN3 (+16 bp) polymorphisms had no significant impact on survival in the sporadic breast cancer patients.',\n",
       " 'However, this study observed that the patients which www.impactjournals.com oncotarget 4606  Expression of EGFR Family Members in Breast Cancer Patients.',\n",
       " 'The EGFR family consists of four members.',\n",
       " 'For the sake clarity in this review, we will refer to them as EGFR1 (a.k.a., EGFR, HER1, c-erbB1), HER2 (a.k.a., EGFR2, c-erbB2), EGFR3 (a.k.a., c-erbB3, HER3) and EGFR4 (a.k.a., c-erbB4, HER4).',\n",
       " 'The expression of the EGFR1, HER2, EGFR3 and EGFR4 were examined by immunohistochemistry in 220 breast cancer carcinomas [75].',\n",
       " 'Increased expression of EGFR1 was detected in Oncotarget  \\x0c16.4% of the tumors, increased expression of HER2 was observed in 22.8% of the tumors, increased expression of EGFR3 was detected in 17.5% of the tumors, and increased expression of EGFR4 was observed in 11.9% of tumors.',\n",
       " 'Breast cancer patients with tumors that overexpressed EGFR1, HER2 or EGFR3 had reduced survival.',\n",
       " 'In contrast, those breast cancer patients whose tumors displayed elevated levels EGFR4 had better survival than the breast cancer patient that expressed EGFR1, HER2 or EGFR3.',\n",
       " 'Thus overexpression of EGFR1, HER2 and EGFR3 was association with a poor prognosis in breast cancer patients.',\n",
       " 'In contrast, overexpression of EGFR4 is associated with a good prognosis.',\n",
       " 'In addition, this group investigated the association of ER expression with the different EGFR molecules.',\n",
       " 'Expression of EGFR1, HER2 and EGFR3 was associated with ER negativity in the breast cancer patients.',\n",
       " 'Moreover those breast cancer patients that were ER+ and also EGFR1+, HER2+ or EGFR3+ had poorer survival than those breast cancer patients which were either ER+ and HER2+ or EGFR4+.',\n",
       " 'Additional studies were performed by the same group on the proliferative potential of the different types of EGFR-expressing cells [76].',\n",
       " 'These important studies suggest that EGFR1, HER2 and EGFR3 are linked with tumor proliferation, while EGFR4 did not appear to drive proliferation but may even play protective roles.',\n",
       " 'In another study, the expression of the EGFR1 and EGFR3 proteins were examined in 104 primary breast carcinomas comprising nine comedo ductal carcinoma in situ (DCIS), 91 invasive ductal carcinomas and four invasive lobular carcinoma by histochemistry.',\n",
       " 'Increased expression of EGFR3 was observed in 67% of comedo DCIS, 52% of invasive ductal carcinomas, 71% of carcinomas containing both the in situ and invasive lesions and 25% of invasive lobular carcinomas [77].',\n",
       " '59% of ERtumors, 63% of lymph node+ tumors and 63% of HER2 tumors were positive for EGFR3 expression.',\n",
       " '67% of EGFR1+, 67% of HER2+ (67%), 75% TP53+ and 60% of cathepsin-D+ DCIS were positive for EGFR3 expression.',\n",
       " 'The expression of EGFR1, HER2, EGFR3 and EGFR4 was examined in 100 breast cancer patients [78].',\n",
       " 'By using immunohistochemistry techniques, 36% of the breast cancer samples were positive for EGFR1, 27%, of the breast cancer samples were positive of HER2, 26% of the breast cancer samples were positive for EGFR3 and 82% of the breast cancer samples were positive for EGFR4.',\n",
       " 'The expression of these genes was also examined by RT-PCR and the similar results were observed.',\n",
       " 'An association between decreased disease-free survival and expression of HER2 was observed.',\n",
       " 'Co-expression of EGFR1 and HER2 was associated with a worse prognosis.',\n",
       " 'In contrast, expression of EGFR4 was associated with a better outcome.',\n",
       " 'EGFR4 expression appeared to antagonize the effects on HER2 on clinical outcome in the breast cancer patients which expressed both.',\n",
       " 'The link between ER-alpha and EGFR4 expression www.impactjournals.com oncotarget 4607  was examined in 103 breast cancer samples by both immunohistochemistry and RT-PCR [79].',\n",
       " 'In 25% of the breast cancer samples ER-alpha were not expressed.',\n",
       " 'In addition, approximately 25% of them did not express EGFR4.',\n",
       " 'About one-half of the ER-alpha- tumors did not express EGFR4 at both mRNA and protein levels.',\n",
       " 'The luminal breast cancer cell lines MCF-7 and T47D cells expressed both ER-alpha and EGFR4.',\n",
       " 'While the basal breast cancer line MDA-MB-231 and the HER2overexpressing SK-BR-3 line expressed neither.',\n",
       " 'These results have suggested roles for EGFR4 in ER-alpha mediated signal transduction in breast cancer.',\n",
       " 'The expression patterns of EGFR1, HER2, EGFR3 and EGFR4 were examined by real-time RT-PCR in 365 unselected primary breast cancers [80].',\n",
       " 'EGFR1 and HER2 were negatively associated with ER+ and PR+ breast cancers.',\n",
       " 'In contrast, EGFR3 and EGFR4 were positively associated with ER+ and PR+ breast cancers.',\n",
       " 'In a subsequent study with the ER-alpha+ breast cancer cell line MCF-7, beta-estradiol down regulated the expression of all four EGFR family member receptors as determined by RT-PCR.',\n",
       " 'In contrast, the ER antagonist 4-hydroxy tamoxifen (4HT) inhibited the downregulation induced by beta-estradiol [81].',\n",
       " 'Gatekeeper Mutations in HER2.',\n",
       " 'The T798M mutation in HER2 is considered a gatekeeper mutation as it affects the ability of small molecule inhibitors such as lapatinib to bind the ATPbinding pocket and the mutation confers resistance to the inhibitor [82].',\n",
       " 'In studies by a different research group which were aimed at determining the mechanism of resistance of breast cancer cells containing the HER2 T798M mutation to lapatinib, it was shown that the cells overexpressed EGFR ligands [83].',\n",
       " 'In BT474 and MCF10A breast cancer cells transfected with the construct encoding HER2-T798M mutation, elevated HER2 kinase activity was detected and lapatinib did not block phosphorylation of HER2, EGFR3 or downstream Akt and ERK1 2.',\n",
       " 'Increased levels of EGFR2 associated with PI3Kp85 were detected in the transfected cells and the BT474 HER2T798M cells were also resistant to Herceptin.',\n",
       " 'However these cells were sensitive to the pan-phosphoinositol-3kinase (PI3K) inhibitors BKM120 and XL147 but not the mitogen-activated protein kinase kinase1 2 (MEK1 2) inhibitor CI-1040.',\n",
       " 'The BT474 HER2-T798M transfected cells expressed elevated levels of the EGFR ligands EGF, TGF-alpha, amphiregulin (AR), and heparin binding EGF (HB-EGF).',\n",
       " 'Thus targeting breast cancers with both HER2 inhibitors and PI3K inhibitors may be an appropriate technique to treat those breast cancer patients with HER2T798M mutations [84].',\n",
       " 'Other growth factors also bind the EGFR family of receptors including: EGF, schwannomaderived growth factor (SDGF), vaccinia growth factor (VGF), Neu differentiation factor (NDF) or neuregulins, Oncotarget  \\x0cand heregulin.',\n",
       " 'A diagram of EGFR family members is presented in Figure 1.',\n",
       " 'Mutations at the EGFR Gene Family in Breast Cancer.',\n",
       " 'HER2 is amplified in 20-25% of breast cancers.',\n",
       " 'However, the roles of mutations amplifications of other EGFR family members are not so clear.',\n",
       " 'Mutations and amplifications of EGFR1 have been detected in breast cancer [86, 87].',\n",
       " 'In a study consisting of 70 TNBC patient samples, mutations in the kinase domain were detected in 11% of the samples.',\n",
       " 'This study amplified the region spanning exons 18 to 21 of the EGFR1 gene.',\n",
       " 'The investigators detected deletions in exon 19, which encodes part of the kinase domain.',\n",
       " 'However, the authors indicated that these mutations appeared to be independent of the expression levels of the EGFR1 protein that were detected by immunohistochemistry.',\n",
       " 'A diagram illustrating some of the effects of mutations amplifications in EGFR1 HER2 and other genes in breast and other cancers on signal transduction pathways is presented in Figure 2.',\n",
       " 'In addition, amplification of the EGFR1 gene has been detected in certain breast cancers [88].',\n",
       " 'The EGFR1 gene was amplified in approximately 6% of the breast cancer patients they examined (n =175) and the samples displayed increased EGFR1 expression.',\n",
       " 'They also examined patient samples which had amplified the EGFR1 gene for hot-spot mutations in the EGFR1 gene.',\n",
       " 'No mutations were detected in exons 19 and 21.',\n",
       " 'The presence of EGFR1 or HER2 gene amplification or overexpression was investigated in a series of metaplastic breast carcinomas (MBC) [89].',\n",
       " 'MBC are basal like tumors that account for less than 1% of all invasive mammary carcinomas.',\n",
       " 'Nineteen of the 25 (76%) MBC examined overexpressed EGFR1.',\n",
       " 'EGFR1 gene amplifications were detected in 37% of the tumors which overexpressed EGFR1.',\n",
       " 'In contrast, only one case exhibited HER2 overexpression but HER2 gene amplification was not detected in this study of MBC.',\n",
       " 'The authors pointed out that some of the tumors that overexpressed EGFR1, but did not have EGFR1 gene amplification, the expression of EGFR1 could have resulted from activating mutations in EGFR1, which resulted in its expression.',\n",
       " 'In a subsequent study by the same group, the presence of EGFR1 amplification and activating mutations was examined in 47 MBC [90].',\n",
       " '32% of the samples exhibited overexpression of EGFR1.',\n",
       " 'Within the subset that had EGFR overexpression, 34% had EGFR1 gene amplification.',\n",
       " 'This group examined these MBC samples for the presence of activating mutations in exons 18, 19, 20, and 21 of the EGFR1 gene.',\n",
       " 'No activating mutations of EGFR1 in exons 18, 19, 20 and 21 were detected in these MBC patient samples.',\n",
       " 'The epidermal growth factor receptor variant III (EGFRvIII) is a genetic truncation of the EGFR1 gene.',\n",
       " 'It encodes a constitutively-active truncated EGFR1 protein which has been implication in many types of cancers, (e.g., brain, breast, prostate and others) [91-103].',\n",
       " '4608 Oncotarget  Roles of Aberrant mRNA Splicing in Sensitivity to Herceptin.',\n",
       " 'The splice variant delete16HER2, which results from exon 16 skipping, has been shown to increase the transformation frequency of cancer cells.',\n",
       " 'This missplicing results in resistance to Herceptin.',\n",
       " 'In contrast, retention of intron 8 of the HER2 gene after mRNA splicing results in the creation of herstatin which inhibits tumor cell proliferation.',\n",
       " 'Likewise retention of intron 15 during splicing of HER2 results in the p100 protein which also suppresses tumor cell proliferation [85].',\n",
       " 'Figure 1: Epidermal Growth Factor Receptor Family.',\n",
       " 'Conserved domains of the four different EGFR family members are indicated by similar shading.',\n",
       " 'L = ligand binding domain, CR = cysteine-rich domains.',\n",
       " 'TM = transmembrane domain.',\n",
       " 'CT = C-terminal domain which contains the phosphorylation sites.',\n",
       " 'JM = juxta membrane domain.',\n",
       " 'HER2 does not bind a ligand.',\n",
       " 'The kinase domain in ERB3 is defective.',\n",
       " 'Alternative names for each receptor are written underneath each receptor.',\n",
       " 'Ligands which bind the receptors are indicated in purple circles above the receptors.',\n",
       " 'Epidermal growth factor (EGF), vaccinia growth factor (VGF), amphiregulin (AR), heparin binding-EGF (HB-EGF), schwannoma-derived growth factor (SDGF), Neu differentiation factor (NDF), heregulin, neuregulin.',\n",
       " 'Arrows between receptors indicate possible heterodimer formation between various EGFR family receptors.',\n",
       " 'www.impactjournals.com oncotarget  \\x0cIntroduction of a construct encoding EGFRvIII into MCF-7 breast cancer cells resulted in HER2 phosphorylation, which the authors suggested occurred through heterodimerization and cross-talk [91].',\n",
       " 'The MCF7 EGFRvIII transfectants had approximately a 3-fold increase in colony formation.',\n",
       " 'MCF-7 EGFRvIII were more tumorigenic that MCF-7 cells in athymic nude The EGFRvIII protein is detected in various human cancers, but not in normal tissues.',\n",
       " 'The presence of EGFRvIII mRNA was examined in primary invasive breast cancer by utilizing laser capture microdissection (LCM) RT-PCR.',\n",
       " 'A high incidence (67.8%) of EGFRvIII transcripts was observed in pure breast cancer cells [92].',\n",
       " 'Furthermore, 57.1% of the infiltrating breast carcinomas expressed both EGFR1 WT and EGFRvIII mRNA in the same tumor.',\n",
       " 'In contrast, no detectable EGFRvIII mRNA was seen in samples of normal breast tissue.',\n",
       " 'These expression results were further confirmed by immunohistochemical analysis.',\n",
       " 'Thus there was co-expression of EGFRvIII and EGFR WT in some human invasive breast cancer tissue but not in normal breast samples.',\n",
       " 'It must be pointed out that other studies consisting of 55 breast cancer cell lines and 170 primary breast cancer did not observe similar results and the authors concluded that expression of EGFRvIII is extremely rare in breast cancer [93].',\n",
       " 'However, in a study by a different research group, the expression of EGFRvIII mRNA in women with breast cancer was examined by an RT-nested PCR.',\n",
       " 'EGFRvIII mRNA was detected in the peripheral blood of 30% of 33 low risk, early stage patients, in 56% of 18 patients selected for neoadjuvant chemotherapy, in 63.6% of 11 patients with disseminated disease but not in any of 40 control women [94].',\n",
       " 'Interesting in the low risk, early stage patients, EGFRvIII expression was associated with ER- or HER2+.',\n",
       " 'In another study of 225 breast cancer patients, the expression of EGFR1, phosphorylated EGFR1, and EGFRvIII, was examined by immunohistochemistry and the patient outcomes were also followed [95].',\n",
       " '48% of the patients displayed EGFR1, 54% of the patients were positive for phospho-EGFR, and 4% of the patients were positive for EGFRvIII.',\n",
       " 'EGFR1 expression correlated with negative hormone receptor status, worse relapse-free survival and overall survival than those patients with did not express detectable EGFR1 [95].',\n",
       " 'Interestingly there did not appear to be any association between expression of phosphoEGFR or EGFRvIII expression with clinical outcome.',\n",
       " 'This study indicated that prognostic value of EGFR1 expression was most important in the HER2+ and the ER-  PR- subgroups.',\n",
       " 'EGFRvIII expression may down regulate PR expression in certain luminal B tumors [96].',\n",
       " 'These tumors are ER+ but are characterized as having an aggressive behavior which is 4HT resistant.',\n",
       " 'This subset of breast cancers displays increased EGFR1, HER2 and downstream PI3K PTEN Akt mTORC1 pathway activation [96,97].',\n",
       " 'www.impactjournals.com oncotarget 4609  EGFRvIII interacts with HER2 [98] and chemokine (C-X-C motif) receptor 4 (CXCR4) [99,100] to activate signaling pathways important in migration, invasion and tumorigenesis.',\n",
       " 'These interactions between EGFRvIII and HER2 and CXCR4 may be prolonged in comparison to interactions between EGFR1 and HER2 and CXCR4 as it is more difficult to down regulate the constitutive nature of EGFRvIII.',\n",
       " 'CXCR4 is highly expressed in breast cancers and implicated in metastasis [101] and cancer initiating cells (CICs) [102].',\n",
       " 'Recently EGFRvIII has been implicated in primary breast cancers and breast CICs.',\n",
       " 'Its expression has been associated with the Wnt beta-catenin pathway and downstream beta-catenin target gene expression and the expression of genes associated with self-renewal.',\n",
       " 'Its expression has been linked with increased in vitro mammosphere formation and tumor formation [103].',\n",
       " 'HER2 normally has to heterodimerize with another EGFR family member for activity.',\n",
       " 'However, if the HER2 gene is amplified, HER2 activity is induced and the abnormal breast tumor growth is dependent on HER2 activity for growth [104].',\n",
       " 'An activating mutation has been detected in HER2+ lung cancer patients in the germline from a Japanese cancer patient with a familiar history of lung cancer as well as sporadic cancer patients [105].',\n",
       " 'The mutation occurred in the transmembrane region and may be responsible for increased stability and dimerization.',\n",
       " 'This mutation activated downstream HER2 signaling molecules including Akt and p38MAPK.',\n",
       " 'Somatic mutations in the kinase domain of the HER2 gene, which result in its activation have been detected in lung cancer patients [106].',\n",
       " 'In a study involving non small cell lung cancers (NSCLC), HER2 mutations were detected in 1.6% (11 of 671) of NSCLC cancer specimens examined but were absent in other types of cancers.',\n",
       " 'The HER2 mutations were in-frame insertions in exon 20 corresponding to a similar region as in the EGFR gene where insertions were detected in NSCLC patients.',\n",
       " 'Interestingly the HER2 mutations were more frequent in the lung cancer patients that never smoked.',\n",
       " 'HER2 mutations were detected more frequently in patients of Oriental ethnicity than other ethnicities.',\n",
       " 'The mutations were also more frequently detected in females.',\n",
       " 'Recently mutations in HER2 have been detected in breast cancer patient samples which lack HER2 gene amplification.',\n",
       " 'Thirteen HER2 mutations were characterized from twenty-five patient samples which had HER2 mutations but lacked HER2 gene amplification.',\n",
       " '7 mutations were activating and resulted from point mutations and in-frame deletions.',\n",
       " 'Some mutations (L755S) resulted in lapatinib resistance; however this was not an activating mutation.',\n",
       " 'All of the cells containing the HER2 mutations were sensitive to the irreversible HER2 kinase inhibitor, neratinib [107].',\n",
       " 'Loss of HER2 activity also results in loss of phosphorylated EGFR3.',\n",
       " 'It has been shown that HER2 Oncotarget  \\x0ccan dimerize with EGFR3 to drive breast cancer proliferation [104].',\n",
       " 'Loss of EGFR3 activity in HER2+ breast cancer cells inhibited their growth.',\n",
       " 'This growth inhibition mediated by loss of HER2 or EGFR3 and could be overcome by introduction of a construct encoding activated Akt.',\n",
       " 'Importantly this group demonstrated that a key function of EGFR3 was to couple the response of HER2 to the PI3K PTEN Akt mTORC pathway.',\n",
       " 'Less is known about the roles of EGFR3 and EGFR4 in breast cancer [108].',\n",
       " 'EGFR3 is naturally kinase-inactive.',\n",
       " 'However it interacts with the EGFR family members and serves to transduce signals to the PI3K PTEN Akt  mTORC1 pathway.',\n",
       " 'Since EGFR3 is kinase-inactive, it will be more difficult to isolate inhibitors specific to it.',\n",
       " 'However, it is still important and it can serve in pathways responsible for drug resistance [109].',\n",
       " 'The functions of EGFR4 in breast and other cancers have been investigated and recently summarized [110].',\n",
       " 'Similar to the EGFR1 gene, EGFR4 encodes a protein contains an extracellular ligand-binding domain, a hydrophobic transmembrane domain, an intracellular tyrosine kinase domain, and carboxyl-terminal tyrosine residues [110,111].',\n",
       " 'These carboxyl-terminal tyrosine residues are phosphorylated after ligand binding and allows coupling of the receptor with other signaling molecules [111].',\n",
       " 'Ligand binding to EGFR4 stimulates either homodimerization or heterodimerization of EGFR4 with another EGFR family member.',\n",
       " 'Activation of EGFR4 signaling results in cleavage and release of the EGFR4 cytoplasmic domain from the membrane, which then may traffic to the nucleus and mitochondria to exert additional biological effects [110,112,113].',\n",
       " 'The roles of EGFR4 in cancer are complex.',\n",
       " 'In some cancers (larynx, pancreatic and prostate) EGFR4 may function as a tumor suppressor [110,114-116].',\n",
       " 'EGFR4 expression in breast, cervical and ovarian cancers is associated with a favorable prognosis [110, 113,117,118].',\n",
       " 'The EGFR4 Q646C mutant results in an EGFR4 protein which undergoes ligand-independent homodimerization and tyrosine phosphorylation which interestingly suppresses colony formation of breast, pancreatic and prostate cell lines [110, 119-121].',\n",
       " 'Introduction of the constitutively active EGFR4 I658Q mutant into breast, ovarian and prostate cell lines induces apoptosis [122].',\n",
       " 'However, in some cell types and biological situations EGFR4 has oncogenic activities [120].',\n",
       " 'EGFR4 is overexpressed in certain cancers including ependymomas and medulloblastomas [111,116,123].',\n",
       " 'Overexpression of EGFR4 in lung carcinomas results in increased proliferation [116.123].',\n",
       " 'Furthermore, overexpression of HER2, EGFR4, and the EGFR4 ligand NRG1-beta in medulloblastoma results in increased metastasis [124].',\n",
       " 'Likewise over-expression of EGFR4 in conjunction with EGFR1 and HER2 results in breast cancers correlates with poor prognosis [125].',\n",
       " 'In contrast overexpression of EGFR4 by itself results in a more www.impactjournals.com oncotarget 4610  favorable outcome.',\n",
       " 'Silencing endogenous EGFR4 expression in ER+ MCF7 and T47D breast tumor cell lines reduces anchorage-independent proliferation stimulated by an EGFR4 ligand such as neuregulin-2beta [110, 126].',\n",
       " 'HER2 tyrosine kinase activity, rather than EGFR4 tyrosine kinase activity was required for neuregulin-2beta to stimulate cell proliferation [110].',\n",
       " 'Interesting the sites of EGFR4 tyrosine phosphorylation, but not the sites of HER2 phosphorylation, were required for neuregulin2beta to regulate cell proliferation [110].',\n",
       " 'Thus the roles of EGFR4 in cancer remain complex and are influenced by the expression of other EGFR family members.',\n",
       " 'Neurofibromin 1 and GTPase Activating Proteins (GAP).',\n",
       " 'Neurofibromin 1 (NF1) is a tumor suppressor gene.',\n",
       " 'It encodes a GTPase which normally serves to regulate Ras signaling.',\n",
       " 'It is mutated in neurofibromatosis patients.',\n",
       " 'These patients have elevated (constitutive) Ras activation.',\n",
       " 'The tumor suppressor NF1 has been implicated in sporadic breast cancer.',\n",
       " 'Recent studies have suggested that NF1 is a breast cancer driver gene.',\n",
       " 'NF1 is deleted or mutated in 27.7% of all breast carcinomas [127].',\n",
       " 'In a study of inbred Chaos3 mice, investigators noted high levels of genetic instability which can lead to mammary tumors.',\n",
       " 'The genomically-characterized mammary adenocarcinomas from these mice displayed deletions of certain genes, and the NF1 gene was deleted in the vast majority of the mouse mammary adenocarcinomas.',\n",
       " 'These mammary adenocarcinomas exhibited constitutive Ras hyperactivation and sensitivity to Ras pathway inhibitors.',\n",
       " 'Also alternative mRNA splicing events can give rise to different NF1 isoforms.',\n",
       " 'NF1 and Ras expression were examined in 22 sporadic breast cancers, 18 benign lesions and 6 normal breast tissues by tissue microarrays.',\n",
       " 'NF1 and CELF3-6 RNA expression was examined by RT-PCR in the breast samples.',\n",
       " 'NF1 and Ras expression displayed no difference in expression when examined by immunohistochemistry assays.',\n",
       " 'In contrast, NF1 isoforms were determined to shift from the type II mRNA isoform in normal breast in normal breast cancer to the type I mRNA isoform in breast carcinoma.',\n",
       " 'The authors did not detect a shift in CELF mRNA cofactor expression that was related to the shift in NF1 mRNA isoforms.',\n",
       " 'The authors suggest that there is a NF1 isoform shift in expression from type II to type I which could be important in the development and progression of sporadic breast cancer [128].',\n",
       " 'NF1 normally serves to regulate Ras and thus is implicated in the regulation of both the PI3K PTEN  Akt mTORC1 and Raf MEK ERK pathways.',\n",
       " 'Successful targeting of Ras may improve the therapy of patients with NF1 mutations.',\n",
       " 'Alternatively, these patients may Oncotarget  \\x0cbe sensitive to combined treatment with MEK and PI3K pathway inhibitors.',\n",
       " 'Genes that serve to regulate the Ras gene may be tumor suppressors as when they are mutated the Ras pathway is turned on.',\n",
       " 'The RasGAP2 gene (RASAL2) is one such gene, it is a tumor and metastasis suppressor which is mutated or suppressed in breast cancer.',\n",
       " 'Inactivation of RASAL2 was shown to be associated with tumor growth, progression and metastasis in animal models.',\n",
       " 'In human RASAL2 loss was associated with metastatic disease and decreased expression of RASAL2 was associated with the recurrence of luminal B breast tumors [129].',\n",
       " 'The methylation state of the Ras association domaincontaining protein 1 (RASSF1A) gene was examined in 36 breast cancer patients in breast cancer tissue as well as their adjacent normal tissues [130].',\n",
       " 'RASSF1A may function as a tumor suppressor gene.',\n",
       " 'Methylation of the RASSF1A gene was detected in 61.1% of the breast tissues but not in their normal adjacent tissues.',\n",
       " 'The methylation of the RASSF1A gene resulted in deceases in mRNA and protein levels of 33.3 and 44.4% respectively.',\n",
       " 'In contrast, the RASSF1A protein was detected in normal tissues.',\n",
       " 'Methylation of the RASSF1A gene did not appear to be associated with clinical parameters, such as age, histological types, TNM stages and lymph node metastases.',\n",
       " 'RAS Gene Family Mutations Alterations in Breast Cancer.',\n",
       " 'In a genetic study examining the mutational status of the Ras and PI3K pathway genes in 40 breast cancer cell lines, mutations were detected at approximately 25% in Ras pathway members (KRAS, HRAS, NRAS, and BRAF) and 54% of PI3K pathway members (PTEN, PIK3CA).',\n",
       " 'However, unlike the mutational status of these two families in colo-rectal cancer, this study did not detect mutation in both pathway family genes in single cell lines very frequently [131].',\n",
       " 'In subsequent studies by the same group with 41 breast cancer cell lines, this group found 146 mutations among twenty-seven cancer causing genes, which resulted in an average of 3.6 mutations per cell line.',\n",
       " 'Mutations in TP53, RB and PI3K pathways were frequently detected in the breast cancer cell lines.',\n",
       " 'Importantly these investigators could identify mutational profiles that were associated with luminal-type and basaltype breast cancer cell lines.',\n",
       " 'The luminal mutational profile included E-cadherin (CDH1) and mitogen-activated protein kinase kinase 4 (MAP2K4 a.k.a.',\n",
       " 'MEK4) genes and amplifications of the cyclin D1, (CCND1), HER2 and mouse double minute 2 homolog (MDM2), while the basal mutational profile included: BRCA1, RB1, RAS and BRAF gene mutations and deletions of tumor suppressors p16 (CDKN2A) and p14ARF (CDKN2A a.k.a.',\n",
       " 'Activation of the Ras pathway is often detected in breast cancer in the absence of mutations of the RAS genes www.impactjournals.com oncotarget 4611  [133].',\n",
       " 'Gene expression signature analysis has revealed that Ras pathway dependence can predict the sensitivity to inhibitors targeting the Raf MEK ERK and PI3K PTEN  Akt mTORC1 pathways.',\n",
       " 'The Ras pathway activation is associated with sensitivity to MEK inhibitors but resistance to Akt inhibitors in breast and lung tumors.',\n",
       " 'The Ras pathway signature is a better indicator of Ras pathway dependency than mutations at KRAS, as there can be many genetic mutations which can result in Ras pathway dependency (e.g., upstream receptors, EGFR1 and others as well as mutations in BRAF and other downstream signaling molecules).',\n",
       " 'The Ras pathway signature was determined to be activated in breast cancers which were sensitive to MEK inhibitors but resistant to Akt inhibitors.',\n",
       " 'The Ras pathway signature was shown to be increased in ER- breast cancers and lung adenocarcinomas.',\n",
       " 'The Ras pathway signature also predicts resistant to the EGFR1-targeting agent cetuximab (Erbitux) in metastatic colorectal cancer [134].',\n",
       " 'Upon novel global gene expression profiling on 47,293 gene transcripts in 128 invasive breast cancers, the Ras-like, estrogen-regulated, growth-inhibitor (RERG) gene was determined to be a key marker of the luminal BC class and could be used to separate distinct prognostic subgroups [135].',\n",
       " 'These observations were further explored by performing immunohistochemistry on tissue microarrays containing 1,140 invasive breast cancers [135].',\n",
       " 'These results showed that the RERG gene is one of the highest ranked genes to differentiate between ER+ luminal and ER- non-luminal cancers.',\n",
       " 'RERG expression was positively associated with the following markers of luminal differentiation: ER+, the cytokeratins (CK7 8, CK18 and CK19) and FOXOA1.',\n",
       " 'RERG expression was also associated with other markers of good prognosis namely, small size, lower histologic grade an d AR, BRCA1, fragile histidine triad protein (FHIT, Bis(5-adenosyl)-triphosphatase), and p21Waf-1 and p27Kip-1 and inversely associated with the proliferation markers MIB1 (a monoclonal antibody which is directed to a different epitope on the Ki67 protein than the original Ki67 antibody) and TP53.',\n",
       " 'Importantly RERG expression was associated with longer survival.',\n",
       " 'The Rab-coupling protein RCP (a.k.a RAB11FIP1), is located at a chromosomal region frequently amplified in breast cancer.',\n",
       " 'Introduction of constructs encoding RCP into normal human mammary epithelial cells MCF-10A cells lead to cells with tumorigenic properties.',\n",
       " 'Likewise knock-down of RCP in breast cancer cell lines decreased the tumorigenic properties of the cells.',\n",
       " 'Therefore RCP is a frequently amplified gene in breast cancer  and suggest roles for the Rab family in carcinogenesis [136].',\n",
       " 'Oncotarget  \\x0cInteractions between Ras and Bmi-1 in Breast Cancer.',\n",
       " 'The B-lymphoma Moloney murine leukemia virus insertion region-1 (Bmi-1) gene functions in stem cell maintenance.',\n",
       " 'It is a member of the polycomb group of transcription repressors.',\n",
       " 'Bmi-1 exerts it effects by suppression of the p16Ink4A ARF tumor suppressor.',\n",
       " 'Increased expression of Bmi-1 is detected in many cancers.',\n",
       " 'The effects on Bmi-1 overexpression were examined on MCF-10A cells.',\n",
       " 'While Bmi-1 overexpression by itself did not result in the oncogenic transformation of MCF-10A cell, co-expression of activated H-Ras (RasG12) resulted in the oncogenic transformation of MCF-10A cells.',\n",
       " 'The Bmi-1 H-Ras transformed cells exhibited properties of cells that had undergone the epithelial to mesenchymal transition (EMT).',\n",
       " 'Bmi-1 inhibited senescence and allowed the proliferation of cells expressing high levels of activated H-Ras [137].',\n",
       " 'Subsequent studies by the same group have indicated that knock-down of Bmi-1 in breast cancer cell lines decreased their aggressive nature in vivo in tumor transplant studies [138].',\n",
       " 'Mutations in Components of the PI3K PTEN Akt mTOR Pathway in Breast Cancer.',\n",
       " 'In a study which examined MBC, PIK3CA mutations were detected in 47.4% of the cancers which were aggressive and also chemoresistant.',\n",
       " 'In contrast, PI3KCA mutations were detected in 34.5% of hormone receptor-positive cancers, 22.7% of 75 HER2-positive cancers, 8.3% of basal-like cancers and none of claudinlow tumors examined in this study.',\n",
       " 'MBCs and claudin-low breast cancer subsets displayed enrichment for markers linked to stem cell function and EMT.',\n",
       " 'It was postulated that MBCs and claudin-low tumors are enriched with CICs and may arise from an earlier, more chemoresistant breast epithelial precursor than either basal-like or luminal cancers.',\n",
       " 'However in this study, no mutations at PIK3CA were detected in the claudin-low cancers, so the roles of PIK3CA mutations in these cancers are not clear [139].',\n",
       " 'In a study with 547 human breast cancer patient samples and 41 established cell lines, the mutational status of PIK3CA, AKT and PTEN were analyzed as well as the effects of pathway mutations on the sensitivity to PI3K inhibitor LY294002 [140].',\n",
       " 'PIK3CA mutations were determined to be more frequent in ER+ (34.5%) and HER2+ (22.7%) than in basal-like tumors (8.3%).',\n",
       " 'AKT1 (1.4%) and PTEN (2.3%) mutations were determined to be restricted to ER+ breast cancers.',\n",
       " 'Interestingly cells with PIK3CA mutations were less sensitive to the PI3K inhibitor LY294002 inhibitor than breast cancer cells which had loss of PTEN activity.',\n",
       " 'Thus, PI3K pathway aberrations likely play a distinct role in the pathogenesis of different breast cancer subtypes.',\n",
       " 'The specific aberration www.impactjournals.com oncotarget 4612  present may have implications for the selection of PI3Ktargeted therapies in hormone receptor-positive breast cancer.',\n",
       " 'In a study which examined the genetic structure of the PIK3CA gene in 452 breast cancer patients, PIK3CA mutations were observed in 33.4% of the breast cancer patients [141].',\n",
       " 'PIK3CA mutations were more frequently detected in ERalpha+ and PR+ breast cancers (41.1%), than in TNBCs (ER-,PR-, HER2-) (12.5%.).',\n",
       " 'Patients which had PIK3CA mutations had a longer metastasis-free survival period than the overall population.',\n",
       " 'The mutational status of the PIK3CA gene was examined in eighty HER2+ breast cancer patients as well as clinical outcome of the patients after Herceptin treatment [142].',\n",
       " 'The PIK3CA gene was determined to be mutated in 21.3% of HER2+ breast cancer patients that had been treated with herceptin for one year.',\n",
       " 'Improved disease free survival was observed in those patients with WT PIK3CA as opposed to those patients with mutant PIK3CA.',\n",
       " 'The PIK3CA gene is also mutated in some ovarian cancer patients [143].',\n",
       " 'PI3K-p110 (PIK3CA) protein expression was examined in 1,394 early stage breast cancer samples.',\n",
       " 'Elevated PI3K-p110 was associated with the basal-like breast cancers, HER2+ breast cancer, and triple negative non-basal breast cancers.',\n",
       " 'In contrast, the luminal class of breast cancers had reduced levels of PI3K-p110 in comparison to the other classes.',\n",
       " 'PI3K-p110+ breast cancer patients had shorter disease free survival.',\n",
       " 'Thus these studies demonstrated that PI3K-p110 expression is a biomarker associated with poor prognosis in breast cancer [144].',\n",
       " 'Interactions between PIK3CA Mutations and HER2 Amplification in Breast Cancer.',\n",
       " 'Introduction of genetic constructs containing PIK3CA mutations E545K and H1047R into MCF-10A human mammary epithelial cells that also overexpress HER2 conferred enhanced growth properties to MCF10A  HER2 cells in comparison to cells which lacked the introduced mutant PIK3CA gene [145].',\n",
       " 'Upon introduction of mutant PIK3CA (H1047) into MCF-10A cells which overexpress HER2, the expression of EGFR3 EGFR4 ligand heregulin (HRG) was detected.',\n",
       " 'In contrast, introduction of mutant PIK3CA (E545K) gene into the MCF-10A cells which overexpressed HER2 did not result in the expression of HRG.',\n",
       " 'Silencing HRG with siRNA inhibited the growth of MCF-10A HER2 PIK3CAH1047R cells but not MCF-10A HER2 PIK3CA-E545Kexpressing cells.',\n",
       " 'The HRG siRNA synergized with the HER2 inhibitors herceptin (trastuzumab) and lapatinib.',\n",
       " 'Treatment of the cells with a PI3K inhibitor (BEZ235) suppressed HRG and P-AKT levels.',\n",
       " 'When the cells were cotreated with BEZ235 and lapatinib, a complete Oncotarget  \\x0csuppression of growth was observed.',\n",
       " 'These important result document interactions between HER2 and certain PIK3CA mutations and point to the possible co-targeting of HER2 and PI3K in certain breast cancers.',\n",
       " 'Deregulation of the PI3K PTEN Akt mTORC1 pathway by gene mutations has been estimated to occur in >70% of breast cancers [146].',\n",
       " 'In ER+ breast cancers, activation of the PI3K PTEN Akt mTORC1 pathway can result in both estrogen-dependent and estrogenindependent ER activity and contribute to estrogenindependence and potentially loss of sensitivity to hormonal based therapies.',\n",
       " 'Activation of the PI3K PTEN  Akt mTORC1 pathway can also result in resistance to HER2 inhibitors in HER2+ cells.',\n",
       " 'Inhibition of the PI3K  PTEN Akt mTORC1 pathway can overcome resistant to hormonal and anti-HER2 targeted therapies [146].',\n",
       " 'Encouraging results have been obtained in breast cancer clinical trials with combinations of various inhibitors.',\n",
       " 'These results predict that combinations of HER2, PI3K, mTORC2 inhibitors and hormonal based therapeutics may be appropriate for the treatment of certain breast cancer patients which are resistant to current therapies.',\n",
       " 'The transforming activity of mutant PI3K has been linked to its ability to bind phosphorylated YXXM motifs present in activated receptor tyrosine kinases (RTK) or adaptor molecules, potentially in association with the PI3K-regulatory subunit, p85.',\n",
       " 'EGFR3 is a potent activator of the PI3K PTEN Akt mTORC1 pathway.',\n",
       " 'The requirement for EGFR3 in PI3K-mediated transformation of mammary epithelium was investigated [147].',\n",
       " 'Conditional loss of EGFR3 in mammary epithelium lead to a delay in PI3K-H1047R-mediated mammary hyperplasia.',\n",
       " 'However, these studies demonstrated that tumor latency and PI3K signaling were not perturbed.',\n",
       " 'In the EGFR3-deficient mammary tumors, the PI3K-H1407R protein was determined to be associated with several tyrosyl phosphoproteins.',\n",
       " 'These studies also demonstrated that inhibition of other EGFR family members with lapatinib did not inhibit the mutant PI3K-mediated signaling.',\n",
       " 'However, co-inhibition of PI3K with the PI3Kalpha specific inhibitor BYL719 and EGFR-mediated signaling with lapatinib was more effective in suppressing growth than treatment with either the PI3K-alpha inhibitor or lapatinib alone.',\n",
       " 'Additional studies by this group determined that co-inhibition of PI3K and EGFR suppressed growth and PI3K signaling in human breast cancer cells containing the mutant PIK3CA-H1047R gene.',\n",
       " 'These studies point to the possibility of co-targeting of PI3K and EGFR in certain breast cancers.',\n",
       " 'to herceptin.',\n",
       " 'PTEN was identified as a modulator of resistance to Herceptin in cell cultures experiments.',\n",
       " 'Likewise oncogenic PIK3CA mutations would confer resistance to Herceptin in cell cultures [148].',\n",
       " 'In a screening of samples from 55 breast cancer patients, PIK3CA mutations or low expression of PTEN was associated with resistance to Herceptin.',\n",
       " 'Deregulation of PI3K and PTEN in Metastatic Breast Cancer.',\n",
       " 'The mutational status of the PIK3CA gene and expression of PTEN was examined in breast cancer specimens that differed in the state of malignancy (e.g., primary vs. metastatic breast cancer from the same cancer patient) [149].',\n",
       " 'The PIK3CA gene was determined to be mutated in 19 (40%) of primary tumors and 21 (42%) of metastatic cancers.',\n",
       " 'PTEN expression was examined by immunohistochemistry and PTEN expression was lost in 14 (30%) primary tumors and 13 (25%) metastases.',\n",
       " 'Thus the PI3KCA gene is frequently mutated and PTEN expression is often lost in breast cancer.',\n",
       " 'Epigenetic Regulation of Additional and Novel Genes Associated with Breast Cancer.',\n",
       " 'Epigenetic profiling has recently resulted in the identification of genes associated with breast cancer tumorigenicity that were methylated [150].',\n",
       " 'This study identified 264 hypermethylated loci in genomic CpG islands.',\n",
       " 'Hierarchical clustering in terms of the levels of methylation at the loci resulted in generation of at least three distinct groups of breast cancer patients.',\n",
       " 'Namely the methylation levels were distributed into three different groups, ER+ PR+ breast cancer patients, time to tumor relapse and lymph note metastasis.',\n",
       " 'Methylation of six genes (RECK, SFRP2, UAP1L1, ACADL, ITR, and UGT3A1) was associated with decreased relapse free survival.',\n",
       " 'Reversion-inducing-cysteine-rich protein with kazal motifs (RECK) is thought to be a metastasissuppressor gene and may interact negatively with matrix metalloproteinase (MMP9).',\n",
       " 'Secreted frizzled-related protein 2 (SFRP2) is a soluble modulator of Wnt signaling.',\n",
       " 'Methylation of this gene has been association with breast and other cancers [151].',\n",
       " 'UAP1L1 encodes a UDP-Nacteylglucosamine pyrophosphorylase 1-like 1 protein.',\n",
       " 'ACADL is a member of the acyl-CoA dehydrogenase family.',\n",
       " 'This is a family of mitochondrial flavoenzymes involved in fatty acid and branched chain amino-acid metabolism.',\n",
       " 'The ACADL gene product is associated with long-chain 3-hydroxyacyl-coenzyme A dehydrogenase deficiency.',\n",
       " 'The ITR gene (ITR GPR180) is a G proteincoupled receptor).',\n",
       " 'The UGT3A1 gene is a member of the UDP glycosyltransferase 3 family.',\n",
       " 'Mutations at PIK3CA and PTEN Can Confer Resistance to Herceptin in HER2+ Breast Cancer.',\n",
       " 'An RNA interference screen was performed to identify some of the genes involved in resistance www.impactjournals.com oncotarget 4613  Oncotarget  \\x0cDrug Resistance and CICs.',\n",
       " 'Drug resistance breast cancer cells are enriched in populations of cells which have characteristics of cancer stem cells.',\n",
       " 'These cancer cells with stem like characteristics are referred to cancer initiating cells (CICs) [152-175].',\n",
       " 'In the study by Britton and colleagues with both clinical fine needle aspirates obtained from breast cancer patients as well as established breast cancer cell lines, the expression of ATP-binding cassette sub-family G member 2 (ABCG2=BCRP1) was monitored [175].',\n",
       " 'In the studies with the side populations from MCF-7 and MDA-MB-231 cells, increased ABCG2 was observed in the side populations as well as elevated resistance to the chemotherapeutic drug mitoxantrone.',\n",
       " 'The increase in the side populations may be due to increased drug transporter expression in the cells with the CIC phenotype as the drug transporter would exclude the drugs from the cells.',\n",
       " 'The presence of the side populations in the fine needle aspirates was associated with ER-negative breast cancers and TNBCs which also had elevated ABCG2 protein expression.',\n",
       " 'Breast CICs can be characterized by increased expression of CD44 and decreased expression of CD24 (CD44 CD24) compared to the remaining cancer cells which are referred to as the bulk cancer cells (BCs).',\n",
       " 'The CICs and BCs differ in their gene expression patterns which may have resulted from epigenetic and other mechanisms [176].',\n",
       " 'Thus the CICs and BCs will likely have different signaling pathways activated suppressed which will require different therapeutic approaches to eliminate both populations of cancer cells.',\n",
       " 'breast cancer patients which lack amplification of HER2, documenting a role for HER2 in the growth of these normally HER2-negative cancers [181].',\n",
       " 'These authors demonstrated that HER2 was expressed in ER+, HER2luminal breast cancers and regulates the self renewal of the CIC sub-population.',\n",
       " 'HER2 expression was not due to gene amplification but was determined to result from receptor activation of NF-kappaB (RANK)-ligand in the bone microenvironment.',\n",
       " 'PI3K Pathway in Breast CICs.',\n",
       " 'The PIK3CA gene may be mutated in some breast CICs [182].',\n",
       " 'The PI3K PTEN Akt mTORC pathway has been reported to be important in breast CICs.',\n",
       " 'Sidepopulation positive MCF-7 breast cancer cells [MCF-7  (CIC)] were isolated from parent MCF-7 bulk cancer [MCF-7 (BC)] [183].',\n",
       " 'The MCF-7 (CIC) cells displayed increased drug transporter activity and enhanced colonyformation ability in vitro and greater tumorigenicity in vivo than the MCF-7 (BC).',\n",
       " 'The expression of critical pathways were compared between MCF-7 (CIC) and MCF-7 (BC).',\n",
       " 'The PI3K PTEN Akt mTORC1 and STAT3 pathways were shown to be differentially expressed in the MCF-7 (CIC) population and responsible in part for their enhanced survival [183].',\n",
       " 'Deregulation of Akt in Breast Cancer.',\n",
       " 'Mutations of Akt are relatively rare in breast cancer, however, aberrant activation of either upstream PIK3CA or polymorphism of the PH domain and leucine rich repeat protein phosphatase 2 (PHLPP2) gene can result in Akt expression.',\n",
       " 'In a study which examined the mutational status and polymorphism of genes in the PI3K PTEN Akt mTORC1 pathway, DNA was isolated from fine needle aspirations of 267 stage I-III breast cancers [184].',\n",
       " 'In this study, 28 genes were examined for 163 known cancer-related DNA sequence variations by Sequenom technology.',\n",
       " 'The PI3K pathway was determined to be frequently altered in breast cancers as at least one mutation in 38 alleles corresponding to 15 genes in 108 (40%) of the breast cancer samples.',\n",
       " 'The PIK3CA gene was determined to be the most frequently mutated (16.1% of all samples), the F-box and WD repeat domain containing 7, E3 ubiquitin protein ligase (FBXW7) gene, the second (8%), the BRAF gene, the third (3.0%), the EGFR1 gene, the fourth (2.6%), the AKT1 and CTNNB1 genes (beta-catenin) the fifth and sixth (1.9% each), the KIT and KRAS genes (1.5% each), and the PDGFRA gene, the seventh (1.1%).',\n",
       " 'Polymorphism at the PHLPP2 phosphatase which activates Akt was observed in 13.5% of the patient samples.',\n",
       " 'PIK3CA mutations were observed more frequently in ER+ cancers compared to TNBC (19 vs. 8%).',\n",
       " 'Interesting a high frequency of PIK3CA 4614 Oncotarget  Involvement of HER2 in Breast CICs.',\n",
       " 'HER2 is expressed in the CIC population [177, 178].',\n",
       " 'The expression of HER2 is modulated by the tumor microenvironment.',\n",
       " 'Targeting of HER2 in these CICs may be an appropriate therapeutic approach.',\n",
       " 'Interestingly herceptin suppressed tumor growth of the CIC in mouse xenograft models but in not established breast tumors [176].',\n",
       " 'Initially was thought that herceptin would only be effective in breast cancer patients which overexpressed HER2, some clinical studies have shown that herceptin may target breast cells which did not overexpress HER2 [179].',\n",
       " 'These results have led to the hypothesis that HER2 is an important molecule expressed on breast CICs and further studies have suggested that HER2 expression may be induced by signals in the microenvironment in breast CICs which lack HER2 gene amplification.The effectiveness of herceptin is thought to be due to its ability to target the breast CIC population as well as the PI3K  PTEN Akt mTORC and other signaling pathways [176, 177, 180].',\n",
       " 'Herceptin may also be effective in the treatment of www.impactjournals.com oncotarget  \\x0cmutations (28%) was observed in HER2+ breast tumors.',\n",
       " 'In TNBC, FBXW7 mutations were significantly more frequent compared to ER+ tumors (13 vs. 5%).',\n",
       " 'FBXWZ is a component of ubiquitin ligase (SKP-cullin-F-box) and may bind cyclin E and target it for ubiquitin-mediated degradation (Figure 2).',\n",
       " 'Some investigators have suggested that other molecules besides Akt are important in breast cancer and have proposed Akt-independent signaling mechanisms [185].',\n",
       " 'This group demonstrated that in some breast tumors cultured in an anchorage-independent fashion displayed minimal Akt activation and decreased reliance on Akt for growth.',\n",
       " 'In contrast, these cells had strong PDK1 activation and membrane localization and were dependent on the activated PDK1 substrate serum glucocorticoid regulated kinase family, member 3 (SGK3), which is related in structure to Akt.',\n",
       " 'SGK3 has been shown in additional studies to be linked with breast cancer and is induced by estrogen in breast cancer cells and is associated with ER expression [186].',\n",
       " 'Like Akt, SGK3 can phosphorylate GSK-3beta and TSC-2 which results in their inactivation and activation of mTORC1 and stimulation of protein translation [187, 188].',\n",
       " 'Knowledge of the particular kinase or other type of protein, responsible for the malignant potential of breast cancer cells could aid therapy by the use of more effective inhibitors which target the particular enzyme affected and responsible for the abnormal growth.',\n",
       " 'Activated Akt as a Marker for Sensitivity to Drug Therapy.',\n",
       " 'Phosphorylation (activation) of Akt has been shown to predict the effectiveness of paclitaxel chemotherapy in node-positive breast cancer patients [189].',\n",
       " 'In the National Surgical Adjuvant Breast and Bowel Project (NSABP) B-28 trial, the effectiveness of adding paclitaxel to doxorubicin (a.k.a Adriamycin) plus cyclophosphamide (AC) was examined in breast cancer patients (median follow up 9.1 years).',\n",
       " 'Enhanced effectiveness of adding paclitaxel to AC was observed in those breast cancer patients who expressed elevated P-Akt.',\n",
       " 'In contrast, no enhanced effectiveness of adding paclitaxel to AC was detected in the breast cancer patients who did not express activated P-Akt.',\n",
       " 'Thus addition of paclitaxel to breast cancer patients which are P-Akt- does not appear to increase the effect of AC therapy, while addition of paclitaxel to those breast cancer patients which are P-Akt+ does appear to improve therapy.',\n",
       " 'progenitor cells both in vitro and in vivo.',\n",
       " 'This increase in progenitor cells was mediated by increased Akt activation which resulted in the phosphorylation of GSK-3beta which in turn led to activation of the Wnt beta-catenin pathway.',\n",
       " 'The increases in progenitor cells could be suppressed by the Akt inhibitor perifosine [190].',\n",
       " 'PTEN is also important in the resistance of breast cancers to herceptin and other therapeutic approaches [191].',\n",
       " 'Interleukin-6 (IL-6) is cytokine which is an important immune-regulator.',\n",
       " 'IL-6 is also important in HER2-resistance as it can expand the CIC population [192-194].',\n",
       " 'Decreases in PTEN expression has been implicated in herceptin-resistance.',\n",
       " 'In HER2+ cell lines generated by knocking down PTEN, which were resistant to herceptin, the resistance was shown to be due to activation of an IL-6 inflammatory feed back loop.',\n",
       " 'This IL-6 inflammatory loop resulted in the expansion of breast CIC which have an EMT phenotype and secrete 100-fold more IL-6 than in the parental cells which did not have PTEN knocked down.',\n",
       " 'The authors of this important study also determined than treatment with an IL-6R Ab inhibited this IL-6 regulatory loop and reduced the CIC population and importantly reduced tumor growth and metastasis in mouse xenographs.',\n",
       " 'A figure depicting the effects of PTEN on certain CICs as well as ER-signaling is present in Figure 3.',\n",
       " 'The tumor microenvironment is important in CICs.',\n",
       " 'Interactions between the interleukin-8 receptor (IL-8R) and HER2 have been determined to be important in the survival of breast CICs [195].',\n",
       " 'NF-kappaB has been shown to be important in breast cancer CIC survival and HER2-dependent tumorigenesis.',\n",
       " 'NF-kappaB activity can be regulated by the PI3K PTEN Akt mTORC1 pathway [193].',\n",
       " 'Targeting PI3K Pathway to Inhibit Breast Cancer Resistance to Therapy.',\n",
       " 'Treatment of breast cancer patients with the aromatase inhibitor (AI) letrozole has been shown to result in suppression of the PI3K PTEN Akt mTORC1 pathway [194].',\n",
       " 'This clinical study examined the expression of PI3K (p110), P-Akt, and P-mTOR by immunohistochemistry on breast cancer samples from 113 patients.',\n",
       " 'The patients had been enrolled in a phase II study of letrozole or letrozole and cyclphosphamide.',\n",
       " 'Either letrozole or letrozole plus cyclophosphamide-treated patients displayed a reduction in PI3K and P-mTOR expression.',\n",
       " 'In contrast, expression of P-Akt did not change in the letrozoletreated patients whereas it decreased in the letrozole and cyclophosphamide-treated patients.',\n",
       " 'The reduction of P-Akt expression was associated with a better response rate and reduction in Ki67 staining.',\n",
       " 'The reduction in P-mTOR expression was associated with a longer diseasefree survival.',\n",
       " 'Thus the AI letrozole targets key components 4615 Oncotarget  PTEN and Cytokines and their Involvement in HER2-Resistance and CIC Survival.',\n",
       " 'PTEN is important in breast CIC survival.',\n",
       " 'Knockdown of PTEN expression was shown to result in increases in normal and malignant human mammary stem  www.impactjournals.com oncotarget  \\x0cof the PI3K PTEN Akt mTORC1 pathway which may be important in the successful treatment of certain breast cancers.',\n",
       " 'This same group developed some letrozoleresistant MCF-7 cells by culturing the cells for prolonged periods of time in letrozole.',\n",
       " 'The letrozole-resistant cells displayed elevated expression of key components of the PI3K PTEN Akt mTORC1 pathway.',\n",
       " 'They observed that suppression of the PI3K PTEN Akt mTORC1 pathways with PI3K or mTORC1 inhibitors reversed the acquired letrozole- resistance [195].',\n",
       " 'A diagram of the effects of letrozole on ER and PI3K PTEN Akt mTORC signaling and letrozole-resistance is presented in Figure 4.',\n",
       " 'Effects of PI3K PTEN Akt mTORC1 Pathway Activation and ER Expression on Breast Cancer.',\n",
       " 'PIK3CA mutations in ER+ HER2- luminal breast cancer actually result in low levels of mTORC1 expression and these breast cancers have some of the better treatment successes after 4HT therapy [196].',\n",
       " 'These PIK3CA mutations are predicted to render these breast cancers sensitive to targeted therapy.',\n",
       " 'The authors of this study have devised a PIK3CA gene expression pattern.',\n",
       " 'This pattern was associated with prognosis in those breast cancers with PIK3CA mutations that remained ER+  HER2-, it was not associated with prognosis in breast cancers which were either HER2+ or ER-.',\n",
       " 'An inverse relationship between PI3K pathway activation and ER expression was observed in ER+ breast cancers.',\n",
       " 'When the PI3K pathway was activated, lower levels of ER were detected, indicating a negative correlation.',\n",
       " 'Treatment of breast cancer cells with insulin like growth factor-1 (IGF-1), which activated the PI3K pathway resulted in decreased ER expression.',\n",
       " 'Likewise treatment of breast cancers with the PI3K inhibitor BEZ-235 resulted in increased ER activity as well as ER-regulated gene expression [197].',\n",
       " 'The PI3K pathway activity was increased in ER+ tumors and cell lines of the more aggressive luminal B subtype versus those of the less aggressive luminal A subtype.',\n",
       " 'The PI3K inhibitor would increase the effects of 4HT on the more aggressive luminal B breast cancer, potentially by increasing ER expression and restoring sensitivity to hormone based therapies.',\n",
       " 'This study suggests a potential therapeutic approach by combining PI3K inhibitors and 4HT.',\n",
       " 'The PI3K PTEN Akt mTORC1 pathway is also important in the growth of breast cancers which have become hormone-independent.',\n",
       " 'In four hormoneindependent breast cancer cell lines created after long term estrogen deprivation, increased phosphorylation of p70S6K, p85S6K and Akt was observed [198].',\n",
       " 'Inhibition of the PI3K pathway in these hormone-independent cells resulted in apoptosis.',\n",
       " 'These results indicate that some hormone-independent breast cancers may be sensitive to www.impactjournals.com oncotarget 4616  the combination of ER and PI3K pathway inhibitors.',\n",
       " 'Inhibition of PTEN activity has been associated with resistance to chemotherapeutic drugs, hypersensitivity to the mTORC1 inhibitor rapamycin as well as hormonal based therapies [199,200] Inhibition of PTEN expression by shRNA resulted in three ERalpha+ breast cancer cell lines that were able to grow in the absence of hormone.',\n",
       " 'Furthermore the cells were resistant to 4HT and fulvestrant.',\n",
       " 'Fulvestrant is an ER antagonist that functions by down regulating the ER.',\n",
       " 'Knock down of PTEN also increased ERalpha transcriptional activity in MCF-7 cells, but decreased ER protein levels and transcriptional activity in T47D and MDA-361 cells.',\n",
       " 'Inhibition of PTEN by shRNA increased basal and ligand-induced activation of IGF-1R and EGFR-3, documenting the effects that PTEN can have on the regulation of these upstream tyrosine kinases.',\n",
       " 'Inhibition of IGF-1R or EGFR-3 restored hormonal dependency and the effects of hormonal therapy on the breast cancer cells with PTEN-knocked down.',\n",
       " 'These studies indicate a possible treatment strategy for breast cancers which are either PTEN-negative or have decreased PTEN expression.',\n",
       " 'These results document the complex interactions between hormonal- and growth factor-dependent signaling.',\n",
       " 'Association between PIK3CA Mutations and AR Expression in Breast Cancer.',\n",
       " 'An association between AR expression and PIK3CA mutations was observed in a study which examined AR and ER expression and PIK3CA mutational status in 347 breast cancer patients [201].',\n",
       " 'AR expression was higher in those breast cancers which also expressed ER and PR.',\n",
       " 'In those samples that expressed AR, mutations in the PIK3CA kinase domain were more frequently detected than mutations in the helical domain or those lacking PIK3CA mutations.',\n",
       " 'High AR expression was linked with an improved recurrence-free survival in 207 patients with early-stage ER+ PR+ positive tumors after hormonalbased therapy.',\n",
       " 'Higher AR expression was also associated with PIK3CA mutations and not with PIK3CA WT or TNBCs.',\n",
       " 'These studies indicate that AR and PIK3CA (as well as ER and PR) may be prognostic markers for breast cancer [201].',\n",
       " 'Dysregulation of the PI3K PTEN Akt mTORC1 Pathway in Endometrial Cancer.',\n",
       " 'Endometrial cancer is one of the most frequent gynecological malignancies [202].',\n",
       " 'Many signaling pathways have been implicated in endometriod cancer [203-206].',\n",
       " 'Greater than 90% of endometriod cancers suffer from some type of mutation in the PI3K PTEN Akt  mTORC1 pathway.',\n",
       " 'Thus, this pathway is a key therapeutic target in endometriod cancer [207,208].',\n",
       " 'Endometrioid Oncotarget  \\x0cendometrial cancers (EEC) frequently have multiple mutations at PTEN, PIK3CA, PIK3R1 and KRAS.',\n",
       " 'The effects of the pan-class I PI3K inhibitor GDC-0941 and the mTORC1 inhibitor temsirolimus were examined on EEC cells with concurrent PIK3CA and PTEN mutations [209].',\n",
       " 'ECC with PIK3CA mutations were sensitive to GDC0941, while ECC with PTEN mutations were sensitive to the mTORC1 blocker.',\n",
       " 'Only 2 of 6 EEC cells with KRAS mutations were sensitive to MEK inhibitors.',\n",
       " 'The PI3K p110alpha selective inhibitor A66 was more effective than the PI3K p110beta inhibitors GSK2636771 and AZD6482 in suppressing the growth of the PTEN-mutant EEC cell lines.',\n",
       " 'Deregulation of Downstream Components of the PI3K PTEN Akt mTORC1 Pathway Involved in the Regulation of mRNA Translation in Breast Cancer.',\n",
       " 'The PI3K PTEN Akt mTORC1 pathway serves to regulate the translation of certain mRNAs which are considered difficult to translate due to their structures.',\n",
       " 'The PI3K PTEN Akt mTORC1 pathway can regulate the activity of key components of the translational apparatus such as eIF4E, eIF4G, 4E-BP1, rpS6, programmed cell death protein 4 (pdcd4), eEF2 and eEF2K.',\n",
       " 'The expression of eIF4E, eIF4G, 4E-BP1, p4E-BP1 (T37 46), p4EBP1 (S65), p4E-BP1 (T70), S6, pS6 (S235 236), pS6 (S240 244), pdcd4, eEF2 and eEF2K was examined in 190 hormone receptor-positive breast cancer patients [212].',\n",
       " 'This study followed the course of the breast cancer patients for 96 months.',\n",
       " 'Elevated eEF2K, rpS6, and p4E-BP and decreased pdcd4 were associated with poor prognosis in hormone receptor+ breast cancer.',\n",
       " 'These molecules may be prognostic markers and therapeutic targets for certain classes of breast cancer (e.g., hormoneresponsive breast cancers).',\n",
       " 'Expression of the PI3K-p110 alpha and beta Subunits in Breast Carcinomas and EEC.',\n",
       " 'A study was performed on 315 invasive breast carcinomas to compare the expression of the PI3K-p110 alpha and beta subunits in these breast cancer patients.',\n",
       " 'Then the expression results were correlated with clinical outcomes [210].',\n",
       " 'This immunohistochemistry study determined that overall the p110 subunits were expressed in 23.8% of invasive breast carcinomas.',\n",
       " 'PI3K-p110-alpha was expressed in 11.8% and 15.2% expressed PI3K-p110beta.',\n",
       " 'This study observed that PI3K-p110-alpha expression was associated with hormone receptor expression but was not associated with overall survival.',\n",
       " 'PI3K-p110beta expression was linked to HER2 overexpression and lack of hormone receptor expression.',\n",
       " 'PI3K-p110-beta+ breast cancer patients had lower age of onset, lymph node involvement and distant metastasis.',\n",
       " 'Those breast cancer patients that expressed membrane PI3K-p110-beta had a worse prognosis and overall survival.',\n",
       " 'These important clinical studies point to the possibility of co-targeting of HER2 and PI3K-p110-beta in certain breast cancer patients.',\n",
       " 'Recently it was determined that the PIK3R1 (p85alpha) and PIK3R2 (p85beta) regulatory subunits are mutated in EEC [211].',\n",
       " 'PIK3R1 mutations were reported to occur at a higher rate in EEC than any other cancer type.',\n",
       " 'Also this study demonstrated that the PIK3R2 gene is mutated in EEC, which was previously not thought to be a cancer gene.',\n",
       " 'Many PIK3R1 and PIK3R2 mutations are gain of function mutations.',\n",
       " 'Some PIK3R1 mutations bind and stabilize PTEN.',\n",
       " 'KRAS mutations are also common in EECs.',\n",
       " 'PI3K pathway mutations can occur in the presence of WT PTEN and they phenocopy PTEN loss as the pathway is activated.',\n",
       " 'Involvement of GSK-3 in Breast Cancer.',\n",
       " 'Phosphorylation by Akt also inhibits the activity of many key molecules involved in signaling and apoptosis.',\n",
       " 'Activated Akt can stimulate carcinogenesis by inactivating proteins that normally function to limit cell growth and regulate apoptosis [213].',\n",
       " 'GSK-3beta is a down stream target of Akt.',\n",
       " 'Introduction of kinasedead GSK-3beta [GSK-3beta(KD)] into epithelial cells promoted tumorigenesis of breast and skin tumors [214].',\n",
       " 'Overexpression of constitutively-active GSK-3beta altered chemosensitivity, cell cycle arrest and tumorigenicity of breast cancers [215-218].',\n",
       " 'Inhibition of GSK-3beta, by small molecule inhibitors induced epithelial mesenchymal transition (EMT) and invasion in breast cancer [219].',\n",
       " 'The localization of GSK-3beta was altered in a study of invasive ductal carcinomas (IDC).',\n",
       " 'A reduction or loss of cytoplasmic GSK-3beta in was observed in 53% of IDC examined [220].',\n",
       " 'Nuclear accumulation of GSK3beta was detected in 35% of the IDC samples examined.',\n",
       " 'This nuclear accumulation of GSK-3beta was associated positively with tumor grade [220].',\n",
       " 'A downstream target of GSK-3 is p27Kip-1 which is also implicated in breast cancer [221].',\n",
       " 'In addition, p70S6K can be regulated by GSK-3 and is involved in breast cancer [222].',\n",
       " 'Figure 2 depicts some of the interactions with GSK-3 and these and other signaling molecules.',\n",
       " 'GSK-3beta and beta-catenin can regulate cadherin-11 post-transcriptionally in breast and prostate cancer cells.',\n",
       " 'Inactivation of GSK-3beta lead to repression of cadherin-11 mRNA and protein levels [223].',\n",
       " 'Loss of 4617 Oncotarget  www.impactjournals.com oncotarget  \\x0ccytoplasmic GSK-3beta may promote EMT in breast cancer [223].',\n",
       " 'GSK-3 can regulate c-Myb which is important in EMT in breast cancer [224-227].',\n",
       " 'A studyl performed on breast cancer biopsies observed that inactivation of GSK-3beta was associated with elevated levels of the prolactin receptor, which is implicated in tumorigenesis [228].',\n",
       " 'The breast cancer resistance protein (BCRP) was determined to be downregulated in breast cancer cells when GSK-3beta was active, documenting that GSK-3beta can suppress active drug efflux [229].',\n",
       " 'Suppression of GSK-3beta activity by Akt phosphorylation enriched for mammary stem cells in both normal and breast cancer cells through activation of beta-catenin [230].',\n",
       " 'Thus at least with regards to mammary epithelial cells GSK-3beta activity appears to limit proliferation and suppress the stem-like cell population.',\n",
       " 'Often GSK-3beta is thought to have roles in tumor suppression.',\n",
       " 'However, this is not always the case.',\n",
       " 'In certain cancer types (e.g., pancreatic cancer) GSK-3beta was shown to participate in pro-inflammatory and antiapoptotic processes by positively regulating NF-kappaB activity in the nucleus [231-233].',\n",
       " 'The roles of GSK-3beta in cancer progression remain controversial and extremely complex and may depend on the cancer type.',\n",
       " 'The cellular localization of GSK-3beta is an important factor in controlling GSK-3beta ability to provide growth-limiting and survival-promoting activities.',\n",
       " 'Aberrant nuclear accumulation of GSK-3beta may be important in many cancers.',\n",
       " 'The biochemical events leading to loss of functional GSK-3beta activity may also enrich for a subpopulation of CICs that demonstrate enhanced motility, clonogenicity, and drug resistance.',\n",
       " 'However, the production of CICs may depend on the cancer type as well as stage of differentiation.',\n",
       " 'Micro RNAs (miRs) also play important roles in CICs.',\n",
       " 'TNBCs often have a high percentage of CICs [234].',\n",
       " 'Different miRs are expressed in TNBC than other breast cancer.',\n",
       " 'Breast CICs often express high levels of CD44 and lower levels of CD24 (CD44CD24) than the non-CIC population which is referred frequently to as the bulk cancer.',\n",
       " 'Autophagy is important in the regulation of the CIC population [235].',\n",
       " 'GSK-3 plays important roles in the regulation of autophagy and other age-related diseases [236].',\n",
       " 'Pharmacological inhibition of GSK-3 by small molecule inhibitors or deletion of GSK-3alpha leads to the prevention of autophagy.',\n",
       " 'We determined recently that GSK-3beta play important a role in MCF-7 breast cancer clonogenicity, drug resistance, and cell signaling [218].',\n",
       " 'Introduction of GSK-3beta(KD) into MCF-7 cells resulted in increases in both anchorage-dependent and anchorage-independent clonogenicity compared to cells transfected with the GSK3beta(WT) construct.',\n",
       " 'This increase in clonogenicity was observed when doxorubicin was absent or present.',\n",
       " 'More colonies were observed in MCF-7 GSK-3beta(KD) cells than MCF-7 GSK-3beta(WT) or MCF-7 GSK-3beta(A9) www.impactjournals.com oncotarget 4618  cells when the cells were treated with doxorubicin.',\n",
       " 'MCF7 GSK-3beta(A9) cells with constitutive GSK-3beta kinase activity displayed higher anchorage-dependent, but not anchorage-independent, clonogenicity than MCF7 GSK-3beta(WT) cells.',\n",
       " 'Anchorage-independent growth is one measurement of transformed cells.',\n",
       " 'In essence, it measures the ability of the cells to grow without adherence to a tissue culture plate surface.',\n",
       " 'Cells which are more transformed will form more colonies in soft agar than cells which are either not-tranformed or less-transformed.',\n",
       " 'GSK-3beta may play multiple roles by both limiting proliferation under certain conditions and allowing cell growth in others.',\n",
       " 'In some cases GSK-3beta functions as a tumor promoter by phosphorylation of Axin which leads to beta-catenin stabilization and potentially cancer.',\n",
       " 'In contrast, other cases GSK-3beta acts as a tumor suppressor and induces beta-catenin phosphorylation and proteasomal degradation.',\n",
       " 'When the cells were plated n doxorubicin, the anchorage-independent colony forming ability of MCF7 GSK-3beta(KD) cells in doxorubicin was enhanced five-fold in comparison to MCF-7 GSK-3beta(WT) cells.',\n",
       " 'Our results indicated that GSK-3beta activity influences breast cancer proliferation, motility and response to chemotherapy.',\n",
       " 'Loss of GSK-3beta kinase activity may confer survival advantages by upregulating factors involved in cell cycle progression, prevention of apoptosis and anchorage-independence.',\n",
       " 'Resistance to doxorubicin and 4HT was increased in MCF-7 GSK-3beta(KD) cells in comparison to MCF7 GSK-3beta(WT) cells.',\n",
       " 'However, drug resistant MCF-7  GSK-3beta(KD) cells responded to mTOR inhibition by treatment with rapamycin.',\n",
       " 'Additionally, a combination treatment of consisting of a MEK inhibitor and doxorubicin or 4HT was determined to have a synergistic effect that eliminated drug resistance in MCF-7 GSK3beta(KD) cells.',\n",
       " 'Targeting signaling molecules involved with the PI3K PTEN Akt mTORC1 and Raf MEK ERK pathways may be appropriate to overcome resistance to chemo- and hormonal therapy.',\n",
       " 'Genetic alterations which result in abnormal GSK-3beta activity should be taken into considered when designing a course of breast cancer therapy.',\n",
       " 'Combination drug inhibitor treatments could be advantageous by lowering the concentration of the chemotherapeutic drug and reduce therapy-related side effects.',\n",
       " 'There are alternate routes of GSK-3beta activation that are independent of Akt.',\n",
       " 'To add to the complexity, GSK-3beta plays roles in the Wnt beta-catenin pathway by being a component of the beta-catenin destruction complex as well as in the phosphorylation of Axin which results in the stabilization of Axin.',\n",
       " 'This interaction of GSK-3 with the Axin protein complex prevents Akt from accessing and phosphorylating GSK-3beta.',\n",
       " 'There are likely different cellular pools of GSK-3beta that are under separate regulation by Wnt and Akt, in which the activation of one Oncotarget  \\x0cdoes not affect the other.',\n",
       " 'Simultaneous Akt and GSK-3beta activity has been observed in pancreatic and colon cancer cell lines, indicating that increased Akt expression does not always result with decreased GSK-3beta activity [231].',\n",
       " 'Since certain pools of GSK-3beta may remain active in cancer, the precise cellular localization of these pools may also be an important factor involved in their regulation.',\n",
       " 'GSK-3beta is thought to be active predominantly in the cytoplasm, but it can also translocate to the nucleus.',\n",
       " 'The nuclear activities of GSK-3beta may appear to conflict with its cytosolic roles.',\n",
       " 'Aberrant nuclear accumulation of GSK-3beta has been observed in certain cancers [231].',\n",
       " 'A nuclear localization signal (NLS) is present in the basic domain of GSK-3beta [237].',\n",
       " 'Increased translocation of GSK-3beta to the nucleus may be involved with cancer progression.',\n",
       " 'Involvement of the Wnt Signaling Pathway in ERalpha+ Breast Cancers with PIK3CA Mutations.',\n",
       " 'The PIK3CA gene is mutated in 30-40% of ERalpha+ breast cancers.',\n",
       " 'Gene expression profiling of 249 ER-alpha+ breast tumors revealed that nineteen genes were differently expressed in PIK3CA-mutated tumors in comparison to samples lacking PIK3CA mutations.',\n",
       " 'Interesting PIK3CA mutations were associated with over-expression of several genes critical for the Wnt signaling pathway [WNT5A, transcription factor 7-like 2 (T-cell specific, HMG-box (TCF7L2), msh homeobox 2 (MSX2), tumor necrosis factor receptor superfamily, member 11b (TNFRSF11B)], as well as genes important in the regulation of gene transcription [SEC14-like 2 (SEC14L2), transcription factor AP-2 beta (activating enhancer binding protein 2 beta) (TFAP2B), nuclear receptor interacting protein 3 (NRIP3)] and metal ion binding [cytochrome P450, family 4, subfamily Z, polypeptide 1 (CYP4Z1), cytochrome P450, family 4, subfamily Z, polypeptide 2, pseudogene (CYP4Z2P), solute carrier family 40 (iron-regulated transporter), member 1 (SLC40A1), lactotransferrin (LTF), LIM and calponin homology domains 1 (LIMCH1) [238].',\n",
       " 'These studies suggest that targeting the Wnt signaling pathway may be an appropriate approach to treat certain ERalpha+ breast cancers containing PIK3CA mutations.',\n",
       " 'A figure illustrating some of the sites of targeting these pathways is present in Figure 5.',\n",
       " 'A recent component which has been shown to be involved in breast cancer drug resistance is the phosphatase DUSP4 which removes the phosphate from active ERK1,2.',\n",
       " 'Decreased expression of DUSP4 is associated with breast cancer drug resistance and increased expression of active ERK1,2 [241].',\n",
       " 'Other studies have indicated that increased MAPK signaling is a predicator of resistance to successful endocrine therapy of breast cancer patients [242].',\n",
       " 'This study included 114 women who were ER-alpha+ that were randomly assigned to neoadjuvant letrozole or letrozole plus metronomic cyclophosphamide (frequent low doses of cyclophosphamide).',\n",
       " 'The expression of twenty-four proteins involved in apoptosis, cell survival, hypoxia, angiogenesis and signaling were examined by immunohistochemistry in pretreatment samples.',\n",
       " 'The ages, sizes of the tumors, the nodal status of the tumor, tumor grade, histological type and treatment were also followed.',\n",
       " '81% of the patients had a disease response, 43% achieved a complete clinical response and 19% did not respond.',\n",
       " 'This study determined that increased phosphorylated ER-alpha and decreased phosphorylated ERK1,2 were factors associated with complete response to therapy.',\n",
       " 'Phosphorylated and activated ER-alpha was an independent factor for sensitivity to chemoendocrine therapy.',\n",
       " 'In contrast, HIF-alpha and phosphorylated ERK1 2 were independent factors associated with resistance to therapy.',\n",
       " 'These and other results indicating that targeting ERK (e.g., with MEK inhibitors) may be effective in reducing the drug resistance of certain breast cancers.',\n",
       " 'Interactions Between Raf-1 and Aurora Kinase Signaling in ER+ Breast Cancer.',\n",
       " 'Recently, it has been shown that there is a Raf-1mediated involvement of Aurora-A kinase signaling which is important in regulating the balance between EMT and MET in ER+ breast cancer cells.',\n",
       " 'This interaction is also important in chemoresistance and regulating the expression of many genes involved in EMT and MET.',\n",
       " 'The interactions between Raf-1 and Aurora-1 kinase signaling altered the metastatic potential of the cells.',\n",
       " 'Constitutive activation of Raf-1 oncogenic signaling induced HER2 overexpression and resulted in the development of distant metastases in ER+ MCF-7 Raf-1 breast cancer xenografts.',\n",
       " 'These distant metastases in xenograft models were associated with activation of MET as characterized by reduced expression of EMT-inducer genes (TGFB2, TWIST1 and FOXC1) while overexpression of the BMB7, CXCR7 and EGR genes.',\n",
       " 'Constitutive activation of Raf  MEK MAPK oncogenic signaling during tumor growth promoted the development of metastatic lesions from 4619 Oncotarget  Interactions between PI3K PTEN Akt mTORC1 and Raf MEK ERK Pathways in Breast Cancer and Drug Resistance.',\n",
       " 'The Raf MEK ERK pathway is also important in breast cancer drug resistance [239,240].',\n",
       " 'The Raf MEK  ERK pathway often interacts with the PI3K PTEN Akt  mTORC1 pathway and the two pathways often co regulate many signaling molecules such as p70S6K, eIF-4B, eIFwww.impactjournals.com oncotarget  \\x0cprimary tumors by activating MET [174].',\n",
       " 'Further studies by this same group demonstrated that constitutive activation of Raf-1 oncogenic signaling induced stabilization and accumulation of Aurora-A mitotic kinase.',\n",
       " 'This promoted EMT and stemness in ER+ MCF-7 Raf-1 cells.',\n",
       " 'The EMT transition was associated by reduced expression of CD24 and ERalpha, while HER2 and CD44 were upregulated.',\n",
       " 'The upregulation of the stemness gene SOX2 was linked to acquisition of multiple stem cell-like properties.',\n",
       " 'Namely the cells displayed an enhanced ability to form mammospheres in vitro and self-renewal in vivo.',\n",
       " 'The aberrant Aurora-A kinase activity induced phosphorylation and nuclear translocation of SMAD5, indicating a novel interplay between Aurora-A and SMAD5 signaling pathways in the development of EMT, stemness and ultimately tumor progression.',\n",
       " 'Pharmacological and molecular inhibition of Aurora-A kinase activity restored the CD24 epithelial phenotype that was coupled with ERalpha expression, downregulation of HER2, inhibition of EMT and impaired self-renewal ability and suppression of distant metastases.',\n",
       " 'These findings demonstrated the importance of Aurora-A kinase in the activation of EMT pathway responsible for the development of distant metastases in ERalpha+ breast cancer cells.',\n",
       " 'This study has translational implications because it highlighted the mitotic kinase Aurora-A as a novel, promising therapeutic target to eliminate invasive breast cancer cells and improve the disease-free and overall survival of ERalpha+ breast cancer patients resistant to conventional endocrine therapy [164].',\n",
       " 'Further studies indicated that cross-talk between ERalpha+ and Raf MEK ERK signaling pathways is a key oncogenic axis which is responsible in part for the development of ER-independent growth of breast cancers which were initially ERalpha+ and hormone sensitive.',\n",
       " 'In a metastatic breast cancer xenograft model harboring constitutive activation of Raf-1, the link between aberrant Raf MEK ERK signaling and development of endocrine resistance through abrogation of the ER signaling axis was elucidated.',\n",
       " 'The Aurora-A mitotic kinase was demonstrated to be important in the development of endocrine resistance.',\n",
       " 'Activation of SMAD5 nuclear signaling was detected as well as down-regulation of ERalpha.',\n",
       " 'These results suggest that the development of novel molecular therapies targeting the Aurora-A  SMAD5 oncogenic axis may be beneficial for the selective eradication of endocrine resistant ERalpha- cancer cells from the bulk tumor with benefits for breast cancer patients [243].',\n",
       " '(CIN) that affects cancer development and progression.',\n",
       " 'The relationships between induction of genotoxic stress, activation of cyclin-A Cdk2 and Aurora-A and development of centrosome amplification were investigated in MCF-7 breast cancer cell lines harboring a dominant negative (DN) p53 mutation (vMCF-7DNp53).',\n",
       " 'Genotoxic stress was induced in the MCF-7 cells harboring the DN p53 by treatment with hydroxyurea (HU) which induced centrosome amplification.',\n",
       " 'Aurora-A kinase activity was linked with centrosome amplication.',\n",
       " 'The Aurora-A kinase-induced centrosome amplification was determined to be mediated by Cdk2 kinase.',\n",
       " 'Molecular inhibition of Cdk2 activity by SU9516 suppressed Aurora-A centrosomal localization and consequent centrosome amplification.',\n",
       " 'MCF-7 cells harboring constitutively-activated Raf-1 displayed high levels of endogenous cyclin-A.',\n",
       " 'Targeting of Aurora-A by the Aurora kinase inhibitor Alisertib reduced cyclin-A expression.',\n",
       " 'Thus there is a positive feed-back loop between cyclin-A  Cdk2 and Aurora-A pathways in the development of centrosome amplification in breast cancer cells which may be which may provide additional approaches to target drug resistant breast cancer patients [244].',\n",
       " 'These studies document the key interactions between the Raf MEK ERK and Aurora kinase pathways which may be important in regulating breast cancer progression and serve as targets for therapeutic intervention.',\n",
       " 'Breast Cancer Therapy.',\n",
       " 'The most commonly utilized treatment for breast cancer is surgical resection with adjuvant chemotherapy, hormone therapy or radiation (National Cancer Institute, 2013).',\n",
       " 'Radiation and chemotherapy are effective in killing or limiting the growth of actively dividing cancer cells through various mechanisms including the production of oxygen free radicals, DNA damage, and subsequent apoptosis [245-251].',\n",
       " 'Common chemotherapeutic drugs used to treat breast cancer include the anthracyclines, taxanes, 5-fluorouracil, cyclophosphamide, and methotrexate [252].',\n",
       " 'Doxorubicin, also known as adriamycin, is in the anthracycline class of antibiotic chemotherapeutic drugs.',\n",
       " 'These compounds work by intercalating between adjacent DNA base pairs and inhibiting topoisomerase II, thereby interfering with DNA, RNA, and protein synthesis.',\n",
       " 'The effects of combining chemotherapeutic drugs have been analyzed by clinical trials.',\n",
       " 'A phase III, clinical trial with 1491 patients with node-positive, early breast cancer were randomly assigned to adjuvant treatment with docetaxel, doxorubicin, and cyclophosphamide (ACT) or fluorouracil, doxorubicin, and cyclophosphamide (FAC) every 3 weeks for six 4620 Oncotarget  Relationship between Raf-1 p53 Aurora Kinase and Centrosome Amplification in Breast Cancer.',\n",
       " 'Centrosome amplification has been shown to play key roles in the origin of chromosomal instability www.impactjournals.com oncotarget  \\x0ccycles.',\n",
       " 'The 10 year follow-up analysis of disease-free survival, overall survival, and long-term safety in the clinical trial was reported [253].',\n",
       " 'In general, breast cancer patients in ACT treatment group had better disease-free survival relative to breast cancer patients in the FAC treatment group.',\n",
       " 'In this study, hormone receptor, and HER2 status and nodal involvement status did not appear to change the disease-free survival between the enhanced survival in ACT treatment group vs. the FAC treatment group.',\n",
       " 'This study also documented that there were a significant percentage of patients who had a decrease in ventricular ejections secondary to anthracycline therapy.',\n",
       " 'An additional study evaluated the effects of ACT and FAC treatments on node negative breast cancer patients over a 5 year observation period.',\n",
       " 'This study was performed as it was known previously that ACT is superior to FAC when used as adjuvant therapy in women with node-positive breast cancer but the effects of taxanes on breast cancer patients with node cancer were not known [254].',\n",
       " 'The breast cancer patients in the two treatment groups were examined for 77 months.',\n",
       " 'The disease-free survival was higher in the ACT group (87.8%) than the FAC group (81.8%).',\n",
       " 'Thus the ACT treatment may be appropriate for some high risk, lymph node negative breast cancer patients.',\n",
       " 'Previously it was thought that hormonal therapy would only be effective in treatment of breast cancer cells which express ER-alpha.',\n",
       " 'Triple negative breast and HER2+ cancers often do not expresses ER-alpha, but some express ER-beta and may be sensitive to hormonal therapy [262].',\n",
       " 'The expression of hormonal receptors (ER, PR) and HER2 may change with the progression of breast cancer into metastatic cancer.',\n",
       " 'In addition, the combination of hormonal based therapy can be enhanced by the addition of herceptin and mTORC1 blockers in some cases.',\n",
       " 'Hormonal Therapy Combined with mTORC1 Blockage.',\n",
       " 'Another common treatment option for breast cancer consists of a hormonal-based approach [255].',\n",
       " 'Many early stage breast cancers overexpress various isoforms of the ER, making their growth dependent on estrogen.',\n",
       " '4HT can block estrogen signaling by competitively binding the ER and antagonizing its proliferative effects [256].',\n",
       " 'Once a breast cancer has undergone additional molecular changes allowing it to overcome estrogen-dependence, hormonal therapy is no longer effective, however, certain inhibitors can reverse the estrogen-independence (see Figures 3 and 4) [257261].',\n",
       " 'Exemestane is an AI and prevents conversion of testosterone into estrogen.',\n",
       " 'The quality of life was observed to be better in hormone-responsive metastatic breast cancer patients treated with exemestane and the mTORC1 blocker everolimus than patients treated with exemestane by itself.',\n",
       " 'In the BOLERO-3 study with herceptinresistant metastatic breast cancer patients, treatment with everolimus, herceptin and vinorelbine was more effective than treatment of with herceptin and vinorelbine [262].',\n",
       " 'Vinorelbine (Navelbine) is anti-mitotic drug [261].',\n",
       " 'It is used to treat non-small cell lung cancer and metastatic breast cancer.',\n",
       " 'Thus suppressing mTORC1 activity could enhance the effectiveness of exemestane and Herceptin in different types of breast cancer patients.',\n",
       " 'www.impactjournals.com oncotarget 4621  Figure 2: Dysregulated Expression of Upstream Receptors and Kinases Can Result in Activation of the Ras Raf MEK ERK and Ras PI3K PTEN Akt  mTOR and Other Signaling Pathways and Contribute to Malignant Transformation.',\n",
       " 'Sometimes dysregulated  expression of growth factor receptors occurs by increased expression, genetic translocations or genomic amplifications which can lead to activation of the Ras Raf MEK ERK, Ras PI3K PTEN Akt mTOR and other signaling pathways.',\n",
       " 'Alternatively chromosomal translocations can occur in nonreceptor kinases and other genes which result in activation of these pathways.',\n",
       " 'Genes in the Ras Raf MEK ERK and Ras PI3K  PTEN Akt mTOR pathways that have activating mutations detected in human cancer and proliferative diseases are indicated in red ovals and squares.',\n",
       " 'Tumor suppressor genes inactivated in certain cancer are indicated in black squares or octagons.',\n",
       " 'Other key genes are indicated in green ovals.',\n",
       " 'Red arrows indicate activating events in pathways.',\n",
       " 'Blocked black arrows indicating inactivating events in pathways.',\n",
       " 'Oncotarget  \\x0cAntibody Therapy of Breast Cancer.',\n",
       " 'The genetically engineered antibody herceptin (trastuzumab) is used to treat HER2+ breast cancers.',\n",
       " 'Combining herceptin with chemotherapy increases survival and response rates [263].',\n",
       " 'However a significant problem with chemotherapy is cardiotoxicity.',\n",
       " 'Herceptin therapy is also expensive and not always covered by insurance companies [264].',\n",
       " 'A diagram depicting the targeting of HER2 and other signaling molecules and sites where herceptin-resistance occurs is presented in Figure 6.',\n",
       " 'Pertuzumab (Perjeta(®) is a newer antibody developed by Genentech Roche.',\n",
       " 'Pertuzumab and herceptin binding to different sites on HER2 and the combined administration of pertuzumab and herceptin had synergistic inhibitory effects on HER2+ breast cancers [265].',\n",
       " 'Pertuzumab prevents the dimerization of HER2 with other EGFR family members (EGFR1, EGFR3, EGFR4) [266].',\n",
       " 'Pertuzumab prevents the interaction of HER2 with EGFR3 and subsequent activation of the PI3K PTEN   Akt mTOR pathway.',\n",
       " 'This has been proposed to be in part responsible to the anti-cancer effects of pertuzumab [267].',\n",
       " 'Initial studies with just pertuzumab did not reveal impressive results in suppressing cancer growth.',\n",
       " 'However, in the more recent phase III CLEOPATA trial have revealed that combination of pertuzumab with herceptin and docetaxel were promising and significantly improved prolonged progression-free survival for a first-line treatment of HER2+ metastatic breast cancer and importantly no increase in cardio-toxic effects were observed [268].',\n",
       " 'Pertuzumab received approval by the FDA to treat HER2+ metastatic breast cancer patients in 2012.',\n",
       " 'Pertuzumab in combination with herceptin and a taxol has been approved by the FDA and is being evaluated as a first line treatment option for HER2+ metastatic breast cancer patients who have not been previously treated with antiHER2 therapy or chemotherapy [269].',\n",
       " 'Pertuzumab is being further evaluated in the following clinical trials: MARIANNE, for advanced breast cancer, NEOSPHERE, for early breast cancer, TRYPAHENA, for HER2+ stage II III breast cancer and APHINITY for HER2+ nonmetastatic breast cancer [270].',\n",
       " 'Figure 3: Effects of Targeting PI3K PTEN Akt mTORC1 and IL-6 on Breast Cancer Hormonal Dependency, EMT and CICs.',\n",
       " 'Mutations at PIK3CA can alter hormonal dependency.',\n",
       " 'Silencing PI3K with PI3K inhibitors can restore hormonal dependency.',\n",
       " 'Silencing PTEN can also result in hormonal-independenc, 4HT and fulvestrant resistance.',\n",
       " 'Silencing PTEN can in some cases also lead in some cases to IL-6 production and an IL-6 inflammatory feed back loop which results in EMT and CIC formation.',\n",
       " 'Silencing of IL-6 with monoclonal antibodies (MoAb) can prevent this loop and result in the death of the breast CICs.',\n",
       " 'www.impactjournals.com oncotarget 4622  Oncotarget  \\x0cTrastuzumab emtansine is an antibody-drug modification of herceptin which is now linked with the cytotoxic agent mertansine (DM1).',\n",
       " 'It is abbreviated T-DM1 (trastuzumab-DM1) and also called Kadcyla, ado-trastuzumab emtansine and PRO132365.',\n",
       " 'Mertansine inhibits cell growth by binding to tubulin [271].',\n",
       " 'Herceptin targets the antibody-drug conjugate to HER+ cells [272].',\n",
       " 'T-DM1 is more effective than Herceptin in the treatment of HER+ advanced metastatic breast cancer patients.',\n",
       " 'It was evaluated in the EMILIA phase III clinical trial.',\n",
       " 'This trial consisted of 991 cancer patients with unrectable, locally advanced or metastatic HER+ breast cancer which had been previously treated with herceptin and taxanes with breast cancer patients treated with capecitabine (Xeloda) plus lapatinib (Tykerb).',\n",
       " 'Progression free survival was enhanced in breast cancer patients treated with T-DM1.',\n",
       " 'It has been approved by the FDA for the treatment of HER2+ advanced metastatic breast cancer patients.',\n",
       " 'T-DM1 has also been evaluated in other clinical trials such as MARIANNE, which compares the effectiveness of the taxanes (docetaxel or paclitaxel) combined with Herceptin vs T-DM1 by itself vs T-DM1 plus pertuzumab as a first line therapy for HER2+ breast cancers.',\n",
       " 'Patients  in this study are HER2+, which have either unresectable locally advanced or metastatic breast cancer.',\n",
       " 'T-DM1 is also being investigated in the TH3RESA clinical trial which is evaluating the effectiveness T-DM1 in HER2+ metastatic breast cancer patients that were previously treated with Herceptin and the EGFR2 kinase inhibitor lapatinib.',\n",
       " 'T-DM1 is also being evaluated in gastric cancers in clinical trials.',\n",
       " 'Tables 1-3 and Supplementary Tables 1-3 list the various clinical trials in breast cancer which are listed on the ClinicalTrial.gov website.',\n",
       " 'EGFR, PI3K, and mTORC1 inhibitors are being evaluated to treat breast cancer patients.',\n",
       " 'Drug targeting the ER have been developed and many have been evaluated in clinical trials [273].',\n",
       " 'The ER modulators: 4HT, raloxifene, and lasofoxifene) and AIs (anastrozole, letrozole, and exemestane) have been evaluated extensively [273].',\n",
       " 'An important pitfall with these drugs is they were thought to be only effective in ER+ cancers.',\n",
       " 'Drugs such as herceptin are more effective on HER2+ breast cancers.',\n",
       " 'Lapatinib is a small molecule HER2 EGFR1 dual kinase inhibitor that has also shown be effective in inhibiting the growth of HER2+ breast cancers.',\n",
       " 'TNBCs do not express either ER or HER2, thus novel therapies  Figure 4: Effects of PI3K PTEN Akt mTORC1 on Aromatase Resistance and Sensitivity to Tamoxifen (4HT).',\n",
       " 'The aromatase inhibitor (AI) letrozole prevents the conversion of testosterone (TEST) into estrogen (ER) and hence there is no ERalpha mediated gene transcription.',\n",
       " 'Letrozole alters the expression of the PI3K PTEN Akt mTORC1 pathway.',\n",
       " 'PI3K inhibitors will restore ERalpha mediated gene expression in Letrozole resistant cells and the cells revert to hormonal sensitivity.',\n",
       " 'Likewise PI3K inhibitors will restore the sensitivity of 4HT-resistant cells to 4HT.',\n",
       " 'Breast cells with mutations at PIK3CA may develop resistance to 4HT and PI3K inhibitors may restore sensitivity to 4HT.',\n",
       " 'www.impactjournals.com oncotarget 4623  Oncotarget  \\x0cneed to be developed for this class of breast cancer.',\n",
       " 'Drugs being evaluated to treat TNBC include poly ADP-ribose polymerase (PARP) inhibitors, vitamin D, and rexinoids, which activate the vitamin D and retinoid X receptors.',\n",
       " 'Mechanisms of Lapatinib-Resistance in HER2+ Cells: Activation of PI3K and Src.',\n",
       " 'The PI3K pathway and Src activation may be a mechanism by which some HER2+ cells grow in  response to HER2 inhibition by treatment with small molecule HER2 inhibitors such as lapatinib.',\n",
       " 'Lapatinibresistant HER2+ breast cancer cell lines were generated by culturing the cells in the presence of lapatinib for prolonged periods of time.',\n",
       " 'These cells did not express activated HER2, but they did express the PI3K PTEN Akt  mTORC and Raf MEK ERK pathways which is believed to be due to activated Src family members.',\n",
       " 'Treatment of the lapatinib-resistant HER2+ cells with Src inhibitors suppressed the PI3K PTEN Akt mTOR pathway as well as  Figure 5: Sites of Targeting the EGFR PI3K PTEN Akt mTORC Pathway with Small Molecule Membrane-Permeable Inhibitors and Monoclonal Antibodies (MoAbs).',\n",
       " 'The PI3K PTEN Akt mTORC1 pathway is regulated by Ras (indicated in green ovals), as well as various upstream growth factor receptors (indicated in purple) and PTEN indicated in a black rectangle.',\n",
       " 'Sites where various small molecule inhibitors suppress this pathway are indicated by red octagons.',\n",
       " 'The downstream transcription factors regulated by this pathway are indicated in yellow diamond shaped outlines.',\n",
       " 'The Ras Raf MEK ERK pathway also interacts with key proteins involved in protein translation (indicated in green ovals).',\n",
       " 'The two pathways aid in the assembly of the protein translation complex (indicated in purple ovals) responsible for the translation of weak mRNAs (indicated in a red line folding over on itself) important in the prevention of apoptosis.',\n",
       " 'Other key proteins inactivated by PI3K PTEN Akt mTORC1 pathway (e.g., TSC, PTEN, beta-catenin, Foxo and 4E-BP1 are indicated by black ovals, diamonds and rectangles.',\n",
       " 'Red arrows indicate activating events in pathways.',\n",
       " 'Blocked black arrows indicating inactivating events in pathways.',\n",
       " 'GF = growth factor, GFR = growth factor receptor.',\n",
       " 'www.impactjournals.com oncotarget 4624  Oncotarget  \\x0cFigure 6: Sites of Resistance in the HER2 PI3K PTEN Akt mTORC Pathway and Potential Sites for Intervention with Small Molecule Membrane-Permeable Inhibitors and Monoclonal Antibodies (MoAbs).',\n",
       " 'The HER2 EGFR receptor  is indicated in a blue figure.',\n",
       " 'The downstream PI3K PTEN Akt mTORC1 pathway is regulated by Ras (indicated in green ovals), PTEN indicated in a black octagon, activated Src is indicated by a red oval, IRS1 is indicated by an orange oval.',\n",
       " 'Sites where various small molecule inhibitors suppress this pathway are indicated by red octagons.',\n",
       " 'Sites which stimulate proteins involved in autophagy are indicated by green octagons and ovals.',\n",
       " 'The Serine threonine-protein kinase ULK1 (ULK1) which is regulated by mTORC1 is indicated in a black oval.',\n",
       " 'The downstream transcription factors regulated by this pathway are indicated in yellow diamond shaped outlines.',\n",
       " 'The Ras Raf  MEK ERK pathway also interacts with key proteins involved in protein translation (indicated in green ovals).',\n",
       " 'The two pathways aid in the assembly of the protein translation complex (indicated in purple ovals) responsible for the translation of weak mRNAs (indicated in a red line folding over on itself) important in the prevention of apoptosis.',\n",
       " 'Other key proteins inactivated by PI3K PTEN Akt mTORC1 pathway (e.g., TSC, PTEN, GSK-3beta, beta-catenin, Foxo and 4E-BP1 are indicated by black ovals, diamonds and rectangles.',\n",
       " 'Red arrows indicate activating events in pathways.',\n",
       " 'Blocked black arrows indicating inactivating events in pathways.',\n",
       " 'www.impactjournals.com oncotarget 4625 Oncotarget  \\x0cTable 1: Clinical Trials with Single Targeted Agent Treatments (Monotherapy) Official Trial Name Clinical Trial # Phase Type of Cancer Patient in Trial of Trial Status of Trial Intervention Publications  Phase II Trial of Akt Inhibitor MK2206 in Patients With Advanced NCT01277757 Breast Cancer Who Have Tumors II With a PIK3CA Mutation, or an AKT Mutation, and or PTEN Loss PTEN Mutation Pre-surgical Evaluation of MK-2206 NCT01319539 in Patients With Operable Invasive II Breast Cancer A Phase I, Open-Label, Two-Stage Study to Investigate the Safety, Tolerability, Pharmacokinetics and NCT00920257 I Pharmacodynamics of the Oral AKT Inhibitor GSK2141795 in Subjects With Solid Tumors or Lymphomas A Phase II Trial of BKM120 (a PI3K Inhibitor) in Patients With Triple NCT01629615 II Negative Metastatic Breast Cancer A Phase I II, Multi-center, Openlabel Study of BEZ235, Administered Orally on a Continuous Daily Dosing Schedule in Adult Patients With Advanced Solid Malignancies Including Patients With Advanced Breast Cancer A Phase II Trial of Short-Term Everolimus (RAD001) to Predict Response in Women With Operable Breast Cancer Phase II Trial of CCI-779 (Temsirolimus) in Patients With Locally Advanced or Metastatic Breast Cancer A Phase Ib II Study Investigating the Combination of Everolimus With Trastuzumab and Paclitaxel in Patients With HER2-overexpressing Metastatic Breast Cancer  Male Breast Cancer (BC), recurrent BC, stage IIIB BC, stage IIIC BC, Recruiting stage IV BC  MK2206 Akt inhibitor  Not Provided  ER- BC, ER+ BC, HER2- BC, Ongoing, but HER2+ BC, PR- BC, PR+ BC, stage MK2206 not recruiting IB BC, stage II BC, stage IIIA BC, participants stage IIIB BC, stage IIIC BC, TNBC  Not provided  Cancer  Completed  GSK2141795 Akt Inhibitor  Not provided  BC  Currently recruiting participants  BKM120 Not PI3K Inhibitor provided  NCT00620594 I  BC, Advanced Solid Tumors, Cowden Syndrome Completed  BEZ235 PI3K mTOR Inhibitor  Not provided  NCT00855114  II  BC  Withdrawn  Everolimus (mTORC1 blocker)  Not provided  NCT00376688 II  Temsirolimus Male BC, Recurrent BC, Stage IIIA, Ongoing but (mTORC1 Stage IIIB, Stage IIIC, Stage IV BC not recruiting blocker)  Not provided  NCT00426556  I  Metastatic BC  Completed  Everolimus  Not provided  A Phase II Trial of RAD001 in Triple II Negative Metastatic Breast Cancer NCT00827567 A Phase Ib Study Investigating the Combination of Everolimus With Trastuzumab and Vinorelbine in I NCT00426530 Patients With HER2-overexpressing Metastatic Breast Cancer Neoadjuvant Phase II Study Of Everolimus Plus Cisplatin In Triple Negative Breast Cancer Patients II NCT01931163 With Residual Disease After Standard Chemotherapy Randomized, Double Blind, Multicentric Phase III Trial Evaluating the Safety and Benefit of Adding Everolimus to Adjuvant Hormone Therapy in Women With III NCT01805271 Poor Prognosis, ER+ and HER2Primary Breast Cancer Who Remain Free of Disease After Receiving 3 Years of Adjuvant Hormone Therapy  Breast cancer  Terminated  RAD 001 (everolimus)  [337-339]  Breast Neoplasms, Metastasis  Neoplasm  Completed  Everolimus  [340]  BC, TNBC  Recruiting  Everolimus  Not provided  ER+, HER2- BC  Recruiting  Everolimus  Not provided  www.impactjournals.com oncotarget  4626  Oncotarget  \\x0cRADAR: A Randomized Discontinuation Phase II Study to Determine the Efficacy of RAD001 II NCT00466102 in Breast Cancer Patients With Bone Metastases Phase II Trial of RAD001 Plus Carboplatin in Patients With TripleII NCT01127763 Negative Metastatic Breast Cancer A Randomized Phase II Study of Two Different Schedules of RAD001C in II Patients With Recurrent Metastatic NCT00255788 Breast Cancer A Randomized Study of mTOR Inhibition by RAD001 (Everolimus) in Invasive Breast Cancer Patients II After Pre-operative Use of NCT01088893 Anthracycline and or Taxane-based Chemotherapy A Multicenter Randomized, Double Blind, Placebo- Controlled, Phase II Study to Compare Endocrine Treatment Alone Versus Endocrine Treatment With Everolimus in III NCT01773460 Patients With HR+ HER2- Metastatic Breast Cancer and Progression After Previous Treatment With Exemestane and Everolimus Influence of Exceptional Patient Characteristics on Everolimus IV NCT01948960 Exposure A Phase Ib Study Administering Rapamycin (Sirolimus) With I Grapefruit Juice in Patients With NCT00375245 Advanced Malignancies A Phase Ib, Open-label Study to Evaluate RAD001 as Monotherapy Treatment in Chinese Patients I II NCT01175096 With Advanced Pulmonary Neuroendocrine Tumor A Phase 1 2, Multi-Center, OpenLabel, Dose Finding Study to Assess the Safety, Tolerability, Pharmacokinetics and Preliminary Efficacy of the mTOR Kinase I II NCT01177397 Inhibitor CC-223 Administered Orally to Subjects With Advanced Solid Tumors, Non-Hodgkin Lymphoma or Multiple Myeloma.',\n",
       " 'A Randomized Study of mTOR Inhibition by RAD001 (Everolimus) in Invasive Breast Cancer Patients II After Pre-operative Use of NCT01088893 Anthracycline and or Taxane-based Chemotherapy  BC  Ongoing, but RAD001 not recruiting  Not provided  BC  Ongoing, but RAD001 not recruiting  Not provided Not provided  BC  Completed  Everolimus  BC  Unknown because information Everolimus has not been verified recently  Not provided  Metastatic BC  Recruiting  Everolimus  Not provided  BC  Recruiting  Everolimus Rapamycin (sirolimus) Other: Grapefruit Juice  Not provided  Tumors, Neoplasm Metastasis  Completed  Not provided  Recruitment unknown because RAD001 Neuroendocrine Tumors, Carcinoid information (everolimus, Tumor has not been Afinitor®) verified recently Multiple Myeloma, Diffuse Large B-Cell Lymphoma, Glioblastoma Multiforme, Hepatocellular Currently Carcinoma, Non-Small Cell Lung recruiting Cancer, Neuroendocrine Tumors of participants Non-Pancreatic Origin, Hormone Receptor-Positive BC  Not provided  CC-223  Not provided  BC  Recruitment unknown because the information Everolimus has not been verified recently  Not provided  www.impactjournals.com oncotarget  4627  Oncotarget  \\x0cgrowth and restored sensitivity to lapatinib.',\n",
       " 'Treatment of primary HER2+ tumors with lapatinib actually resulted in the expression of Src-family kinases as detected by mRNA analysis.',\n",
       " 'Treatment of HER2+ BT-474 cells with lapatinib and Src inhibitors was more effective in suppressing the growth of these cells in xenograft models than treatment with lapatinib alone.',\n",
       " 'These studies provide a rationale for the treatment of certain HER2+ cells with the combination of lapatinib and Src inhibitors as a means to prevent drug resistance of HER2+ cells [274].',\n",
       " 'Thus activation of Src family kinases may be a mechanism by which some HER2+ cells become resistant to HER2 inhibitors such as lapatinib.',\n",
       " 'Cotargeting of mTORC1 and Other Signaling Pathways in Breast Cancer Therapy.',\n",
       " 'The possibility of targeting mTORC1 and other signaling pathways such as Raf MEK ERK, PI3K, IGF1R to treat breast cancer is also being examined [275].',\n",
       " 'In some cases, treatment with rapalogs induces the IGF-1R or PDGFR pathways which in turn will activate the Raf  MEK ERK pathway.',\n",
       " 'This type of approach may also be appropriate to over come the rapamycin-resistance that certain breast cancers have developed.',\n",
       " 'There are clinical trials in progress evaluating the effectiveness of co targeting mTORC1 and pathways such as IGF-1R, PI3K  Table 2: Combined Targeted Agent Treatments Official Trial Name An Open-Label, Two Part, Phase I II Study to Investigate the Safety, Pharmacokinetics, Pharmacodynamics, and Clinical Activity of the MEK Inhibitor GSK1120212 in Combination With the AKT Inhibitor GSK2110183 in Subjects With Solid Tumors and Multiple Myeloma A Phase Ib, Open-label, Multi-center, Dose-escalation and Expansion Study of an Orally Administered Combination of BKM120 Plus MEK162 in Adult Patients With Selected Advanced Solid Tumors A Phase Ib, Open-label, Multi-center, Dose-escalation Study of Oral BKM120 in Combination With Oral GSK1120212 in Adult Patients With Selected Advanced Solid Tumors.',\n",
       " 'A Phase I II Trial of an Oral MTOR Protein Kinase Inhibitor (Everolimus, RAD001) in Combination With an Oral EGFR Tyrosine Kinase Inhibitor (Erlotinib, TarcevaTM) In Patients With Metastatic Breast Cancer An Open-label, Multi-center Phase I Dose-finding Study of RAD001 (Everolimus, Afinitor®) in Combination With BEZ235 in Patients With Advanced Solid Tumors Clinical Trial # Phase of Type of Cancer Patient in Trial Trial Status of Trial Intervention Publications  NCT01476137  I  Cancer  Completed  GSK1120212, GSK2110183  Not provided  NCT01363232  I  BKM120 (a Advanced Solid Tumors, Selected Solid Ongoing, pan-class I PI3K Not Tumors but not inhibitor)+ provided participating MEK162 (MEK inhibitor)  NCT01155453  I  Advanced Solid Tumors, Selected Solid Ongoing, but BKM120, Tumors not recruiting GSK1120212  Not provided  NCT00574366  I  BC  Completed  Erlotinib (EGFR1 inhibitor), Not Everolimus provided (RAD001)  NCT01482156  I  RAD001 + Advanced Solid Tumors, Metastatic BC, Ongoing, but BEZ235, a dual Not Metastatic Renal Cell Carcinoma not recruiting inhibitor of PI3K provided and mTOR Cixutumumab (an IGF-1R Ongoing, but Not inhibitor), not recruiting provided temsirolimus  Phase I II Trial of IMC-A12 in Combination With Temsirolimus in Patients With NCT00699491 Metastatic Breast Cancer  I II  Male BC, Recurrent BC, Stage IV BC  www.impactjournals.com oncotarget  4628  Oncotarget  \\x0cPhase I Clinical Trial of Temsirolimus and Vinorelbine in Advanced Solid Tumors.',\n",
       " 'NCT01155258  I  Extensive Stage Small Cell Lung Cancer, Hereditary Paraganglioma, Male BC, Malignant Paraganglioma, Metastatic Gastrointestinal Carcinoid Tumor, Metastatic Pheochromocytoma, Pancreatic Polypeptide Tumor, Recurrent Breast Cancer, Recurrent Cervical Cancer, Recurrent Endometrial Carcinoma, Recurrent Gastrointestinal Carcinoid Tumor, Recurrent Islet Cell Carcinoma Recurrent Neuroendocrine Carcinoma of the Skin, Recurrent Nonsmall Cell Lung Cancer, Recurrent Ovarian Epithelial Cancer Recurrent Ovarian Germ Cell Tumor, Recurrent Pheochromocytoma, -Recurrent Prostate Cancer, Recurrent Renal Cell Cancer, Recurrent Small Cell Lung Cancer, Recurrent Uterine Sarcoma, Regional Gastrointestinal Carcinoid Tumor, Regional Pheochromocytoma, Stage III Ongoing, but Cervical Cancer, Stage III Endometrial not recruiting Carcinoma, Stage III Neuroendocrine Carcinoma of the Skin, Stage III Ovarian Epithelial Cancer, Stage III Ovarian Germ Cell Tumor, Stage III Prostate Cancer, Stage III Renal Cell Cancer, Stage III Uterine Sarcoma, Stage IIIA BC, Stage IIIA Non-small Cell Lung Cancer, Stage IIIB BC, Stage IIIB Nonsmall Cell Lung Cancer, Stage IIIC BC, Stage IV BC, Stage IV Endometrial Carcinoma, Stage IV Neuroendocrine Carcinoma of the Skin, Stage IV Nonsmall Cell Lung Cancer, Stage IV Ovarian Epithelial Cancer, Stage IV Ovarian Germ Cell Tumor, Stage IV Prostate Cancer, Stage IV Renal Cell Cancer, Stage IV Uterine Sarcoma, Stage IVA Cervical Cancer, Stage IVB Cervical Cancer Thyroid Gland Medullary Carcinoma BC, Endometrial Cancer, Kidney Cancer Lung Cancer, Lymphoma, Unspecified Completed Adult Solid Tumor, Protocol Specific  Temsirolimus, vinorelbine Not ditartrate (an provided anti-mitotic vinca alkaloid)  A Phase I Study of Temsirolimus in Combination With Metformin in Advanced NCT00659568 Solid Tumours An Open-Label, Phase Ib Dose Escalation Trial of Oral Combination Therapy With MSC1936369B and NCT01390818 SAR245409 in Subjects With Locally Advanced or Metastatic Solid Tumors A Phase I Study of BKM120 and Everolimus in Advanced NCT01470209 Solid Malignancies A Phase Ib, Open-label, Multi-center, Dose-escalation and Expansion Study of an Orally Administered Combination of BEZ235 Plus NCT01337765 MEK162 in Adult Patients With Selected Advanced Solid Tumors  I  Metformin hydrochloride, temsirolimus  Not provided  I  MSC1936369B Locally Advanced Solid Tumor, (a MEK Metastatic Solid Tumor, BC, Non Small Ongoing, but inhibitor) and Not Cell Lung Cancer, Melanoma, Colorectal not recruiting SAR245409 (a provided Cancer dual PI3K mTOR inhibitor) Solid Tumors Recruiting participants BKM120, everolimus Not provided  I  I  Unspecified Adult Solid Tumor, Protocol Completed Specific Solid Tumor  BEZ235, MEK162  Not provided  www.impactjournals.com oncotarget  4629  Oncotarget  \\x0cA Clinical Trial to Qualify the Growth Factor Signature (GFS) as an Intermediate Biomarker of Response NCT01220570 for Development of PI3KPathway Inhibitors in Patients With Breast Cancer Phase I Parallel Protocol of MK-8669 (Ridaforolimus) + MK-2206 and MK-8669 (Ridaforolimus) + MKNCT01295632 0752 Doublets (MK-MK) in Patients With Advanced Cancer  I  BC  Completed  Ridaforolimus (mTORC1 blocker), Dalotuzumab (IGF-1R inhibitor)  Not provided  I  Advanced Cancer  Ridaforolimus, MK-0752 (gamma secretase Ongoing, but inhibitor, a Not not recruiting Notch signaling provided pathway inhibitor).',\n",
       " 'MK2206  and Raf MEK ERK (find out how many trials).',\n",
       " 'Combinations of Herceptin with pertuzumab, or T-DM1 and mTOR inhibitors added to an aromatase inhibitor are new therapeutic approaches for the treatment of HER2+ breast cancers [276].',\n",
       " 'Inhibition of mTORC1 in combination with endocrine therapy may be an approach for treatment of metastatic breast cancer patients which are resistant to aromatase in inhibitors.',\n",
       " 'Two clinical trials (TAMRAD and BOLERO-2) have revealed significant effects [277].',\n",
       " 'Novel Approaches to Treat TNBC Patients.',\n",
       " 'While therapies for ER+ and HER+ breast cancer patients have improved over the years, the treatment possibilities for TNBC are more limited.',\n",
       " 'A recent review has summarized the clinical results with more novel approaches to treat TNBCs which include inhibitors of enzymes such as PARP and HDAC, kinases such as Jak2 and Src and receptor kinases as well as biological processes such as angiogenesis [281].',\n",
       " 'Combining Herceptin with PI3K, mTORC1 Inhibitors or Chemotherapy to Improve Breast Cancer Therapy.',\n",
       " 'Recently it was demonstrated that combining a pan PI3K inhibitor (XL 147) with Herceptin may overcome herceptin-resistance in breast cancer by suppressing HER2 PI3K FOXO survivin signaling [278].',\n",
       " 'Clinical trials have been performed examining the ability to combine lapatinib, herceptin with paclitaxel in first line HER2+ positive breast cancer patients.',\n",
       " 'The dose limiting toxicity observed in the study was diarrhea [279].',\n",
       " 'Clinical studies examining the combination of the mTORC1 blocker everolimus and either 4HT or exemestane (Aromasin) have been reviewed [280].',\n",
       " 'Previous clinical studies combining endocrine therapy with rapalogs to treat metastatic hormone receptor+ breast cancer patients yielded variable results.',\n",
       " 'However, two recent independent trials which selected patients refractory to endocrine therapy, demonstrated that combining the rapalog everolimus (Afinitor) with 4HT or combining everolimus with exemestane (Aromasin) was more effective than either endocrine agent alone.',\n",
       " 'The rapalogs may sensitize the normally endocrine therapy resistant patients to endocrine therapy.',\n",
       " 'Likewise additional clinical trials with HER2+ breast cancer patients and the inclusion of PI3K mTOR inhibitors may improve therapy of the HER2 inhibitors.',\n",
       " 'Resistance to Therapy.',\n",
       " 'Rb is a critical protein involved in the regulation of cell cycle progression and is a tumor suppressor.',\n",
       " 'The Rb pathway is involved in sensitivity to tamoxifen.',\n",
       " 'Loss of functional Rb activity is associated with resistance to tamoxifen [282,284].',\n",
       " 'Thus oncogenes and tumor suppressor genes often are important in the sensitivity to therapy [168,285].',\n",
       " 'A diagram depicting some of the sites where mutations result in resistance to therapy is presented in Figure 7.',\n",
       " 'Breast cancers can be inherently drug resistance or develop an acquired resistance after exposure to the particular drug [213,252].',\n",
       " 'Resistance can develop in patients and breast cancer cells when treated with anthracycline chemotherapy drugs, such as doxorubicin, daunorubicin, and epirubicin as well as hormonal based therapies [199,285,286].',\n",
       " 'Therefore, cancer cells are capable of undergoing calculated changes that confer survival advantages in otherwise nutrient-restricted or toxic environments [287,288].',\n",
       " 'Thus novel targets and approaches to treat breast cancer are being evaluated [167,289-308].',\n",
       " 'TGF-beta and Smads are also important targets in breast cancer [309-312].',\n",
       " 'The anti-diabetes drug metformin is showing some successes in treating breast cancer including Herceptin-resistant breast cancers [313317] A target of metformin is AMPK which is involved in regulation of components of the PI3K Akt mTORC1 and GSK-3 pathways.',\n",
       " 'An additional target of metformin is Stat3 and it has been shown that metformin can inhibit the growth of TNBC [318].',\n",
       " 'Additional Stats may be important 4630 Oncotarget  www.impactjournals.com oncotarget  \\x0cFigure 7: Induction of the Ras Raf MEK ERK Pathway after Leukemia Therapy and Subsequent Effects on Cell Cycle Progression, Survival Pathways and Protein Translation.',\n",
       " 'After chemotherapy or radiotherapy there can be activation of signaling pathways which can actually promote cell survival and may lead to therapy resistance.',\n",
       " 'Chemotherapeutic drug treatment (shown in irregular black oval) frequently results in the induction of reactive oxygen species (ROS) (shown in black square).',\n",
       " 'ROS can induce the calcium calmodulin kinase (CaM-K) cascade which can induce Ras which can subsequently activate both the Raf MEK ERK and PI3K  PTEN Akt mTOR cascades (most components of the two cascades which promote signaling are show in green ovals, transcription factors activated by events are shown in yellow diamonds, transcription factors inactivated by events shown in black diamonds).',\n",
       " 'Induction of the Ras Raf MEK ERK and PI3K PTEN Akt mTOR pathways can result in the activation of many survival pathways, and regulate both cell cycle progression as well as protein translation.',\n",
       " 'Some of the phosphorylation events mediated by Akt actually serve to inhibit the activities of key proteins such as the Foxo transcription factors and the murine double minute (MDM2) ubiquitin ligase (depicted in black ovals).',\n",
       " 'MDM2 serves to regulate p53 protein stability by ubiquitination, however when it is phosphorylated by Akt it is inactivated.',\n",
       " 'Moreover chemotherapeutic drugs and radiotherapy can induce the ataxia telangiectasia mutated (ATM) protein shown in a black oval, which can in turn phosphorylate and regulate p53.',\n",
       " 'p53 can have complex positive and negative effects on cell growth (depicted in yellow diamond), it can regulate the expression of p21 cyclin dependent kinase inhibitory protein-1 (p21Cip1) which controls cell cycle progression.',\n",
       " 'p53 can also control the transcription of genes such as Puma, Noxa and Bax which are involved in apoptosis (all of these molecules are shown in black ovals, as they tend to inhibit cell cycle progression or promote apoptosis).',\n",
       " 'Both Akt and ERK can phosphorylate p21Cip1 which alters its activity and ability to inhibit cell growth (shown as black phosphorylation sites) and subsequently influence cell growth and therapy resistance.',\n",
       " 'p27Kip1 can also be phosphorylated by both Akt and ERK, however the effects of these phosphorylation events are unclear.',\n",
       " 'Akt phosphorylation of p27Kip1 may result in its cytoplasmic localization, while ERK phosphorylation of p27Kip1 may result in elevated levels of the protein.',\n",
       " 'Hence phosphorylation of proteins by ERK and Akt can have dramatic effects on cell proliferation and contribute to the therapy resistance.',\n",
       " 'Chemo- and radiotherapy also induce breaks in the DNA.',\n",
       " 'In the presence of functional BRCA1 and BRCA2 (indicated in yellow ovals) these breaks may be repaired and normal gene transcription can occur.',\n",
       " 'However when BRCA1 or BRCA2 are mutated, the repair of these genes may not occur and proper gene transcription might not occur.',\n",
       " 'This figure serves as an introduction as to how activation of these pathways by chemotherapy and radiotherapy may contribute to therapeutic resistance.',\n",
       " 'www.impactjournals.com oncotarget 4631 Oncotarget  \\x0cTable 3: Targeted Agents in Combination with Cytotoxic Therapy Official Trial Name Clinical Trial # Phase of Type of Cancer Status of Trial Trial Patient in Trial Intervention Publications  Phase Ib Dose Escalation and Biomarker Study of MK-2206 in Combination With Standard Doses of Weekly Paclitaxel NCT01263145 in Patients With Locally Advanced or Metastatic Solid Tumors With an Expansion in Advanced Breast Cancer A Phase I-II Study of Triciribine Phosphate Monohydrate (TCN-PM) Plus Sequential Weekly Paclitaxel Followed by DoseNCT01697293 Dense Doxorubicin and Cyclophosphamide in Patients With Metastatic and Locally Advanced Breast Cancer A Phase Ib II Trial of GDC-0941 (a PI3K Inhibitor) in Combination With Cisplatin in NCT01918306 Patients With Androgen Receptor Negative Triple Negative Metastatic Breast Cancer *NeoPHOEBE: PI3k Inhibition in Her2 OverExpressing Breast cancEr: A Phase II, Randomized, Parallel Cohort, Two Stage, Double-blind, Placebo-controlled Study of Neoadjuvant Trastuzumab Versus NCT01816594 Trastuzumab + BKM120 in Combination With Weekly Paclitaxel in HER2-positive, PIK3CA Wild-type and PIK3CA Mutant Primary Breast Cancer Neoadjuvant Phase II Trial of Paclitaxel in Combination With BKM120 in Endocrine Resistant Clinical Stage II or III Estrogen NCT01953445 Receptor-Positive and HER2 Negative Breast Cancer A Phase Ib Trial of Gemcitabine and Cisplatin With RAD001 in Patients With Metastatic Triple Negative Breast Cancer Proceeding NCT01939418 to an Open Label Randomized Phase II Trial Comparing Gemcitabine Cisplatin With or Without RAD001.',\n",
       " 'A Randomized, Double-Blind, PlaceboControlled Phase II Trial of Weekly Paclitaxel Bevacizumab + - Everolimus as NCT00915603 First-Line Chemotherapy for Patients With HER2-Negative Metastatic Breast Cancer (MBC) *A Phase II Study Evaluating The Efficacy And Tolerability Of Everolimus (RAD001) In Combination With Trastuzumab And NCT01305941 Vinorelbine In The Treatment Of Progressive HER2-Positive Breast Cancer Brain Metastases *Phase II Study of Everolimus in Combination With Exemestane Versus Everolimus Alone Versus Capecitabine in the Treatment of Postmenopausal Women NCT01783444 With ER+Locally Advanced, Recurrent, or Metastatic Breast Cancer After Recurrence or Progression on Prior Letrozole or Anastrozole.',\n",
       " 'I  Recurrent BC, -Stage IV BC, Active, not MK2206, paclitaxel Not Unspecified Adult recruiting Provided Solid Tumor Triciribine (an Akt inhibitor), Not paclitaxel, provided doxorubicin, cyclophosphamide Cisplatin, 0941, a inhibitor GDCNot PI3K provided  I  II  Metastatic Carcinoma Stage IV  BC, Currently Breast recruiting participants  I II  ER- BC, HER- BC, TNBC, Recurrent Currently BC, Stage IV BC recruiting participants  II  HER2+, Newly Diagnosed, Primary BC Currently Neoadjuvant recruiting Therapy, participants Trastuzumab  BKM120, trastuzumab (trastuzumab = Not herceptin, a antiprovided HER2 MoAb), paclitaxel  II  BC  Not yet open Not for participant Paclitaxel, BKM120 provided study RAD001 (Afinitor everolimus), Not gemcitabine (a provided nucleoside analog), cisplatin  I II  Metastatic BC  Currently recruiting patients  II  Metastatic BC  Everolimus, Bevacizumab (vascular Ongoing, but Not endothelial not recruiting provided growth factor A inhibitor), paclitaxel  II  HER2+ BC  Recruiting  Everolimus, vinorelbine, Not trastuzumab (HER2 provided inhibibitor, MoAb) Capecitabine (a pro-drug, which is converted to 5-flurouracil (5Not FU) which inhibits provided thymidylate synthase, exemestane, Everolimus  II  BC  Recruiting  www.impactjournals.com oncotarget  4632  Oncotarget  \\x0c*A Phase III Trials Program Exploring the Integration of Bevacizumab, Everolimus (RAD001), and Lapatinib Into Current NCT00567554 Neoadjuvant Chemotherapy Regimes for Primary Breast Cancer  III  BC  Phase I, Open Label, Dose Escalation Study of the Safety, Tolerability, and Pharmacokinetics of the Combination NCT00253318 RAD001 Plus Docetaxel in Patients With Metastatic Breast Cancer Phase II, Open Label, Dose Escalation Study of the Safety, Tolerability, and Pharmacokinetics of the Combination RAD001 Plus Docetaxel in Patients With Metastatic Breast Cancer *Phase 1b 2 Single-arm Trial Evaluating the Combination of Lapatinib, Everolimus and Capecitabine for the Treatment of Patients With HER2-positive Metastatic Breast Cancer With CNS Progression After Trastuzumab *A Phase Ib II Trial of LEE011 in Combination With Everolimus (RAD001) and Exemestane in the Treatment of Postmenopausal Women With Estrogen Receptor Positive, Her2- Locally Advanced or Metastatic Breast Cancer Phase I II Study of Weekly Abraxane and RAD001 in Women With Locally Advanced or Metastatic Breast Cancer.',\n",
       " 'A Study of the Cancer Institute of New Jersey Oncology Group (CINJOG)  I  BC  Epirubicin , cyclophosphamide, docetaxel, Ongoing, bevacizumab, [341,342] but not paclitaxel, recruiting everolimus, trastuzumab, lapatinib Docetaxel, RAD001, dexamethasone, a glucocorticoid   Not Terminated steroidial drugs that provided has anti-inflammatory, immunosuppressant properties Withdrawn Docetaxel, RAD001, Not prior to Dexamethasone provided enrollment  NCT01825265  II  BC  NCT01783756  I II  Central Nervous System Metastases HER2+ BC, Male Currently BC, Recurrent BC, recruiting Stage IV BC  Lapatinib ditosylate, everolimus, Not capecitabine provided  NCT01857193  I II  BC  Currently recruiting  LEE011, a CDK4 6 Not inhibitor, exemestane, provided everolimus  NCT00934895  I II  BC  Currently recruiting  Everolimus, abraxane, abraxane is paclitaxel Not bonded to albumin as provided a delivery vehicle Cisplatin, everolimus, Not paclitaxel provided RAD001 in combination with carboplatin.',\n",
       " 'Not Carboplatin is related provided to cisplatin, but is modified.',\n",
       " 'Paclitaxel, 5-FU, epirubicin, Not cyclophosphamide, provided RAD001  A Phase Ib II Study of Cisplatin, Paclitaxel, and RAD001 in Patients With Metastatic NCT01031446 Breast Cancer Everolimus (RAD001) in Combination With Intravenous Carboplatin in Taxaneand Anthracycline-pretreated Patients With NCT00930475 Progressive Metastatic Breast Cancer Open Label Randomized Clinical Trial of Standard Neoadjuvant Chemotherapy (Paclitaxel Followed by FEC) Versus the Combination of Paclitaxel and RAD001 NCT00499603 Followed by FEC in Women With Triple Receptor-Negative Breast Cancer (CRAD001C24101) A Phase I Study of Cisplatin, Paclitaxel, and RAD001 Patients With Metastatic Breast NCT00680758 Cancer A Phase II Neo-Adjuvant Study of Cisplatin, Paclitaxel With or Without RAD001 in Patients With Triple-negative Locally NCT00930930 Advanced Breast Cancer.',\n",
       " 'A Phase I Pilot Study of the Oral mTOR Inhibitor RAD001 in Combination With NCT00473005 Capecitabine for Metastatic Breast Cancer  I II  BC  Completed  I II  BC  Unknown  II  BC  Completed  I  BC  Completed  Cisplatin, everolimus, Not paclitaxel provided  II  BC  Ongoing, Cisplatin, everolimus, Not but not paclitaxel provided recruiting Terminated due to principle Capecitabine, investigator RAD001 leaving sponsor  I  BC  Not provided  www.impactjournals.com oncotarget  4633  Oncotarget  \\x0cPhase I Study of Combined Temosirolimus, Erlotinib and Cisplatin in Advanced Solid NCT00998036 Tumors A Phase Ib Study of Combination of Temsirolimus (Torisel®) and Pegylated Liposomal Doxorubicin (PLD, Doxil®  NCT00982631 Caelyx®) in Advanced or Recurrent Breast, Endometrial and Ovarian Cancer A Phase I, Open-Label, Multi-center Study to Assess the Safety, Tolerability and Pharmacokinetics of AZD6244 (ARRY- NCT00600496 142886) When Given in Combination With Standard Doses of Selected Chemotherapies to Patients With Advanced Solid Tumors  I  TNBC  Completed  Temsirolimus, cisplatin, erlotinib  Not provided  I  I  A Phase I Study of Lenalidomide in Combination With Bevacizumab, Sorafenib, Temsirolimus, or 5-fluorouracil, Leucovorin, NCT01183663 Oxaliplatin (FOLFOX) in Patients With Advanced Cancers  I  Recruitment is unknown Advanced Recurrent because the BC, Endometrial information Temsirolimus PLD Cancer, Ovarian has not been Cancer verified recently AZD6244 (MEK inhibitor), BC, Colon Cancer, Ongoing, but dacarbazine, Lung Cancer, not recruiting Erlotinib (EGFR1 Melanoma, Kidney participants inhibitor), Cancer docetaxel, temsirolimus Lenalidomide (related to thalidomide), bevacizumab sorafenib (Raf Ongoing, but Advanced Cancers and other kinase not recruiting inhibitor), temsirolimus, oxaliplatin, leucovorin, (Folinic acid), 5-FU Resistant Solid Completed Malignancies  Not provided  [343]  Not provided  Phase I Study of Pegylated Liposomal Doxorubicin and Temsirolimus in Resistant NCT00703170 Solid Malignancies A Phase I Study of the mTOR Inhibitor Rapamycin (Rapamune, Sirolimus) in Combination With Abraxane (Paclitaxel NCT00337376 Protein-bound Particles) in Advanced Solid Cancers A Dose-finding Phase Ib Study Followed by an Open-label, Randomized Phase II Study of BEZ235 Plus Paclitaxel in Patients NCT01495247 With HER2 Negative, Inoperable Locally Advanced or Metastatic Breast Cancer Phase I Study of Docetaxel and Temsirolimus in Resistant Solid Malignancies NCT00703625  I  Temsirolimus, Not pegylated liposomal provided doxorubicin Rapamune, Abraxane Not provided  I  Advanced Cancers  Solid  Terminated  I II  Inoperable Locally Advanced Breast Ongoing, but BEZ235, paclitaxel Cancer, Metastatic not recruiting MBC Resistant Solid Completed Malignancies Temsirolimus, docetaxel  Not provided  I  Not provided  in the progression of breast cancer [319].',\n",
       " 'Drug resistance may involve multiple approaches such as pumping the compound out of the cell, modifying or detoxifying the drug, or activating survival signaling pathways that prevent drug-induced apoptosis [287].',\n",
       " 'ATPbinding cassette transporters such as multidrug resistance protein (MRP1) and the MDR1 product, P-glycoprotein (Pgp), actively expel chemotherapeutic drugs from the cell [239,240].',\n",
       " 'Drug transporters pumps have been detected in a large number of untreated breast cancers, and their expression increases upon chemotherapy exposure [320].',\n",
       " 'Certain (e.g., EGFR inhibitors) targeted therapeutics antagonize drug transporter activity [321].',\n",
       " 'Certain CDK4 inhibitors antagonize the response of breast cancer cells to antracyclines [322].',\n",
       " 'Activation of cell survival pathways such as PI3K PTEN Akt mTORC1 pathway can prevent apoptosis in the presence of a drug by altering www.impactjournals.com oncotarget 4634  mitochondrial bioenergetics and inhibiting the release of cytochrome c and result in resistance to autophagy [323329].',\n",
       " 'The tumor microenvironment and metabolism may influence the response to targeted therapy as well as drug resistance of breast cancer [330].',\n",
       " 'Genes and biotargets are being identified which can confer resistance or sensitivity to other targeted therapies such as PARP inhibitors [331,332].',\n",
       " 'Effects of Metformin on HER2+ Breast CICs which are Resistant to Herceptin.',\n",
       " 'JIMT-1 is a human breast cancer cell line that was isolated from a pleural metastasis of a patient who was resistant to herceptin.',\n",
       " 'CICs were isolated from this cell line JIMT-1(CIC).',\n",
       " 'JIMT-1(CICs) were shown to be Oncotarget  \\x0cpreferentially sensitive to the anti-diabetes drug metformin compared to the non-CIC JIMT(BC) population.',\n",
       " 'Furthermore, inclusion of metformin in tumor xenograft studies increased the ability of Herceptin to suppress the growth of JIMT xenografts.',\n",
       " 'These important studies document the potential usefulness of metformin in the treatment of HER2+ breast cancer patients [298].',\n",
       " 'Similar studies were performed with a herceptinsensitive (SKBR3-TzbS) and a derivative Herceptinresistant (SKBR3-TzbR) breast cancer cell lines.',\n",
       " 'While herceptin inhibited the formation of mammospheres from herceptin-sensitive (SKBR3-TzbS), it did not suppress the formation of mammospheres from herceptin-resistant (SKBR3-TzbR) cells.',\n",
       " 'Metformin would reduce the formation of mamospheres from both cell types but the SKBR3-TzbR cells were more sensitive to metformin.',\n",
       " 'Metformin could be combined with herceptin and synergistically reduced mammosphere formation ability in SKBR-TzbR cells [333].',\n",
       " 'Clinical Trials with Breast Cancer Patients Regarding Key Pathways Discussed.',\n",
       " 'In Tables 1-3 and Supplementary Tables 1-3, we present the listing of clinical trials with breast cancer patients derived from the ClinicalTrials.gov data base.',\n",
       " 'We have focused our searches to the key pathways discussed in this review.',\n",
       " 'Table 1 presents the clinical trials with single inhibitors (mono therapy) such as PI3K, PI3K  mTOR, Akt, mTOR inhibitors and mTORC1 blockers (Everolimus, Temsirolimus, Rapamycin).',\n",
       " 'Some trials have presented publication regarding their clinical trials, either in the form of abstracts or scientific manuscripts [333340].',\n",
       " 'In Table 2, we list the trials which have used 2 or more agents, usually a combination of inhibitors which target two different signaling pathways.',\n",
       " 'Table 3 lists clinical trials which often combine a signal transduction inhibitor with chemotherapy.',\n",
       " 'Some of these clinical trails have presented publications in the database [341-343].',\n",
       " 'In Supplementary Table 1, clinical trials which combine hormonal therapy with targeted or chemotherapy are presented.',\n",
       " 'Some of these clinical trails have presented publications in the database [344-350].',\n",
       " 'Clinical trials using metformin as a single agent are presented in Supplementary Table 2.',\n",
       " 'Finally clinical trails combining metformin and AIs and other drugs are presented in Supplementary Table 3 as well as a publication from one study [351].',\n",
       " 'Induction of EMT can lead to Herceptin Resistance.',\n",
       " 'Induction of EMT in breast cancer is also another mechanism of resistance to such drugs as herceptin [334].',\n",
       " 'HER2 is also associated with breast cancer stem cells and their aggressiveness [335].',\n",
       " 'Inherent herceptin resistance in HER2+ breast cancers is a significant problem and it has been reported to be as high as 70% [334].',\n",
       " 'In the above study, basal HER2+ breast cancer cells resistant to herceptin were infected with lentiviruses containing small hairpin (sh) RNAs specific for EMT-specific transcription factors.',\n",
       " 'The authors had determined appropriate genes to target by analysis of the public Lawrence Berkeley Laboratory (LBL) Breast Cancer Collection database.',\n",
       " 'These authors demonstrated that herceptin sensitivity was restricted to the SLUG SNAIL2-negative subset of luminal HER2+ cell lines.',\n",
       " 'In contrast, breast cancer lines which expressed SLUG SNAIL2 were inherently resistance to herceptin.',\n",
       " 'Knockdown of SLUG  SNAIL2 suppressed the CIC phenotype by upregulating the expression of the luminal epithelial marker CD24 in basal HER2+ cells and these cells were also sensitive to herceptin and underwent the mesenchymal to epithelial transition (MET).',\n",
       " 'This important study also documented a reduction in tumor growth and sensitivity to herceptin when SLUG and SNAIL2 were knocked-down in HER+ cells in tumor xenograft studies.',\n",
       " 'This group has also postulated that other mechanisms may be involved in the induction of herceptin-refractory CICs from more differentiated cells via the activation of intrinsic or microenvironmental paths-to-stemness, also involving EMT [336].',\n",
       " 'SUMMARY HER2, EGFR1, EGFRvIII and the PI3K PTEN Akt  mTORC1 pathway are clearly important not only in the development of breast cancer, but also in the classification and prognosis of breast cancer patients which display mutations or changes in the expression of components of these pathways.',\n",
       " 'Both HER2 and PIK3CA are aberrantly expressed or mutated in many breast cancers.',\n",
       " 'These genes may be altered in breast CICs and successful targeting of them may prove to be effectively therapeutically.',\n",
       " 'HER2 and PI3K control the expression of many downstream genes involved in many different biological processes including protein translation and gene transcription.',\n",
       " 'ACKNOWLEDGEMENTS ABD was supported in part by grants from the USAMRMC BC022276, the Intramural RECDA Award and the Italian Association for Cancer Research (AIRC).',\n",
       " 'LC was supported in part by MIUR-PRIN 2009 and Italian MIUR-FIRB Accordi di Programma 2010.',\n",
       " 'MC and GM were supported in part by grants from the Italian Ministero dellIstruzione, dellUniversità e della Ricerca (Ministry for Education, Universities and Research) \\xad FIRB-MERIT (RBNE08YYBM).',\n",
       " 'MC was also supported 4635 Oncotarget  www.impactjournals.com oncotarget  \\x0cin part by a grant to the CNR from the Italian Ministry of Economy and Finance for the Project FaReBio di Qualità.',\n",
       " 'ML was supported in part by a grant from the Italian Ministry of Health, Ricerca Finalizzata Stemness 2008 entitled Molecular Determinants of Stemness and Mesenchymal Phenotype in Breast Cancer.',\n",
       " 'RM was supported in part by grants from: CRO, the Italian Association for Cancer Research, (AIRC) (RM PI grant and MCO10016), the Italian Ministry of Health, and Regione Friuli Venezia-Giulia.',\n",
       " 'BRCA1 A-complex fine tunes repair functions of BRCA1.',\n",
       " 'Aging.',\n",
       " 'BRCA1 mutations drive oxidative stress and glycolysis in the tumor microenvironment: implications for breast cancer prevention with antioxidant therapies.',\n",
       " 'Cell Cycle.',\n",
       " 'CA1  p220 loss triggers BRCA1-IRIS overexpression via mRNA stabilization in breast cancer cells.',\n",
       " 'Chang S, Sharan SK.',\n",
       " 'Epigenetic control of an oncogenic microRNA, miR-155, by BRCA1.',\n",
       " 'Adams JR, Schachter NF, Liu JC, Zacksenhaus E, Egan SE.',\n",
       " 'Elevated PI3K signaling drives multiple breast cancer subtypes.',\n",
       " 'DNA2 and EXO1 in replication-coupled, homologydirected repair and in the interplay between HDR and the FA BRCA network.',\n",
       " 'Cell Cycle.',\n",
       " 'Tarapore P, Hanashiro K, Fukasawa K. Analysis of centrosome localization of BRCA1 and its activity in suppressing centrosomal aster formation.',\n",
       " 'Cell Cycle.',\n",
       " 'Almeida MI, Reis RM, Calin GA. BRCA1, microRNAs and cancer predisposition: challenging the dogma.',\n",
       " 'Cell Cycle.',\n",
       " 'Rare BRCA1 haplotypes including 3UTR SNPs associated with breast cancer risk.',\n",
       " 'Cell Cycle.',\n",
       " 'Germ line p53 mutations in a familial syndrome of breast cancer, sarcomas, and other neoplasms.',\n",
       " 'Coles C, Condie A, Chetty U, Steel CM, Evans HJ, Prosser J. p53 mutations in breast cancer.',\n",
       " 'Cancer Res.',\n",
       " 'Predictive and prognostic impact of TP53 mutations and MDM2 promoter genotype in primary breast cancer patients treated with epirubicin or paclitaxel.',\n",
       " 'PLoS One.',\n",
       " 'doi: 10.1371  journal.pone.0019249.',\n",
       " 'Tumor suppressor activity of RB and p53 genes in human breast carcinoma cells.',\n",
       " 'Oncogene.',\n",
       " 'Cancer statistics, 2013.',\n",
       " 'CA Cancer J Clin.',\n",
       " 'Weiss JR, Moysich KB, Swede H. Epidemiology of male breast cancer.',\n",
       " 'Cancer Epidemiol Biomarkers Prev.',\n",
       " 'Deb S, Jene N, Kconfab Investigators, Fox SB.',\n",
       " 'Genotypic and phenotypic analysis of familial male breast cancer shows under representation of the HER2 and basal subtypes in BRCA-associated carcinomas.',\n",
       " 'BMC Cancer.',\n",
       " 'Miki Y, Swensen J, Shattuck-Eidens D, Futreal PA, Harshman K, Tavtigian S, Liu Q, Cochran C, Bennett LM, Ding W, Bell R, Rosenthal J, Hussey C, Tran T, McClure M, Fry Cet al.',\n",
       " 'A strong candidate for the breast and ovarian cancer susceptibility gene BRCA1.',\n",
       " 'Cancer risks in two large breast cancer families linked to BRCA2 on chromosome 13q12-13.',\n",
       " 'Am J Hum Genet.',\n",
       " 'Mutation of the PIK3CA gene in ovarian and breast cancer.',\n",
       " 'Cancer Res.',\n",
       " 'Evaluating HER2 amplification and overexpression in breast cancer.',\n",
       " 'Vogel CL, Cobleigh MA, Tripathy D, Gutheil JC, Harris LN, Fehrenbacher L, Slamon DJ, Murphy M, Novotny WF, Burchmore M, Shak S, Stewart SJ, Press M. Efficacy and safety of trastuzumab as a single agent in first-line treatment of HER2-overexpressing metastatic breast cancer.',\n",
       " 'J Clin Oncol.',\n",
       " 'Activating HER2 mutations in HER2 gene amplification negative breast cancer.',\n",
       " 'Cancer Discov.',\n",
       " 'Network modeling of MDM2 inhibitoroxaliplatin combination reveals biological synergy in wtp53 solid tumors.',\n",
       " 'MDM2 promoter SNP285 and SNP309; phylogeny and impact on cancer risk.',\n",
       " 'Bazarov AV, Lee WJ, Bazarov I, Bosire M, Hines WC, Stankovich B, Chicas A, Lowe SW, Yaswen P. The specific role of pRb in p16 (INK4A) -mediated arrest of normal and malignant human breast cells.',\n",
       " 'Cell Cycle.',\n",
       " 'Jiang Z, Jones R, Liu JC, Deng T, Robinson T, Chung PE, Wang S, Herschkowitz JI, Egan SE, Perou CM, Zacksenhaus E. RB1 and p53 at the crossroad of EMT and triple-negative breast cancer.',\n",
       " 'Cell Cycle.',\n",
       " 'Puca R, Nardinocchi L, Porru M, Simon AJ, Rechavi G, Leonetti C, Givol D, DOrazi G. Restoring p53 active conformation by zinc increases the response of mutant p53 tumor cells to anticancer drugs.',\n",
       " 'Cell Cycle.',\n",
       " 'The multiple nuclear functions of BRCA1: transcription, ubiquitination and DNA repair.',\n",
       " 'Current Opin Cell Biol.',\n",
       " 'BASC, a super complex of BRCA1associated proteins involved in the recognition and repair of aberrant DNA structures.',\n",
       " 'Genes Dev.',\n",
       " 'Role of BRCA1 and BRCA2 as regulators of DNA repair, transcription, and cell cycle in response to DNA damage.',\n",
       " 'Cancer Sci.',\n",
       " 'Xia B, Sheng Q, Nakanishi K, Ohashi A, Wu J, Christ N, Liu X, Jasin M, Couch FJ, Livingston DM.',\n",
       " 'Control of BRCA2 cellular and clinical functions by a nuclear partner, PALB2.',\n",
       " 'Cell.',\n",
       " 'Kobayashi H, Ohno S, Sasaki Y, Matsuura M. Hereditary breast and ovarian cancer susceptibility genes.',\n",
       " 'Oncol Rep. 2013; 30: 1019-1029.',\n",
       " ...]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sentences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "files = [doc_names[i] for i in non_empty_rows]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['guidelines.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " 'test.txt',\n",
       " ...]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "vocab_mat_clean=sparse_matrix_functions.remove_zero_rows(vocab_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<2024x267 sparse matrix of type '<class 'numpy.int64'>'\n",
       "\twith 12901 stored elements in Compressed Sparse Row format>"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vocab_mat_clean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  (0, 107)\t1\n",
      "  (0, 229)\t1\n",
      "  (0, 66)\t2\n",
      "  (0, 254)\t1\n",
      "  (0, 220)\t1\n",
      "  (0, 126)\t1\n",
      "  (1, 220)\t1\n",
      "  (1, 8)\t1\n",
      "  (1, 19)\t2\n",
      "  (1, 20)\t1\n",
      "  (1, 0)\t1\n",
      "  (1, 172)\t1\n",
      "  (1, 188)\t1\n",
      "  (1, 53)\t1\n",
      "  (1, 54)\t3\n",
      "  (1, 246)\t1\n",
      "  (1, 170)\t2\n",
      "  (1, 51)\t4\n",
      "  (1, 63)\t1\n",
      "  (1, 5)\t1\n",
      "  (1, 157)\t1\n",
      "  (1, 245)\t1\n",
      "  (1, 247)\t2\n",
      "  (1, 221)\t1\n",
      "  (1, 174)\t1\n",
      "  :\t:\n",
      "  (2015, 54)\t2\n",
      "  (2016, 220)\t1\n",
      "  (2016, 54)\t1\n",
      "  (2016, 44)\t1\n",
      "  (2017, 53)\t1\n",
      "  (2017, 54)\t1\n",
      "  (2017, 210)\t1\n",
      "  (2017, 180)\t1\n",
      "  (2017, 213)\t1\n",
      "  (2017, 231)\t1\n",
      "  (2017, 100)\t2\n",
      "  (2017, 83)\t1\n",
      "  (2017, 22)\t1\n",
      "  (2018, 54)\t1\n",
      "  (2019, 210)\t1\n",
      "  (2020, 110)\t1\n",
      "  (2020, 212)\t2\n",
      "  (2020, 241)\t1\n",
      "  (2021, 216)\t2\n",
      "  (2021, 110)\t1\n",
      "  (2022, 262)\t1\n",
      "  (2022, 187)\t1\n",
      "  (2023, 220)\t1\n",
      "  (2023, 54)\t1\n",
      "  (2023, 44)\t1\n"
     ]
    }
   ],
   "source": [
    "print(vocab_mat_clean)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2024, 267)"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vocab_mat_clean.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "b = vocab_mat_clean.nonzero()[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "c = vocab_mat_clean.nonzero()[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rows=[]\n",
    "for i in b:\n",
    "    rows.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12901"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cols=[]\n",
    "for i in b:\n",
    "    cols.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12901"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "d=[]\n",
    "for i in range(len(cols)):\n",
    "    d.append((rows[i],cols[i]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14643\n"
     ]
    }
   ],
   "source": [
    "print(vocab_mat_clean.sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import sklearn.preprocessing as pp\n",
    "\n",
    "def cosine_similarities(mat):\n",
    "    col_normed_mat = pp.normalize(mat.tocsc(), axis=0)\n",
    "    return col_normed_mat.T * col_normed_mat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/ale/anaconda3/lib/python3.6/site-packages/sklearn/utils/validation.py:429: DataConversionWarning: Data with input dtype int64 was converted to float64 by the normalize function.\n",
      "  warnings.warn(msg, _DataConversionWarning)\n"
     ]
    }
   ],
   "source": [
    "a = cosine_similarities(vocab_mat_clean)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[107, 229, 66, 254, 220, 126]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(vocab_mat_clean.getrow(0).nonzero()[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "Ks = vocab_mat_clean\n",
    "value = Ks.data\n",
    "column_index = Ks.indices\n",
    "row_pointers = Ks.indptr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(12901,)"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "value.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([107, 229,  66, ..., 220,  54,  44], dtype=int32)"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "column_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([    0,     6,    30, ..., 12896, 12898, 12901], dtype=int32)"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "row_pointers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12901"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(column_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2025"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(row_pointers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rows, cols = vocab_matrix.nonzero()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([   0,    0,    0, ..., 2892, 2892, 2892], dtype=int32)"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([126, 220, 254, ...,  44,  54, 220], dtype=int32)"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "266"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "max(cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'resist_col' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-136-e8a08445e206>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mresist_mat\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvocab_matrix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmultiply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresist_col\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'resist_col' is not defined"
     ]
    }
   ],
   "source": [
    "resist_mat = vocab_matrix.multiply(resist_col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'sensit_col' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-59-169b7a443729>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0msensit_mat\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvocab_matrix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmultiply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msensit_col\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'sensit_col' is not defined"
     ]
    }
   ],
   "source": [
    "sensit_mat = vocab_matrix.multiply(sensit_col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#gene_dict.values()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "index() takes at least 1 argument (0 given)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-150-e941db88542b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mvocab_columns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: index() takes at least 1 argument (0 given)"
     ]
    }
   ],
   "source": [
    "vocab_columns.index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_columns_index =[vocab_columns.index(i) for i in vocab_columns if i in gene_values] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_columns_name =[i for i in vocab_columns if i in gene_values] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "drug_columns_name =[i for i in vocab_columns if i in drug_values] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "drug_columns_index =[vocab_columns.index(i) for i in vocab_columns if i in drug_values] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2893, 406)"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vocab_matrix.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "drug_mat=vocab_matrix[:,drug_columns_index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_mat=vocab_matrix[:,gene_columns_index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<328x76 sparse matrix of type '<class 'numpy.int64'>'\n",
       "\twith 536 stored elements in Compressed Sparse Column format>"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mat.T*drug_mat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def drop_columns(X,index_to_drop):\n",
    "    to_keep = list(set(range(X.shape[1]-1))-set(index_to_drop))    \n",
    "    new_X = X[:,to_keep]\n",
    "    return new_X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "resist = drop_columns(vocab_matrix, [vocab_columns.index('sensit')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2893, 405)"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resist.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sensit = drop_columns(vocab_matrix, [vocab_columns.index('resist')])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#sparse_matrix_functions.column_indexing(drug_columns_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def make_network_matrix(X, vocabulary):\n",
    "    #drug_columns_name =[i for i in vocabulary if i in drug_values] \n",
    "    drug_columns_index =[vocabulary.index(i) for i in vocabulary if i in drug_values]    \n",
    "    gene_columns_index =[vocabulary.index(i) for i in vocabulary if i in gene_values] \n",
    "    #gene_columns_name =[i for i in vocabulary if i in gene_values] \n",
    "    gene_mat=X[:,gene_columns_index]\n",
    "    drug_mat=X[:,drug_columns_index]\n",
    "    network_matrix = gene_mat.T*drug_mat\n",
    "    return network_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_network_rows(vocabulary):\n",
    "    genes = {}\n",
    "    genes_columns_names=[i for i in vocabulary if i in gene_values]\n",
    "    for idx,name in enumerate(genes_columns_names):\n",
    "        genes[idx]=name\n",
    "    return genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "network_genes=get_network_rows(vocab_columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_network_columns(vocabulary):\n",
    "    drugs = {}\n",
    "    drugs_columns_names=[i for i in vocabulary if i in drug_values]\n",
    "    for idx,name in enumerate(drugs_columns_names):\n",
    "        drugs[idx]=name\n",
    "    return drugs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "network_drugs =get_network_columns(vocab_columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'#voxtalisib#'"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "network_drugs[75]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'#dgcr2#'"
      ]
     },
     "execution_count": 181,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "network_genes[75]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "97"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col_idx['#dgcr2#']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "396"
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col_idx[network_drugs[75]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sensit_network_matrix = make_network_matrix(sensit, vocab_columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#sensit_network_matrix.nonzero()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<328x76 sparse matrix of type '<class 'numpy.int64'>'\n",
       "\twith 536 stored elements in Compressed Sparse Column format>"
      ]
     },
     "execution_count": 186,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sensit_network_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "536"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(list(sensit_network_matrix.nonzero()[1]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "a =zip(list(sensit_network_matrix.nonzero()[0]),list(sensit_network_matrix.nonzero()[1]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('#abcg2#', '#mitoxantrone#')\n",
      "[[ 1.]]\n",
      "('#acpp#', '#binimetinib#')\n",
      "[[ 6.]]\n",
      "('#acpp#', '#buparlisib#')\n",
      "[[ 6.]]\n",
      "('#acpp#', '#dalotuzumab#')\n",
      "[[ 3.]]\n",
      "('#acpp#', '#everolimus#')\n",
      "[[ 6.]]\n",
      "('#acpp#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 6.]]\n",
      "('#acpp#', '#ridaforolimus#')\n",
      "[[ 18.]]\n",
      "('#acpp#', '#temsirolimus#')\n",
      "[[ 9.]]\n",
      "('#acpp#', '#vinorelbine#')\n",
      "[[ 3.]]\n",
      "('#acpp#', '#voxtalisib#')\n",
      "[[ 6.]]\n",
      "('#ado#', '#trastuzumab emtansine#')\n",
      "[[ 2.]]\n",
      "('#ado#', '#trastuzumab#')\n",
      "[[ 2.]]\n",
      "('#akt1#', '#afuresertib#')\n",
      "[[ 1.]]\n",
      "('#akt1#', '#akt inhibitor gsk2141795#')\n",
      "[[ 10.]]\n",
      "('#akt1#', '#akt inhibitor mk2206#')\n",
      "[[ 17.]]\n",
      "('#akt1#', '#bevacizumab#')\n",
      "[[ 2.]]\n",
      "('#akt1#', '#binimetinib#')\n",
      "[[ 1.]]\n",
      "('#akt1#', '#buparlisib#')\n",
      "[[ 14.]]\n",
      "('#akt1#', '#capecitabine#')\n",
      "[[ 2.]]\n",
      "('#akt1#', '#carboplatin#')\n",
      "[[ 5.]]\n",
      "('#akt1#', '#cisplatin#')\n",
      "[[ 7.]]\n",
      "('#akt1#', '#cyclophosphamide#')\n",
      "[[ 2.]]\n",
      "('#akt1#', '#docetaxel#')\n",
      "[[ 2.]]\n",
      "('#akt1#', '#doxorubicin#')\n",
      "[[ 1.]]\n",
      "('#akt1#', '#everolimus#')\n",
      "[[ 120.]]\n",
      "('#akt1#', '#exemestane#')\n",
      "[[ 7.]]\n",
      "('#akt1#', '#gemcitabine#')\n",
      "[[ 1.]]\n",
      "('#akt1#', '#lapatinib#')\n",
      "[[ 5.]]\n",
      "('#akt1#', '#letrozole#')\n",
      "[[ 9.]]\n",
      "('#akt1#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 5.]]\n",
      "('#akt1#', '#paclitaxel albumin-stabilized nanoparticle formulation#')\n",
      "[[ 1.]]\n",
      "('#akt1#', '#paclitaxel#')\n",
      "[[ 18.]]\n",
      "('#akt1#', '#perifosine#')\n",
      "[[ 1.]]\n",
      "('#akt1#', '#pertuzumab#')\n",
      "[[ 1.]]\n",
      "('#akt1#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 11.]]\n",
      "('#akt1#', '#recombinant interleukin-6#')\n",
      "[[ 1.]]\n",
      "('#akt1#', '#ribociclib#')\n",
      "[[ 1.]]\n",
      "('#akt1#', '#tamoxifen#')\n",
      "[[ 1.]]\n",
      "('#akt1#', '#temsirolimus#')\n",
      "[[ 16.]]\n",
      "('#akt1#', '#trametinib#')\n",
      "[[ 2.]]\n",
      "('#akt1#', '#trans sodium crocetinate#')\n",
      "[[ 2.]]\n",
      "('#akt1#', '#trastuzumab#')\n",
      "[[ 18.]]\n",
      "('#akt1#', '#triciribine phosphate#')\n",
      "[[ 1.]]\n",
      "('#akt1#', '#vinorelbine#')\n",
      "[[ 6.]]\n",
      "('#alb#', '#bevacizumab#')\n",
      "[[ 1.]]\n",
      "('#alb#', '#capecitabine#')\n",
      "[[ 1.]]\n",
      "('#alb#', '#carboplatin#')\n",
      "[[ 1.]]\n",
      "('#alb#', '#cisplatin#')\n",
      "[[ 1.]]\n",
      "('#alb#', '#cyclophosphamide#')\n",
      "[[ 1.]]\n",
      "('#alb#', '#dexamethasone#')\n",
      "[[ 2.]]\n",
      "('#alb#', '#docetaxel#')\n",
      "[[ 3.]]\n",
      "('#alb#', '#epirubicin#')\n",
      "[[ 1.]]\n",
      "('#alb#', '#everolimus#')\n",
      "[[ 8.]]\n",
      "('#alb#', '#exemestane#')\n",
      "[[ 1.]]\n",
      "('#alb#', '#lapatinib#')\n",
      "[[ 2.]]\n",
      "('#alb#', '#paclitaxel albumin-stabilized nanoparticle formulation#')\n",
      "[[ 2.]]\n",
      "('#alb#', '#paclitaxel#')\n",
      "[[ 3.]]\n",
      "('#alb#', '#ribociclib#')\n",
      "[[ 1.]]\n",
      "('#alb#', '#trastuzumab#')\n",
      "[[ 1.]]\n",
      "('#anpep#', '#lapatinib#')\n",
      "[[ 1.]]\n",
      "('#ar#', '#anastrozole#')\n",
      "[[ 1.]]\n",
      "('#ar#', '#exemestane#')\n",
      "[[ 1.]]\n",
      "('#ar#', '#letrozole#')\n",
      "[[ 1.]]\n",
      "('#armc9#', '#akt inhibitor mk2206#')\n",
      "[[ 1.]]\n",
      "('#armc9#', '#bevacizumab#')\n",
      "[[ 2.]]\n",
      "('#armc9#', '#buparlisib#')\n",
      "[[ 2.]]\n",
      "('#armc9#', '#capecitabine#')\n",
      "[[ 2.]]\n",
      "('#armc9#', '#cisplatin#')\n",
      "[[ 2.]]\n",
      "('#armc9#', '#cyclophosphamide#')\n",
      "[[ 1.]]\n",
      "('#armc9#', '#docetaxel#')\n",
      "[[ 2.]]\n",
      "('#armc9#', '#doxorubicin#')\n",
      "[[ 1.]]\n",
      "('#armc9#', '#everolimus#')\n",
      "[[ 14.]]\n",
      "('#armc9#', '#exemestane#')\n",
      "[[ 2.]]\n",
      "('#armc9#', '#gemcitabine#')\n",
      "[[ 1.]]\n",
      "('#armc9#', '#lapatinib#')\n",
      "[[ 2.]]\n",
      "('#armc9#', '#paclitaxel albumin-stabilized nanoparticle formulation#')\n",
      "[[ 1.]]\n",
      "('#armc9#', '#paclitaxel#')\n",
      "[[ 5.]]\n",
      "('#armc9#', '#ribociclib#')\n",
      "[[ 1.]]\n",
      "('#armc9#', '#trastuzumab#')\n",
      "[[ 6.]]\n",
      "('#armc9#', '#triciribine phosphate#')\n",
      "[[ 1.]]\n",
      "('#armc9#', '#vinorelbine#')\n",
      "[[ 1.]]\n",
      "('#birc5#', '#trastuzumab#')\n",
      "[[ 3.]]\n",
      "('#brca1#', '#cisplatin#')\n",
      "[[ 1.]]\n",
      "('#c19orf66#', '#tamoxifen#')\n",
      "[[ 1.]]\n",
      "('#cd24#', '#trastuzumab#')\n",
      "[[ 1.]]\n",
      "('#cdk4#', '#bevacizumab#')\n",
      "[[ 1.]]\n",
      "('#cdk4#', '#capecitabine#')\n",
      "[[ 1.]]\n",
      "('#cdk4#', '#carboplatin#')\n",
      "[[ 1.]]\n",
      "('#cdk4#', '#cisplatin#')\n",
      "[[ 1.]]\n",
      "('#cdk4#', '#cyclophosphamide#')\n",
      "[[ 1.]]\n",
      "('#cdk4#', '#dexamethasone#')\n",
      "[[ 2.]]\n",
      "('#cdk4#', '#docetaxel#')\n",
      "[[ 3.]]\n",
      "('#cdk4#', '#epirubicin#')\n",
      "[[ 1.]]\n",
      "('#cdk4#', '#everolimus#')\n",
      "[[ 8.]]\n",
      "('#cdk4#', '#exemestane#')\n",
      "[[ 1.]]\n",
      "('#cdk4#', '#lapatinib#')\n",
      "[[ 2.]]\n",
      "('#cdk4#', '#paclitaxel albumin-stabilized nanoparticle formulation#')\n",
      "[[ 2.]]\n",
      "('#cdk4#', '#paclitaxel#')\n",
      "[[ 3.]]\n",
      "('#cdk4#', '#ribociclib#')\n",
      "[[ 1.]]\n",
      "('#cdk4#', '#trastuzumab#')\n",
      "[[ 1.]]\n",
      "('#cebpd#', '#folitixorin#')\n",
      "[[ 1.]]\n",
      "('#celp#', '#afuresertib#')\n",
      "[[ 1.]]\n",
      "('#celp#', '#akt inhibitor mk2206#')\n",
      "[[ 1.]]\n",
      "('#celp#', '#binimetinib#')\n",
      "[[ 29.]]\n",
      "('#celp#', '#buparlisib#')\n",
      "[[ 30.]]\n",
      "('#celp#', '#cixutumumab#')\n",
      "[[ 1.]]\n",
      "('#celp#', '#dalotuzumab#')\n",
      "[[ 14.]]\n",
      "('#celp#', '#erlotinib#')\n",
      "[[ 2.]]\n",
      "('#celp#', '#everolimus#')\n",
      "[[ 59.]]\n",
      "('#celp#', '#iproplatin#')\n",
      "[[ 2.]]\n",
      "('#celp#', '#lapatinib#')\n",
      "[[ 3.]]\n",
      "('#celp#', '#letrozole#')\n",
      "[[ 3.]]\n",
      "('#celp#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 2.]]\n",
      "('#celp#', '#niacinamide#')\n",
      "[[ 2.]]\n",
      "('#celp#', '#pi3k-beta inhibitor gsk2636771#')\n",
      "[[ 1.]]\n",
      "('#celp#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 30.]]\n",
      "('#celp#', '#recombinant interleukin-6#')\n",
      "[[ 1.]]\n",
      "('#celp#', '#ridaforolimus#')\n",
      "[[ 84.]]\n",
      "('#celp#', '#tamoxifen citrate#')\n",
      "[[ 4.]]\n",
      "('#celp#', '#tamoxifen#')\n",
      "[[ 2.]]\n",
      "('#celp#', '#temsirolimus#')\n",
      "[[ 45.]]\n",
      "('#celp#', '#trametinib#')\n",
      "[[ 2.]]\n",
      "('#celp#', '#trastuzumab#')\n",
      "[[ 7.]]\n",
      "('#celp#', '#vinorelbine#')\n",
      "[[ 15.]]\n",
      "('#celp#', '#voxtalisib#')\n",
      "[[ 28.]]\n",
      "('#cic#', '#letrozole#')\n",
      "[[ 1.]]\n",
      "('#cic#', '#recombinant interleukin-6#')\n",
      "[[ 6.]]\n",
      "('#cic#', '#trastuzumab#')\n",
      "[[ 3.]]\n",
      "('#cyp19a1#', '#everolimus#')\n",
      "[[ 1.]]\n",
      "('#cyp19a1#', '#fulvestrant#')\n",
      "[[ 1.]]\n",
      "('#cyp19a1#', '#letrozole#')\n",
      "[[ 2.]]\n",
      "('#cyp19a1#', '#pertuzumab#')\n",
      "[[ 1.]]\n",
      "('#cyp19a1#', '#tamoxifen#')\n",
      "[[ 2.]]\n",
      "('#cyp19a1#', '#trastuzumab#')\n",
      "[[ 1.]]\n",
      "('#dcstamp#', '#akt inhibitor mk2206#')\n",
      "[[ 1.]]\n",
      "('#dgcr2#', '#tamoxifen#')\n",
      "[[ 1.]]\n",
      "('#dlx4#', '#trans sodium crocetinate#')\n",
      "[[ 2.]]\n",
      "('#dpm2#', '#recombinant interleukin-6#')\n",
      "[[ 1.]]\n",
      "('#eec1#', '#pi3k-beta inhibitor gsk2636771#')\n",
      "[[ 1.]]\n",
      "('#eec1#', '#temsirolimus#')\n",
      "[[ 1.]]\n",
      "('#egfr#', '#afuresertib#')\n",
      "[[ 1.]]\n",
      "('#egfr#', '#alpelisib#')\n",
      "[[ 1.]]\n",
      "('#egfr#', '#binimetinib#')\n",
      "[[ 1.]]\n",
      "('#egfr#', '#buparlisib#')\n",
      "[[ 2.]]\n",
      "('#egfr#', '#cixutumumab#')\n",
      "[[ 1.]]\n",
      "('#egfr#', '#erlotinib#')\n",
      "[[ 2.]]\n",
      "('#egfr#', '#everolimus#')\n",
      "[[ 7.]]\n",
      "('#egfr#', '#lapatinib#')\n",
      "[[ 5.]]\n",
      "('#egfr#', '#pertuzumab#')\n",
      "[[ 1.]]\n",
      "('#egfr#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 2.]]\n",
      "('#egfr#', '#temsirolimus#')\n",
      "[[ 3.]]\n",
      "('#egfr#', '#trametinib#')\n",
      "[[ 2.]]\n",
      "('#egfr#', '#vinorelbine#')\n",
      "[[ 1.]]\n",
      "('#egr3#', '#bevacizumab#')\n",
      "[[ 2.]]\n",
      "('#egr3#', '#capecitabine#')\n",
      "[[ 2.]]\n",
      "('#egr3#', '#carbendazim#')\n",
      "[[ 1.]]\n",
      "('#egr3#', '#cisplatin#')\n",
      "[[ 4.]]\n",
      "('#egr3#', '#dacarbazine#')\n",
      "[[ 1.]]\n",
      "('#egr3#', '#docetaxel#')\n",
      "[[ 3.]]\n",
      "('#egr3#', '#doxorubicin#')\n",
      "[[ 3.]]\n",
      "('#egr3#', '#erlotinib#')\n",
      "[[ 3.]]\n",
      "('#egr3#', '#everolimus#')\n",
      "[[ 4.]]\n",
      "('#egr3#', '#fluorouracil#')\n",
      "[[ 1.]]\n",
      "('#egr3#', '#lenalidomide#')\n",
      "[[ 2.]]\n",
      "('#egr3#', '#leucovorin calcium#')\n",
      "[[ 2.]]\n",
      "('#egr3#', '#oxaliplatin#')\n",
      "[[ 2.]]\n",
      "('#egr3#', '#paclitaxel albumin-stabilized nanoparticle formulation#')\n",
      "[[ 2.]]\n",
      "('#egr3#', '#paclitaxel#')\n",
      "[[ 5.]]\n",
      "('#egr3#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 2.]]\n",
      "('#egr3#', '#selumetinib#')\n",
      "[[ 2.]]\n",
      "('#egr3#', '#sorafenib tosylate#')\n",
      "[[ 2.]]\n",
      "('#egr3#', '#temsirolimus#')\n",
      "[[ 10.]]\n",
      "('#egr3#', '#thalidomide#')\n",
      "[[ 1.]]\n",
      "('#erbb2#', '#akt inhibitor gsk2141795#')\n",
      "[[ 12.]]\n",
      "('#erbb2#', '#akt inhibitor mk2206#')\n",
      "[[ 24.]]\n",
      "('#erbb2#', '#anastrozole#')\n",
      "[[ 2.]]\n",
      "('#erbb2#', '#bevacizumab#')\n",
      "[[ 18.]]\n",
      "('#erbb2#', '#buparlisib#')\n",
      "[[ 30.]]\n",
      "('#erbb2#', '#capecitabine#')\n",
      "[[ 17.]]\n",
      "('#erbb2#', '#carbendazim#')\n",
      "[[ 6.]]\n",
      "('#erbb2#', '#carboplatin#')\n",
      "[[ 7.]]\n",
      "('#erbb2#', '#cisplatin#')\n",
      "[[ 32.]]\n",
      "('#erbb2#', '#cyclophosphamide#')\n",
      "[[ 10.]]\n",
      "('#erbb2#', '#dacarbazine#')\n",
      "[[ 1.]]\n",
      "('#erbb2#', '#dexamethasone#')\n",
      "[[ 2.]]\n",
      "('#erbb2#', '#docetaxel#')\n",
      "[[ 22.]]\n",
      "('#erbb2#', '#doxorubicin#')\n",
      "[[ 12.]]\n",
      "('#erbb2#', '#epirubicin#')\n",
      "[[ 1.]]\n",
      "('#erbb2#', '#erlotinib#')\n",
      "[[ 3.]]\n",
      "('#erbb2#', '#everolimus#')\n",
      "[[ 242.]]\n",
      "('#erbb2#', '#exemestane#')\n",
      "[[ 21.]]\n",
      "('#erbb2#', '#fluorouracil#')\n",
      "[[ 1.]]\n",
      "('#erbb2#', '#gemcitabine#')\n",
      "[[ 12.]]\n",
      "('#erbb2#', '#lapatinib#')\n",
      "[[ 39.]]\n",
      "('#erbb2#', '#lenalidomide#')\n",
      "[[ 2.]]\n",
      "('#erbb2#', '#letrozole#')\n",
      "[[ 3.]]\n",
      "('#erbb2#', '#leucovorin calcium#')\n",
      "[[ 2.]]\n",
      "('#erbb2#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 6.]]\n",
      "('#erbb2#', '#neratinib#')\n",
      "[[ 2.]]\n",
      "('#erbb2#', '#oxaliplatin#')\n",
      "[[ 2.]]\n",
      "('#erbb2#', '#paclitaxel albumin-stabilized nanoparticle formulation#')\n",
      "[[ 10.]]\n",
      "('#erbb2#', '#paclitaxel#')\n",
      "[[ 62.]]\n",
      "('#erbb2#', '#pertuzumab#')\n",
      "[[ 18.]]\n",
      "('#erbb2#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 14.]]\n",
      "('#erbb2#', '#recombinant interleukin-6#')\n",
      "[[ 2.]]\n",
      "('#erbb2#', '#ribociclib#')\n",
      "[[ 7.]]\n",
      "('#erbb2#', '#selumetinib#')\n",
      "[[ 2.]]\n",
      "('#erbb2#', '#sorafenib tosylate#')\n",
      "[[ 2.]]\n",
      "('#erbb2#', '#tamoxifen#')\n",
      "[[ 1.]]\n",
      "('#erbb2#', '#temsirolimus#')\n",
      "[[ 28.]]\n",
      "('#erbb2#', '#thalidomide#')\n",
      "[[ 1.]]\n",
      "('#erbb2#', '#trastuzumab#')\n",
      "[[ 111.]]\n",
      "('#erbb2#', '#triciribine phosphate#')\n",
      "[[ 12.]]\n",
      "('#erbb2#', '#vinorelbine#')\n",
      "[[ 16.]]\n",
      "('#esr1#', '#iproplatin#')\n",
      "[[ 6.]]\n",
      "('#esr1#', '#niacinamide#')\n",
      "[[ 2.]]\n",
      "('#esr1#', '#tamoxifen citrate#')\n",
      "[[ 2.]]\n",
      "('#esr1#', '#tamoxifen#')\n",
      "[[ 2.]]\n",
      "('#esr2#', '#iproplatin#')\n",
      "[[ 1.]]\n",
      "('#esrra#', '#iproplatin#')\n",
      "[[ 2.]]\n",
      "('#esrrb#', '#iproplatin#')\n",
      "[[ 1.]]\n",
      "('#fancc#', '#cyclophosphamide#')\n",
      "[[ 2.]]\n",
      "('#fancc#', '#docetaxel#')\n",
      "[[ 1.]]\n",
      "('#fancc#', '#doxorubicin#')\n",
      "[[ 2.]]\n",
      "('#fancc#', '#fluorouracil#')\n",
      "[[ 1.]]\n",
      "('#gaa#', '#bevacizumab#')\n",
      "[[ 2.]]\n",
      "('#gaa#', '#capecitabine#')\n",
      "[[ 2.]]\n",
      "('#gaa#', '#carbendazim#')\n",
      "[[ 1.]]\n",
      "('#gaa#', '#cisplatin#')\n",
      "[[ 4.]]\n",
      "('#gaa#', '#dacarbazine#')\n",
      "[[ 1.]]\n",
      "('#gaa#', '#docetaxel#')\n",
      "[[ 3.]]\n",
      "('#gaa#', '#doxorubicin#')\n",
      "[[ 3.]]\n",
      "('#gaa#', '#erlotinib#')\n",
      "[[ 3.]]\n",
      "('#gaa#', '#everolimus#')\n",
      "[[ 4.]]\n",
      "('#gaa#', '#fluorouracil#')\n",
      "[[ 1.]]\n",
      "('#gaa#', '#lenalidomide#')\n",
      "[[ 2.]]\n",
      "('#gaa#', '#leucovorin calcium#')\n",
      "[[ 2.]]\n",
      "('#gaa#', '#oxaliplatin#')\n",
      "[[ 2.]]\n",
      "('#gaa#', '#paclitaxel albumin-stabilized nanoparticle formulation#')\n",
      "[[ 2.]]\n",
      "('#gaa#', '#paclitaxel#')\n",
      "[[ 5.]]\n",
      "('#gaa#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 2.]]\n",
      "('#gaa#', '#selumetinib#')\n",
      "[[ 2.]]\n",
      "('#gaa#', '#sorafenib tosylate#')\n",
      "[[ 2.]]\n",
      "('#gaa#', '#temsirolimus#')\n",
      "[[ 10.]]\n",
      "('#gaa#', '#thalidomide#')\n",
      "[[ 1.]]\n",
      "('#glys1#', '#afuresertib#')\n",
      "[[ 1.]]\n",
      "('#glys1#', '#binimetinib#')\n",
      "[[ 7.]]\n",
      "('#glys1#', '#buparlisib#')\n",
      "[[ 8.]]\n",
      "('#glys1#', '#cixutumumab#')\n",
      "[[ 1.]]\n",
      "('#glys1#', '#dalotuzumab#')\n",
      "[[ 3.]]\n",
      "('#glys1#', '#erlotinib#')\n",
      "[[ 2.]]\n",
      "('#glys1#', '#everolimus#')\n",
      "[[ 13.]]\n",
      "('#glys1#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 8.]]\n",
      "('#glys1#', '#ridaforolimus#')\n",
      "[[ 18.]]\n",
      "('#glys1#', '#temsirolimus#')\n",
      "[[ 12.]]\n",
      "('#glys1#', '#trametinib#')\n",
      "[[ 2.]]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('#glys1#', '#vinorelbine#')\n",
      "[[ 4.]]\n",
      "('#glys1#', '#voxtalisib#')\n",
      "[[ 6.]]\n",
      "('#hdac9#', '#tamoxifen citrate#')\n",
      "[[ 2.]]\n",
      "('#hpp1#', '#anastrozole#')\n",
      "[[ 1.]]\n",
      "('#hpp1#', '#bevacizumab#')\n",
      "[[ 1.]]\n",
      "('#hpp1#', '#capecitabine#')\n",
      "[[ 1.]]\n",
      "('#hpp1#', '#carbendazim#')\n",
      "[[ 1.]]\n",
      "('#hpp1#', '#carboplatin#')\n",
      "[[ 1.]]\n",
      "('#hpp1#', '#cisplatin#')\n",
      "[[ 3.]]\n",
      "('#hpp1#', '#cyclophosphamide#')\n",
      "[[ 1.]]\n",
      "('#hpp1#', '#epirubicin#')\n",
      "[[ 1.]]\n",
      "('#hpp1#', '#everolimus#')\n",
      "[[ 12.]]\n",
      "('#hpp1#', '#exemestane#')\n",
      "[[ 1.]]\n",
      "('#hpp1#', '#letrozole#')\n",
      "[[ 1.]]\n",
      "('#hpp1#', '#paclitaxel#')\n",
      "[[ 7.]]\n",
      "('#hpp1#', '#trastuzumab#')\n",
      "[[ 1.]]\n",
      "('#hpp1#', '#vinorelbine#')\n",
      "[[ 1.]]\n",
      "('#hr#', '#akt inhibitor gsk2141795#')\n",
      "[[ 2.]]\n",
      "('#hr#', '#akt inhibitor mk2206#')\n",
      "[[ 3.]]\n",
      "('#hr#', '#buparlisib#')\n",
      "[[ 2.]]\n",
      "('#hr#', '#carboplatin#')\n",
      "[[ 1.]]\n",
      "('#hr#', '#cisplatin#')\n",
      "[[ 1.]]\n",
      "('#hr#', '#everolimus#')\n",
      "[[ 21.]]\n",
      "('#hr#', '#exemestane#')\n",
      "[[ 1.]]\n",
      "('#hr#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 1.]]\n",
      "('#hr#', '#paclitaxel#')\n",
      "[[ 1.]]\n",
      "('#hr#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 2.]]\n",
      "('#hr#', '#temsirolimus#')\n",
      "[[ 3.]]\n",
      "('#hr#', '#trastuzumab#')\n",
      "[[ 2.]]\n",
      "('#hr#', '#vinorelbine#')\n",
      "[[ 1.]]\n",
      "('#hrg#', '#lapatinib#')\n",
      "[[ 1.]]\n",
      "('#hrg#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 1.]]\n",
      "('#hrg#', '#trastuzumab#')\n",
      "[[ 2.]]\n",
      "('#il36g#', '#binimetinib#')\n",
      "[[ 2.]]\n",
      "('#il36g#', '#buparlisib#')\n",
      "[[ 2.]]\n",
      "('#il36g#', '#dalotuzumab#')\n",
      "[[ 1.]]\n",
      "('#il36g#', '#everolimus#')\n",
      "[[ 2.]]\n",
      "('#il36g#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 2.]]\n",
      "('#il36g#', '#ridaforolimus#')\n",
      "[[ 6.]]\n",
      "('#il36g#', '#temsirolimus#')\n",
      "[[ 3.]]\n",
      "('#il36g#', '#vinorelbine#')\n",
      "[[ 1.]]\n",
      "('#il36g#', '#voxtalisib#')\n",
      "[[ 2.]]\n",
      "('#impact#', '#epirubicin#')\n",
      "[[ 1.]]\n",
      "('#impact#', '#paclitaxel#')\n",
      "[[ 1.]]\n",
      "('#impact#', '#trastuzumab#')\n",
      "[[ 1.]]\n",
      "('#iv#', '#afuresertib#')\n",
      "[[ 1.]]\n",
      "('#iv#', '#akt inhibitor gsk2141795#')\n",
      "[[ 6.]]\n",
      "('#iv#', '#akt inhibitor mk2206#')\n",
      "[[ 12.]]\n",
      "('#iv#', '#bevacizumab#')\n",
      "[[ 7.]]\n",
      "('#iv#', '#binimetinib#')\n",
      "[[ 19.]]\n",
      "('#iv#', '#buparlisib#')\n",
      "[[ 32.]]\n",
      "('#iv#', '#capecitabine#')\n",
      "[[ 7.]]\n",
      "('#iv#', '#carboplatin#')\n",
      "[[ 4.]]\n",
      "('#iv#', '#cisplatin#')\n",
      "[[ 10.]]\n",
      "('#iv#', '#cixutumumab#')\n",
      "[[ 1.]]\n",
      "('#iv#', '#cyclophosphamide#')\n",
      "[[ 4.]]\n",
      "('#iv#', '#dalotuzumab#')\n",
      "[[ 9.]]\n",
      "('#iv#', '#dexamethasone#')\n",
      "[[ 2.]]\n",
      "('#iv#', '#docetaxel#')\n",
      "[[ 9.]]\n",
      "('#iv#', '#doxorubicin#')\n",
      "[[ 3.]]\n",
      "('#iv#', '#epirubicin#')\n",
      "[[ 1.]]\n",
      "('#iv#', '#erlotinib#')\n",
      "[[ 2.]]\n",
      "('#iv#', '#everolimus#')\n",
      "[[ 138.]]\n",
      "('#iv#', '#exemestane#')\n",
      "[[ 10.]]\n",
      "('#iv#', '#gemcitabine#')\n",
      "[[ 3.]]\n",
      "('#iv#', '#lapatinib#')\n",
      "[[ 8.]]\n",
      "('#iv#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 3.]]\n",
      "('#iv#', '#paclitaxel albumin-stabilized nanoparticle formulation#')\n",
      "[[ 5.]]\n",
      "('#iv#', '#paclitaxel#')\n",
      "[[ 21.]]\n",
      "('#iv#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 26.]]\n",
      "('#iv#', '#ribociclib#')\n",
      "[[ 4.]]\n",
      "('#iv#', '#ridaforolimus#')\n",
      "[[ 54.]]\n",
      "('#iv#', '#temsirolimus#')\n",
      "[[ 39.]]\n",
      "('#iv#', '#trametinib#')\n",
      "[[ 2.]]\n",
      "('#iv#', '#trastuzumab#')\n",
      "[[ 25.]]\n",
      "('#iv#', '#triciribine phosphate#')\n",
      "[[ 3.]]\n",
      "('#iv#', '#vinorelbine#')\n",
      "[[ 16.]]\n",
      "('#iv#', '#voxtalisib#')\n",
      "[[ 18.]]\n",
      "('#kit#', '#iproplatin#')\n",
      "[[ 2.]]\n",
      "('#large1#', '#everolimus#')\n",
      "[[ 11.]]\n",
      "('#large1#', '#lapatinib#')\n",
      "[[ 1.]]\n",
      "('#large1#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 1.]]\n",
      "('#lcn2#', '#doxorubicin#')\n",
      "[[ 1.]]\n",
      "('#map2k1#', '#buparlisib#')\n",
      "[[ 1.]]\n",
      "('#map2k1#', '#mek inhibitor ci-1040#')\n",
      "[[ 1.]]\n",
      "('#map2k1#', '#pi3k family kinase inhibitor xl147#')\n",
      "[[ 1.]]\n",
      "('#mapk3#', '#lapatinib#')\n",
      "[[ 1.]]\n",
      "('#mb#', '#mitoxantrone#')\n",
      "[[ 1.]]\n",
      "('#mdm2#', '#epirubicin#')\n",
      "[[ 1.]]\n",
      "('#mdm2#', '#paclitaxel#')\n",
      "[[ 1.]]\n",
      "('#mtor#', '#afuresertib#')\n",
      "[[ 2.]]\n",
      "('#mtor#', '#akt inhibitor gsk2141795#')\n",
      "[[ 6.]]\n",
      "('#mtor#', '#akt inhibitor mk2206#')\n",
      "[[ 9.]]\n",
      "('#mtor#', '#bevacizumab#')\n",
      "[[ 4.]]\n",
      "('#mtor#', '#binimetinib#')\n",
      "[[ 4.]]\n",
      "('#mtor#', '#buparlisib#')\n",
      "[[ 12.]]\n",
      "('#mtor#', '#capecitabine#')\n",
      "[[ 4.]]\n",
      "('#mtor#', '#carbendazim#')\n",
      "[[ 2.]]\n",
      "('#mtor#', '#carboplatin#')\n",
      "[[ 3.]]\n",
      "('#mtor#', '#cisplatin#')\n",
      "[[ 11.]]\n",
      "('#mtor#', '#cixutumumab#')\n",
      "[[ 2.]]\n",
      "('#mtor#', '#clofarabine#')\n",
      "[[ 1.]]\n",
      "('#mtor#', '#cyclophosphamide#')\n",
      "[[ 1.]]\n",
      "('#mtor#', '#dacarbazine#')\n",
      "[[ 2.]]\n",
      "('#mtor#', '#dalotuzumab#')\n",
      "[[ 1.]]\n",
      "('#mtor#', '#docetaxel#')\n",
      "[[ 6.]]\n",
      "('#mtor#', '#doxorubicin#')\n",
      "[[ 6.]]\n",
      "('#mtor#', '#erlotinib#')\n",
      "[[ 10.]]\n",
      "('#mtor#', '#everolimus#')\n",
      "[[ 103.]]\n",
      "('#mtor#', '#exemestane#')\n",
      "[[ 3.]]\n",
      "('#mtor#', '#fluorouracil#')\n",
      "[[ 2.]]\n",
      "('#mtor#', '#lapatinib#')\n",
      "[[ 2.]]\n",
      "('#mtor#', '#lenalidomide#')\n",
      "[[ 4.]]\n",
      "('#mtor#', '#letrozole#')\n",
      "[[ 5.]]\n",
      "('#mtor#', '#leucovorin calcium#')\n",
      "[[ 4.]]\n",
      "('#mtor#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 4.]]\n",
      "('#mtor#', '#oxaliplatin#')\n",
      "[[ 4.]]\n",
      "('#mtor#', '#paclitaxel albumin-stabilized nanoparticle formulation#')\n",
      "[[ 4.]]\n",
      "('#mtor#', '#paclitaxel#')\n",
      "[[ 13.]]\n",
      "('#mtor#', '#pertuzumab#')\n",
      "[[ 2.]]\n",
      "('#mtor#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 16.]]\n",
      "('#mtor#', '#ridaforolimus#')\n",
      "[[ 6.]]\n",
      "('#mtor#', '#selumetinib#')\n",
      "[[ 4.]]\n",
      "('#mtor#', '#sorafenib tosylate#')\n",
      "[[ 4.]]\n",
      "('#mtor#', '#temsirolimus#')\n",
      "[[ 41.]]\n",
      "('#mtor#', '#thalidomide#')\n",
      "[[ 2.]]\n",
      "('#mtor#', '#trametinib#')\n",
      "[[ 4.]]\n",
      "('#mtor#', '#trastuzumab#')\n",
      "[[ 8.]]\n",
      "('#mtor#', '#vinorelbine#')\n",
      "[[ 7.]]\n",
      "('#mtor#', '#voxtalisib#')\n",
      "[[ 2.]]\n",
      "('#nf1#', '#folitixorin#')\n",
      "[[ 1.]]\n",
      "('#nm#', '#resistant starch#')\n",
      "[[ 1.]]\n",
      "('#pdlim5#', '#tamoxifen citrate#')\n",
      "[[ 1.]]\n",
      "('#phospho2#', '#recombinant interferon#')\n",
      "[[ 1.]]\n",
      "('#phospho2#', '#trastuzumab#')\n",
      "[[ 1.]]\n",
      "('#pik3ca#', '#afuresertib#')\n",
      "[[ 3.]]\n",
      "('#pik3ca#', '#akt inhibitor gsk2141795#')\n",
      "[[ 8.]]\n",
      "('#pik3ca#', '#akt inhibitor mk2206#')\n",
      "[[ 13.]]\n",
      "('#pik3ca#', '#alpelisib#')\n",
      "[[ 2.]]\n",
      "('#pik3ca#', '#bevacizumab#')\n",
      "[[ 2.]]\n",
      "('#pik3ca#', '#binimetinib#')\n",
      "[[ 5.]]\n",
      "('#pik3ca#', '#buparlisib#')\n",
      "[[ 27.]]\n",
      "('#pik3ca#', '#capecitabine#')\n",
      "[[ 2.]]\n",
      "('#pik3ca#', '#carbendazim#')\n",
      "[[ 1.]]\n",
      "('#pik3ca#', '#carboplatin#')\n",
      "[[ 4.]]\n",
      "('#pik3ca#', '#cisplatin#')\n",
      "[[ 18.]]\n",
      "('#pik3ca#', '#cixutumumab#')\n",
      "[[ 2.]]\n",
      "('#pik3ca#', '#cyclophosphamide#')\n",
      "[[ 6.]]\n",
      "('#pik3ca#', '#dalotuzumab#')\n",
      "[[ 1.]]\n",
      "('#pik3ca#', '#docetaxel#')\n",
      "[[ 2.]]\n",
      "('#pik3ca#', '#doxorubicin#')\n",
      "[[ 5.]]\n",
      "('#pik3ca#', '#erlotinib#')\n",
      "[[ 4.]]\n",
      "('#pik3ca#', '#everolimus#')\n",
      "[[ 124.]]\n",
      "('#pik3ca#', '#exemestane#')\n",
      "[[ 6.]]\n",
      "('#pik3ca#', '#gemcitabine#')\n",
      "[[ 9.]]\n",
      "('#pik3ca#', '#lapatinib#')\n",
      "[[ 11.]]\n",
      "('#pik3ca#', '#letrozole#')\n",
      "[[ 11.]]\n",
      "('#pik3ca#', '#mek inhibitor ci-1040#')\n",
      "[[ 1.]]\n",
      "('#pik3ca#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 4.]]\n",
      "('#pik3ca#', '#paclitaxel albumin-stabilized nanoparticle formulation#')\n",
      "[[ 1.]]\n",
      "('#pik3ca#', '#paclitaxel#')\n",
      "[[ 25.]]\n",
      "('#pik3ca#', '#pertuzumab#')\n",
      "[[ 1.]]\n",
      "('#pik3ca#', '#pi3k family kinase inhibitor xl147#')\n",
      "[[ 1.]]\n",
      "('#pik3ca#', '#pi3k-beta inhibitor gsk2636771#')\n",
      "[[ 2.]]\n",
      "('#pik3ca#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 15.]]\n",
      "('#pik3ca#', '#recombinant interleukin-6#')\n",
      "[[ 1.]]\n",
      "('#pik3ca#', '#ribociclib#')\n",
      "[[ 1.]]\n",
      "('#pik3ca#', '#ridaforolimus#')\n",
      "[[ 6.]]\n",
      "('#pik3ca#', '#tamoxifen#')\n",
      "[[ 1.]]\n",
      "('#pik3ca#', '#temsirolimus#')\n",
      "[[ 25.]]\n",
      "('#pik3ca#', '#trametinib#')\n",
      "[[ 6.]]\n",
      "('#pik3ca#', '#trans sodium crocetinate#')\n",
      "[[ 2.]]\n",
      "('#pik3ca#', '#trastuzumab#')\n",
      "[[ 38.]]\n",
      "('#pik3ca#', '#triciribine phosphate#')\n",
      "[[ 9.]]\n",
      "('#pik3ca#', '#vinorelbine#')\n",
      "[[ 8.]]\n",
      "('#pik3ca#', '#voxtalisib#')\n",
      "[[ 2.]]\n",
      "('#prdx2#', '#tamoxifen citrate#')\n",
      "[[ 2.]]\n",
      "('#prkaa2#', '#trastuzumab#')\n",
      "[[ 1.]]\n",
      "('#pten#', '#akt inhibitor gsk2141795#')\n",
      "[[ 4.]]\n",
      "('#pten#', '#akt inhibitor mk2206#')\n",
      "[[ 6.]]\n",
      "('#pten#', '#buparlisib#')\n",
      "[[ 4.]]\n",
      "('#pten#', '#carboplatin#')\n",
      "[[ 2.]]\n",
      "('#pten#', '#cisplatin#')\n",
      "[[ 2.]]\n",
      "('#pten#', '#everolimus#')\n",
      "[[ 42.]]\n",
      "('#pten#', '#exemestane#')\n",
      "[[ 2.]]\n",
      "('#pten#', '#fulvestrant#')\n",
      "[[ 1.]]\n",
      "('#pten#', '#lapatinib#')\n",
      "[[ 2.]]\n",
      "('#pten#', '#letrozole#')\n",
      "[[ 7.]]\n",
      "('#pten#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 2.]]\n",
      "('#pten#', '#paclitaxel#')\n",
      "[[ 2.]]\n",
      "('#pten#', '#pertuzumab#')\n",
      "[[ 1.]]\n",
      "('#pten#', '#pi3k-beta inhibitor gsk2636771#')\n",
      "[[ 1.]]\n",
      "('#pten#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 4.]]\n",
      "('#pten#', '#recombinant interleukin-6#')\n",
      "[[ 6.]]\n",
      "('#pten#', '#tamoxifen#')\n",
      "[[ 1.]]\n",
      "('#pten#', '#temsirolimus#')\n",
      "[[ 7.]]\n",
      "('#pten#', '#trans sodium crocetinate#')\n",
      "[[ 4.]]\n",
      "('#pten#', '#trastuzumab#')\n",
      "[[ 11.]]\n",
      "('#pten#', '#vinorelbine#')\n",
      "[[ 2.]]\n",
      "('#raf1#', '#cyclophosphamide#')\n",
      "[[ 2.]]\n",
      "('#raf1#', '#docetaxel#')\n",
      "[[ 1.]]\n",
      "('#raf1#', '#doxorubicin#')\n",
      "[[ 2.]]\n",
      "('#raf1#', '#fluorouracil#')\n",
      "[[ 1.]]\n",
      "('#rnase1#', '#binimetinib#')\n",
      "[[ 2.]]\n",
      "('#rnase1#', '#buparlisib#')\n",
      "[[ 2.]]\n",
      "('#rnase1#', '#dalotuzumab#')\n",
      "[[ 1.]]\n",
      "('#rnase1#', '#everolimus#')\n",
      "[[ 13.]]\n",
      "('#rnase1#', '#lapatinib#')\n",
      "[[ 1.]]\n",
      "('#rnase1#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 1.]]\n",
      "('#rnase1#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 2.]]\n",
      "('#rnase1#', '#ridaforolimus#')\n",
      "[[ 6.]]\n",
      "('#rnase1#', '#temsirolimus#')\n",
      "[[ 3.]]\n",
      "('#rnase1#', '#vinorelbine#')\n",
      "[[ 1.]]\n",
      "('#rnase1#', '#voxtalisib#')\n",
      "[[ 2.]]\n",
      "('#rrad#', '#akt inhibitor gsk2141795#')\n",
      "[[ 2.]]\n",
      "('#rrad#', '#akt inhibitor mk2206#')\n",
      "[[ 3.]]\n",
      "('#rrad#', '#buparlisib#')\n",
      "[[ 2.]]\n",
      "('#rrad#', '#carboplatin#')\n",
      "[[ 1.]]\n",
      "('#rrad#', '#cisplatin#')\n",
      "[[ 1.]]\n",
      "('#rrad#', '#everolimus#')\n",
      "[[ 21.]]\n",
      "('#rrad#', '#exemestane#')\n",
      "[[ 1.]]\n",
      "('#rrad#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 1.]]\n",
      "('#rrad#', '#paclitaxel#')\n",
      "[[ 1.]]\n",
      "('#rrad#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 2.]]\n",
      "('#rrad#', '#temsirolimus#')\n",
      "[[ 3.]]\n",
      "('#rrad#', '#trastuzumab#')\n",
      "[[ 2.]]\n",
      "('#rrad#', '#vinorelbine#')\n",
      "[[ 1.]]\n",
      "('#sgcg#', '#afuresertib#')\n",
      "[[ 1.]]\n",
      "('#sgcg#', '#akt inhibitor gsk2141795#')\n",
      "[[ 2.]]\n",
      "('#sgcg#', '#akt inhibitor mk2206#')\n",
      "[[ 3.]]\n",
      "('#sgcg#', '#binimetinib#')\n",
      "[[ 1.]]\n",
      "('#sgcg#', '#buparlisib#')\n",
      "[[ 8.]]\n",
      "('#sgcg#', '#carboplatin#')\n",
      "[[ 1.]]\n",
      "('#sgcg#', '#cisplatin#')\n",
      "[[ 7.]]\n",
      "('#sgcg#', '#cixutumumab#')\n",
      "[[ 1.]]\n",
      "('#sgcg#', '#cyclophosphamide#')\n",
      "[[ 2.]]\n",
      "('#sgcg#', '#doxorubicin#')\n",
      "[[ 2.]]\n",
      "('#sgcg#', '#erlotinib#')\n",
      "[[ 2.]]\n",
      "('#sgcg#', '#everolimus#')\n",
      "[[ 32.]]\n",
      "('#sgcg#', '#exemestane#')\n",
      "[[ 1.]]\n",
      "('#sgcg#', '#gemcitabine#')\n",
      "[[ 4.]]\n",
      "('#sgcg#', '#mtor kinase inhibitor cc-223#')\n",
      "[[ 1.]]\n",
      "('#sgcg#', '#paclitaxel#')\n",
      "[[ 9.]]\n",
      "('#sgcg#', '#pi3k/mtor inhibitor bez235#')\n",
      "[[ 4.]]\n",
      "('#sgcg#', '#tamoxifen citrate#')\n",
      "[[ 2.]]\n",
      "('#sgcg#', '#tamoxifen#')\n",
      "[[ 1.]]\n",
      "('#sgcg#', '#temsirolimus#')\n",
      "[[ 6.]]\n",
      "('#sgcg#', '#trametinib#')\n",
      "[[ 2.]]\n",
      "('#sgcg#', '#trastuzumab#')\n",
      "[[ 6.]]\n",
      "('#sgcg#', '#triciribine phosphate#')\n",
      "[[ 4.]]\n",
      "('#sgcg#', '#vinorelbine#')\n",
      "[[ 2.]]\n",
      "('#sirt1#', '#iproplatin#')\n",
      "[[ 5.]]\n",
      "('#sirt1#', '#niacinamide#')\n",
      "[[ 3.]]\n",
      "('#sirt1#', '#tamoxifen citrate#')\n",
      "[[ 5.]]\n",
      "('#sirt1#', '#tamoxifen#')\n",
      "[[ 2.]]\n",
      "('#slc7a1#', '#iproplatin#')\n",
      "[[ 2.]]\n",
      "('#sltm#', '#trastuzumab#')\n",
      "[[ 1.]]\n",
      "('#smim10l2a#', '#trastuzumab#')\n",
      "[[ 1.]]\n",
      "('#snai2#', '#trastuzumab#')\n",
      "[[ 8.]]\n",
      "('#spen#', '#tamoxifen#')\n",
      "[[ 1.]]\n",
      "('#src#', '#lapatinib#')\n",
      "[[ 8.]]\n",
      "('#t#', '#akt inhibitor mk2206#')\n",
      "[[ 1.]]\n",
      "('#t#', '#docetaxel#')\n",
      "[[ 3.]]\n",
      "('#t#', '#lapatinib#')\n",
      "[[ 2.]]\n",
      "('#t#', '#paclitaxel#')\n",
      "[[ 3.]]\n",
      "('#t#', '#pertuzumab#')\n",
      "[[ 4.]]\n",
      "('#t#', '#trastuzumab emtansine#')\n",
      "[[ 2.]]\n",
      "('#t#', '#trastuzumab#')\n",
      "[[ 9.]]\n",
      "('#th#', '#tamoxifen#')\n",
      "[[ 1.]]\n",
      "('#tnip1#', '#everolimus#')\n",
      "[[ 1.]]\n",
      "('#tnip1#', '#fulvestrant#')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[ 1.]]\n",
      "('#tnip1#', '#trastuzumab#')\n",
      "[[ 1.]]\n",
      "('#tp53#', '#cyclophosphamide#')\n",
      "[[ 2.]]\n",
      "('#tp53#', '#docetaxel#')\n",
      "[[ 1.]]\n",
      "('#tp53#', '#doxorubicin#')\n",
      "[[ 2.]]\n",
      "('#tp53#', '#epirubicin#')\n",
      "[[ 1.]]\n",
      "('#tp53#', '#fluorouracil#')\n",
      "[[ 1.]]\n",
      "('#tp53#', '#hydroxyurea#')\n",
      "[[ 1.]]\n",
      "('#tp53#', '#iproplatin#')\n",
      "[[ 2.]]\n",
      "('#tp53#', '#paclitaxel#')\n",
      "[[ 1.]]\n"
     ]
    }
   ],
   "source": [
    "\n",
    "for i in a:\n",
    "    pairs = (network_genes[i[0]], network_drugs[i[1]])\n",
    "    r =np.zeros((sensit_network_matrix.shape[0],1))\n",
    "    r[i[0]]=1\n",
    "    c =np.zeros((sensit_network_matrix.shape[1],1))\n",
    "    c[i[1]]=1\n",
    "    counts =(r.T).dot(sensit_network_matrix.dot(c))\n",
    "    print(pairs)\n",
    "    print(counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def extract_gene_drug_pairs(matrix):\n",
    "    a =zip(list(matrix.nonzero()[0]),list(matrix.nonzero()[1])) \n",
    "    pairs = []\n",
    "    counts = []\n",
    "    for pair in a:\n",
    "        gendrug = (network_genes[pair[0]], network_drugs[pair[1]])\n",
    "        r =np.zeros((matrix.shape[0],1))\n",
    "        r[pair[0]]=1\n",
    "        c =np.zeros((matrix.shape[1],1))\n",
    "        c[pair[1]]=1\n",
    "        count =(r.T).dot(matrix.dot(c))\n",
    "        pairs.append(gendrug)\n",
    "        counts.append(count)\n",
    "    return (pairs, counts)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pairs, counts = extract_gene_drug_pairs(sensit_network_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('#abcg2#', '#mitoxantrone#'),\n",
       " ('#acpp#', '#binimetinib#'),\n",
       " ('#acpp#', '#buparlisib#'),\n",
       " ('#acpp#', '#dalotuzumab#'),\n",
       " ('#acpp#', '#everolimus#'),\n",
       " ('#acpp#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#acpp#', '#ridaforolimus#'),\n",
       " ('#acpp#', '#temsirolimus#'),\n",
       " ('#acpp#', '#vinorelbine#'),\n",
       " ('#acpp#', '#voxtalisib#'),\n",
       " ('#ado#', '#trastuzumab emtansine#'),\n",
       " ('#ado#', '#trastuzumab#'),\n",
       " ('#akt1#', '#afuresertib#'),\n",
       " ('#akt1#', '#akt inhibitor gsk2141795#'),\n",
       " ('#akt1#', '#akt inhibitor mk2206#'),\n",
       " ('#akt1#', '#bevacizumab#'),\n",
       " ('#akt1#', '#binimetinib#'),\n",
       " ('#akt1#', '#buparlisib#'),\n",
       " ('#akt1#', '#capecitabine#'),\n",
       " ('#akt1#', '#carboplatin#'),\n",
       " ('#akt1#', '#cisplatin#'),\n",
       " ('#akt1#', '#cyclophosphamide#'),\n",
       " ('#akt1#', '#docetaxel#'),\n",
       " ('#akt1#', '#doxorubicin#'),\n",
       " ('#akt1#', '#everolimus#'),\n",
       " ('#akt1#', '#exemestane#'),\n",
       " ('#akt1#', '#gemcitabine#'),\n",
       " ('#akt1#', '#lapatinib#'),\n",
       " ('#akt1#', '#letrozole#'),\n",
       " ('#akt1#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#akt1#', '#paclitaxel albumin-stabilized nanoparticle formulation#'),\n",
       " ('#akt1#', '#paclitaxel#'),\n",
       " ('#akt1#', '#perifosine#'),\n",
       " ('#akt1#', '#pertuzumab#'),\n",
       " ('#akt1#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#akt1#', '#recombinant interleukin-6#'),\n",
       " ('#akt1#', '#ribociclib#'),\n",
       " ('#akt1#', '#tamoxifen#'),\n",
       " ('#akt1#', '#temsirolimus#'),\n",
       " ('#akt1#', '#trametinib#'),\n",
       " ('#akt1#', '#trans sodium crocetinate#'),\n",
       " ('#akt1#', '#trastuzumab#'),\n",
       " ('#akt1#', '#triciribine phosphate#'),\n",
       " ('#akt1#', '#vinorelbine#'),\n",
       " ('#alb#', '#bevacizumab#'),\n",
       " ('#alb#', '#capecitabine#'),\n",
       " ('#alb#', '#carboplatin#'),\n",
       " ('#alb#', '#cisplatin#'),\n",
       " ('#alb#', '#cyclophosphamide#'),\n",
       " ('#alb#', '#dexamethasone#'),\n",
       " ('#alb#', '#docetaxel#'),\n",
       " ('#alb#', '#epirubicin#'),\n",
       " ('#alb#', '#everolimus#'),\n",
       " ('#alb#', '#exemestane#'),\n",
       " ('#alb#', '#lapatinib#'),\n",
       " ('#alb#', '#paclitaxel albumin-stabilized nanoparticle formulation#'),\n",
       " ('#alb#', '#paclitaxel#'),\n",
       " ('#alb#', '#ribociclib#'),\n",
       " ('#alb#', '#trastuzumab#'),\n",
       " ('#anpep#', '#lapatinib#'),\n",
       " ('#ar#', '#anastrozole#'),\n",
       " ('#ar#', '#exemestane#'),\n",
       " ('#ar#', '#letrozole#'),\n",
       " ('#armc9#', '#akt inhibitor mk2206#'),\n",
       " ('#armc9#', '#bevacizumab#'),\n",
       " ('#armc9#', '#buparlisib#'),\n",
       " ('#armc9#', '#capecitabine#'),\n",
       " ('#armc9#', '#cisplatin#'),\n",
       " ('#armc9#', '#cyclophosphamide#'),\n",
       " ('#armc9#', '#docetaxel#'),\n",
       " ('#armc9#', '#doxorubicin#'),\n",
       " ('#armc9#', '#everolimus#'),\n",
       " ('#armc9#', '#exemestane#'),\n",
       " ('#armc9#', '#gemcitabine#'),\n",
       " ('#armc9#', '#lapatinib#'),\n",
       " ('#armc9#', '#paclitaxel albumin-stabilized nanoparticle formulation#'),\n",
       " ('#armc9#', '#paclitaxel#'),\n",
       " ('#armc9#', '#ribociclib#'),\n",
       " ('#armc9#', '#trastuzumab#'),\n",
       " ('#armc9#', '#triciribine phosphate#'),\n",
       " ('#armc9#', '#vinorelbine#'),\n",
       " ('#birc5#', '#trastuzumab#'),\n",
       " ('#brca1#', '#cisplatin#'),\n",
       " ('#c19orf66#', '#tamoxifen#'),\n",
       " ('#cd24#', '#trastuzumab#'),\n",
       " ('#cdk4#', '#bevacizumab#'),\n",
       " ('#cdk4#', '#capecitabine#'),\n",
       " ('#cdk4#', '#carboplatin#'),\n",
       " ('#cdk4#', '#cisplatin#'),\n",
       " ('#cdk4#', '#cyclophosphamide#'),\n",
       " ('#cdk4#', '#dexamethasone#'),\n",
       " ('#cdk4#', '#docetaxel#'),\n",
       " ('#cdk4#', '#epirubicin#'),\n",
       " ('#cdk4#', '#everolimus#'),\n",
       " ('#cdk4#', '#exemestane#'),\n",
       " ('#cdk4#', '#lapatinib#'),\n",
       " ('#cdk4#', '#paclitaxel albumin-stabilized nanoparticle formulation#'),\n",
       " ('#cdk4#', '#paclitaxel#'),\n",
       " ('#cdk4#', '#ribociclib#'),\n",
       " ('#cdk4#', '#trastuzumab#'),\n",
       " ('#cebpd#', '#folitixorin#'),\n",
       " ('#celp#', '#afuresertib#'),\n",
       " ('#celp#', '#akt inhibitor mk2206#'),\n",
       " ('#celp#', '#binimetinib#'),\n",
       " ('#celp#', '#buparlisib#'),\n",
       " ('#celp#', '#cixutumumab#'),\n",
       " ('#celp#', '#dalotuzumab#'),\n",
       " ('#celp#', '#erlotinib#'),\n",
       " ('#celp#', '#everolimus#'),\n",
       " ('#celp#', '#iproplatin#'),\n",
       " ('#celp#', '#lapatinib#'),\n",
       " ('#celp#', '#letrozole#'),\n",
       " ('#celp#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#celp#', '#niacinamide#'),\n",
       " ('#celp#', '#pi3k-beta inhibitor gsk2636771#'),\n",
       " ('#celp#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#celp#', '#recombinant interleukin-6#'),\n",
       " ('#celp#', '#ridaforolimus#'),\n",
       " ('#celp#', '#tamoxifen citrate#'),\n",
       " ('#celp#', '#tamoxifen#'),\n",
       " ('#celp#', '#temsirolimus#'),\n",
       " ('#celp#', '#trametinib#'),\n",
       " ('#celp#', '#trastuzumab#'),\n",
       " ('#celp#', '#vinorelbine#'),\n",
       " ('#celp#', '#voxtalisib#'),\n",
       " ('#cic#', '#letrozole#'),\n",
       " ('#cic#', '#recombinant interleukin-6#'),\n",
       " ('#cic#', '#trastuzumab#'),\n",
       " ('#cyp19a1#', '#everolimus#'),\n",
       " ('#cyp19a1#', '#fulvestrant#'),\n",
       " ('#cyp19a1#', '#letrozole#'),\n",
       " ('#cyp19a1#', '#pertuzumab#'),\n",
       " ('#cyp19a1#', '#tamoxifen#'),\n",
       " ('#cyp19a1#', '#trastuzumab#'),\n",
       " ('#dcstamp#', '#akt inhibitor mk2206#'),\n",
       " ('#dgcr2#', '#tamoxifen#'),\n",
       " ('#dlx4#', '#trans sodium crocetinate#'),\n",
       " ('#dpm2#', '#recombinant interleukin-6#'),\n",
       " ('#eec1#', '#pi3k-beta inhibitor gsk2636771#'),\n",
       " ('#eec1#', '#temsirolimus#'),\n",
       " ('#egfr#', '#afuresertib#'),\n",
       " ('#egfr#', '#alpelisib#'),\n",
       " ('#egfr#', '#binimetinib#'),\n",
       " ('#egfr#', '#buparlisib#'),\n",
       " ('#egfr#', '#cixutumumab#'),\n",
       " ('#egfr#', '#erlotinib#'),\n",
       " ('#egfr#', '#everolimus#'),\n",
       " ('#egfr#', '#lapatinib#'),\n",
       " ('#egfr#', '#pertuzumab#'),\n",
       " ('#egfr#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#egfr#', '#temsirolimus#'),\n",
       " ('#egfr#', '#trametinib#'),\n",
       " ('#egfr#', '#vinorelbine#'),\n",
       " ('#egr3#', '#bevacizumab#'),\n",
       " ('#egr3#', '#capecitabine#'),\n",
       " ('#egr3#', '#carbendazim#'),\n",
       " ('#egr3#', '#cisplatin#'),\n",
       " ('#egr3#', '#dacarbazine#'),\n",
       " ('#egr3#', '#docetaxel#'),\n",
       " ('#egr3#', '#doxorubicin#'),\n",
       " ('#egr3#', '#erlotinib#'),\n",
       " ('#egr3#', '#everolimus#'),\n",
       " ('#egr3#', '#fluorouracil#'),\n",
       " ('#egr3#', '#lenalidomide#'),\n",
       " ('#egr3#', '#leucovorin calcium#'),\n",
       " ('#egr3#', '#oxaliplatin#'),\n",
       " ('#egr3#', '#paclitaxel albumin-stabilized nanoparticle formulation#'),\n",
       " ('#egr3#', '#paclitaxel#'),\n",
       " ('#egr3#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#egr3#', '#selumetinib#'),\n",
       " ('#egr3#', '#sorafenib tosylate#'),\n",
       " ('#egr3#', '#temsirolimus#'),\n",
       " ('#egr3#', '#thalidomide#'),\n",
       " ('#erbb2#', '#akt inhibitor gsk2141795#'),\n",
       " ('#erbb2#', '#akt inhibitor mk2206#'),\n",
       " ('#erbb2#', '#anastrozole#'),\n",
       " ('#erbb2#', '#bevacizumab#'),\n",
       " ('#erbb2#', '#buparlisib#'),\n",
       " ('#erbb2#', '#capecitabine#'),\n",
       " ('#erbb2#', '#carbendazim#'),\n",
       " ('#erbb2#', '#carboplatin#'),\n",
       " ('#erbb2#', '#cisplatin#'),\n",
       " ('#erbb2#', '#cyclophosphamide#'),\n",
       " ('#erbb2#', '#dacarbazine#'),\n",
       " ('#erbb2#', '#dexamethasone#'),\n",
       " ('#erbb2#', '#docetaxel#'),\n",
       " ('#erbb2#', '#doxorubicin#'),\n",
       " ('#erbb2#', '#epirubicin#'),\n",
       " ('#erbb2#', '#erlotinib#'),\n",
       " ('#erbb2#', '#everolimus#'),\n",
       " ('#erbb2#', '#exemestane#'),\n",
       " ('#erbb2#', '#fluorouracil#'),\n",
       " ('#erbb2#', '#gemcitabine#'),\n",
       " ('#erbb2#', '#lapatinib#'),\n",
       " ('#erbb2#', '#lenalidomide#'),\n",
       " ('#erbb2#', '#letrozole#'),\n",
       " ('#erbb2#', '#leucovorin calcium#'),\n",
       " ('#erbb2#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#erbb2#', '#neratinib#'),\n",
       " ('#erbb2#', '#oxaliplatin#'),\n",
       " ('#erbb2#', '#paclitaxel albumin-stabilized nanoparticle formulation#'),\n",
       " ('#erbb2#', '#paclitaxel#'),\n",
       " ('#erbb2#', '#pertuzumab#'),\n",
       " ('#erbb2#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#erbb2#', '#recombinant interleukin-6#'),\n",
       " ('#erbb2#', '#ribociclib#'),\n",
       " ('#erbb2#', '#selumetinib#'),\n",
       " ('#erbb2#', '#sorafenib tosylate#'),\n",
       " ('#erbb2#', '#tamoxifen#'),\n",
       " ('#erbb2#', '#temsirolimus#'),\n",
       " ('#erbb2#', '#thalidomide#'),\n",
       " ('#erbb2#', '#trastuzumab#'),\n",
       " ('#erbb2#', '#triciribine phosphate#'),\n",
       " ('#erbb2#', '#vinorelbine#'),\n",
       " ('#esr1#', '#iproplatin#'),\n",
       " ('#esr1#', '#niacinamide#'),\n",
       " ('#esr1#', '#tamoxifen citrate#'),\n",
       " ('#esr1#', '#tamoxifen#'),\n",
       " ('#esr2#', '#iproplatin#'),\n",
       " ('#esrra#', '#iproplatin#'),\n",
       " ('#esrrb#', '#iproplatin#'),\n",
       " ('#fancc#', '#cyclophosphamide#'),\n",
       " ('#fancc#', '#docetaxel#'),\n",
       " ('#fancc#', '#doxorubicin#'),\n",
       " ('#fancc#', '#fluorouracil#'),\n",
       " ('#gaa#', '#bevacizumab#'),\n",
       " ('#gaa#', '#capecitabine#'),\n",
       " ('#gaa#', '#carbendazim#'),\n",
       " ('#gaa#', '#cisplatin#'),\n",
       " ('#gaa#', '#dacarbazine#'),\n",
       " ('#gaa#', '#docetaxel#'),\n",
       " ('#gaa#', '#doxorubicin#'),\n",
       " ('#gaa#', '#erlotinib#'),\n",
       " ('#gaa#', '#everolimus#'),\n",
       " ('#gaa#', '#fluorouracil#'),\n",
       " ('#gaa#', '#lenalidomide#'),\n",
       " ('#gaa#', '#leucovorin calcium#'),\n",
       " ('#gaa#', '#oxaliplatin#'),\n",
       " ('#gaa#', '#paclitaxel albumin-stabilized nanoparticle formulation#'),\n",
       " ('#gaa#', '#paclitaxel#'),\n",
       " ('#gaa#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#gaa#', '#selumetinib#'),\n",
       " ('#gaa#', '#sorafenib tosylate#'),\n",
       " ('#gaa#', '#temsirolimus#'),\n",
       " ('#gaa#', '#thalidomide#'),\n",
       " ('#glys1#', '#afuresertib#'),\n",
       " ('#glys1#', '#binimetinib#'),\n",
       " ('#glys1#', '#buparlisib#'),\n",
       " ('#glys1#', '#cixutumumab#'),\n",
       " ('#glys1#', '#dalotuzumab#'),\n",
       " ('#glys1#', '#erlotinib#'),\n",
       " ('#glys1#', '#everolimus#'),\n",
       " ('#glys1#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#glys1#', '#ridaforolimus#'),\n",
       " ('#glys1#', '#temsirolimus#'),\n",
       " ('#glys1#', '#trametinib#'),\n",
       " ('#glys1#', '#vinorelbine#'),\n",
       " ('#glys1#', '#voxtalisib#'),\n",
       " ('#hdac9#', '#tamoxifen citrate#'),\n",
       " ('#hpp1#', '#anastrozole#'),\n",
       " ('#hpp1#', '#bevacizumab#'),\n",
       " ('#hpp1#', '#capecitabine#'),\n",
       " ('#hpp1#', '#carbendazim#'),\n",
       " ('#hpp1#', '#carboplatin#'),\n",
       " ('#hpp1#', '#cisplatin#'),\n",
       " ('#hpp1#', '#cyclophosphamide#'),\n",
       " ('#hpp1#', '#epirubicin#'),\n",
       " ('#hpp1#', '#everolimus#'),\n",
       " ('#hpp1#', '#exemestane#'),\n",
       " ('#hpp1#', '#letrozole#'),\n",
       " ('#hpp1#', '#paclitaxel#'),\n",
       " ('#hpp1#', '#trastuzumab#'),\n",
       " ('#hpp1#', '#vinorelbine#'),\n",
       " ('#hr#', '#akt inhibitor gsk2141795#'),\n",
       " ('#hr#', '#akt inhibitor mk2206#'),\n",
       " ('#hr#', '#buparlisib#'),\n",
       " ('#hr#', '#carboplatin#'),\n",
       " ('#hr#', '#cisplatin#'),\n",
       " ('#hr#', '#everolimus#'),\n",
       " ('#hr#', '#exemestane#'),\n",
       " ('#hr#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#hr#', '#paclitaxel#'),\n",
       " ('#hr#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#hr#', '#temsirolimus#'),\n",
       " ('#hr#', '#trastuzumab#'),\n",
       " ('#hr#', '#vinorelbine#'),\n",
       " ('#hrg#', '#lapatinib#'),\n",
       " ('#hrg#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#hrg#', '#trastuzumab#'),\n",
       " ('#il36g#', '#binimetinib#'),\n",
       " ('#il36g#', '#buparlisib#'),\n",
       " ('#il36g#', '#dalotuzumab#'),\n",
       " ('#il36g#', '#everolimus#'),\n",
       " ('#il36g#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#il36g#', '#ridaforolimus#'),\n",
       " ('#il36g#', '#temsirolimus#'),\n",
       " ('#il36g#', '#vinorelbine#'),\n",
       " ('#il36g#', '#voxtalisib#'),\n",
       " ('#impact#', '#epirubicin#'),\n",
       " ('#impact#', '#paclitaxel#'),\n",
       " ('#impact#', '#trastuzumab#'),\n",
       " ('#iv#', '#afuresertib#'),\n",
       " ('#iv#', '#akt inhibitor gsk2141795#'),\n",
       " ('#iv#', '#akt inhibitor mk2206#'),\n",
       " ('#iv#', '#bevacizumab#'),\n",
       " ('#iv#', '#binimetinib#'),\n",
       " ('#iv#', '#buparlisib#'),\n",
       " ('#iv#', '#capecitabine#'),\n",
       " ('#iv#', '#carboplatin#'),\n",
       " ('#iv#', '#cisplatin#'),\n",
       " ('#iv#', '#cixutumumab#'),\n",
       " ('#iv#', '#cyclophosphamide#'),\n",
       " ('#iv#', '#dalotuzumab#'),\n",
       " ('#iv#', '#dexamethasone#'),\n",
       " ('#iv#', '#docetaxel#'),\n",
       " ('#iv#', '#doxorubicin#'),\n",
       " ('#iv#', '#epirubicin#'),\n",
       " ('#iv#', '#erlotinib#'),\n",
       " ('#iv#', '#everolimus#'),\n",
       " ('#iv#', '#exemestane#'),\n",
       " ('#iv#', '#gemcitabine#'),\n",
       " ('#iv#', '#lapatinib#'),\n",
       " ('#iv#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#iv#', '#paclitaxel albumin-stabilized nanoparticle formulation#'),\n",
       " ('#iv#', '#paclitaxel#'),\n",
       " ('#iv#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#iv#', '#ribociclib#'),\n",
       " ('#iv#', '#ridaforolimus#'),\n",
       " ('#iv#', '#temsirolimus#'),\n",
       " ('#iv#', '#trametinib#'),\n",
       " ('#iv#', '#trastuzumab#'),\n",
       " ('#iv#', '#triciribine phosphate#'),\n",
       " ('#iv#', '#vinorelbine#'),\n",
       " ('#iv#', '#voxtalisib#'),\n",
       " ('#kit#', '#iproplatin#'),\n",
       " ('#large1#', '#everolimus#'),\n",
       " ('#large1#', '#lapatinib#'),\n",
       " ('#large1#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#lcn2#', '#doxorubicin#'),\n",
       " ('#map2k1#', '#buparlisib#'),\n",
       " ('#map2k1#', '#mek inhibitor ci-1040#'),\n",
       " ('#map2k1#', '#pi3k family kinase inhibitor xl147#'),\n",
       " ('#mapk3#', '#lapatinib#'),\n",
       " ('#mb#', '#mitoxantrone#'),\n",
       " ('#mdm2#', '#epirubicin#'),\n",
       " ('#mdm2#', '#paclitaxel#'),\n",
       " ('#mtor#', '#afuresertib#'),\n",
       " ('#mtor#', '#akt inhibitor gsk2141795#'),\n",
       " ('#mtor#', '#akt inhibitor mk2206#'),\n",
       " ('#mtor#', '#bevacizumab#'),\n",
       " ('#mtor#', '#binimetinib#'),\n",
       " ('#mtor#', '#buparlisib#'),\n",
       " ('#mtor#', '#capecitabine#'),\n",
       " ('#mtor#', '#carbendazim#'),\n",
       " ('#mtor#', '#carboplatin#'),\n",
       " ('#mtor#', '#cisplatin#'),\n",
       " ('#mtor#', '#cixutumumab#'),\n",
       " ('#mtor#', '#clofarabine#'),\n",
       " ('#mtor#', '#cyclophosphamide#'),\n",
       " ('#mtor#', '#dacarbazine#'),\n",
       " ('#mtor#', '#dalotuzumab#'),\n",
       " ('#mtor#', '#docetaxel#'),\n",
       " ('#mtor#', '#doxorubicin#'),\n",
       " ('#mtor#', '#erlotinib#'),\n",
       " ('#mtor#', '#everolimus#'),\n",
       " ('#mtor#', '#exemestane#'),\n",
       " ('#mtor#', '#fluorouracil#'),\n",
       " ('#mtor#', '#lapatinib#'),\n",
       " ('#mtor#', '#lenalidomide#'),\n",
       " ('#mtor#', '#letrozole#'),\n",
       " ('#mtor#', '#leucovorin calcium#'),\n",
       " ('#mtor#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#mtor#', '#oxaliplatin#'),\n",
       " ('#mtor#', '#paclitaxel albumin-stabilized nanoparticle formulation#'),\n",
       " ('#mtor#', '#paclitaxel#'),\n",
       " ('#mtor#', '#pertuzumab#'),\n",
       " ('#mtor#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#mtor#', '#ridaforolimus#'),\n",
       " ('#mtor#', '#selumetinib#'),\n",
       " ('#mtor#', '#sorafenib tosylate#'),\n",
       " ('#mtor#', '#temsirolimus#'),\n",
       " ('#mtor#', '#thalidomide#'),\n",
       " ('#mtor#', '#trametinib#'),\n",
       " ('#mtor#', '#trastuzumab#'),\n",
       " ('#mtor#', '#vinorelbine#'),\n",
       " ('#mtor#', '#voxtalisib#'),\n",
       " ('#nf1#', '#folitixorin#'),\n",
       " ('#nm#', '#resistant starch#'),\n",
       " ('#pdlim5#', '#tamoxifen citrate#'),\n",
       " ('#phospho2#', '#recombinant interferon#'),\n",
       " ('#phospho2#', '#trastuzumab#'),\n",
       " ('#pik3ca#', '#afuresertib#'),\n",
       " ('#pik3ca#', '#akt inhibitor gsk2141795#'),\n",
       " ('#pik3ca#', '#akt inhibitor mk2206#'),\n",
       " ('#pik3ca#', '#alpelisib#'),\n",
       " ('#pik3ca#', '#bevacizumab#'),\n",
       " ('#pik3ca#', '#binimetinib#'),\n",
       " ('#pik3ca#', '#buparlisib#'),\n",
       " ('#pik3ca#', '#capecitabine#'),\n",
       " ('#pik3ca#', '#carbendazim#'),\n",
       " ('#pik3ca#', '#carboplatin#'),\n",
       " ('#pik3ca#', '#cisplatin#'),\n",
       " ('#pik3ca#', '#cixutumumab#'),\n",
       " ('#pik3ca#', '#cyclophosphamide#'),\n",
       " ('#pik3ca#', '#dalotuzumab#'),\n",
       " ('#pik3ca#', '#docetaxel#'),\n",
       " ('#pik3ca#', '#doxorubicin#'),\n",
       " ('#pik3ca#', '#erlotinib#'),\n",
       " ('#pik3ca#', '#everolimus#'),\n",
       " ('#pik3ca#', '#exemestane#'),\n",
       " ('#pik3ca#', '#gemcitabine#'),\n",
       " ('#pik3ca#', '#lapatinib#'),\n",
       " ('#pik3ca#', '#letrozole#'),\n",
       " ('#pik3ca#', '#mek inhibitor ci-1040#'),\n",
       " ('#pik3ca#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#pik3ca#', '#paclitaxel albumin-stabilized nanoparticle formulation#'),\n",
       " ('#pik3ca#', '#paclitaxel#'),\n",
       " ('#pik3ca#', '#pertuzumab#'),\n",
       " ('#pik3ca#', '#pi3k family kinase inhibitor xl147#'),\n",
       " ('#pik3ca#', '#pi3k-beta inhibitor gsk2636771#'),\n",
       " ('#pik3ca#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#pik3ca#', '#recombinant interleukin-6#'),\n",
       " ('#pik3ca#', '#ribociclib#'),\n",
       " ('#pik3ca#', '#ridaforolimus#'),\n",
       " ('#pik3ca#', '#tamoxifen#'),\n",
       " ('#pik3ca#', '#temsirolimus#'),\n",
       " ('#pik3ca#', '#trametinib#'),\n",
       " ('#pik3ca#', '#trans sodium crocetinate#'),\n",
       " ('#pik3ca#', '#trastuzumab#'),\n",
       " ('#pik3ca#', '#triciribine phosphate#'),\n",
       " ('#pik3ca#', '#vinorelbine#'),\n",
       " ('#pik3ca#', '#voxtalisib#'),\n",
       " ('#prdx2#', '#tamoxifen citrate#'),\n",
       " ('#prkaa2#', '#trastuzumab#'),\n",
       " ('#pten#', '#akt inhibitor gsk2141795#'),\n",
       " ('#pten#', '#akt inhibitor mk2206#'),\n",
       " ('#pten#', '#buparlisib#'),\n",
       " ('#pten#', '#carboplatin#'),\n",
       " ('#pten#', '#cisplatin#'),\n",
       " ('#pten#', '#everolimus#'),\n",
       " ('#pten#', '#exemestane#'),\n",
       " ('#pten#', '#fulvestrant#'),\n",
       " ('#pten#', '#lapatinib#'),\n",
       " ('#pten#', '#letrozole#'),\n",
       " ('#pten#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#pten#', '#paclitaxel#'),\n",
       " ('#pten#', '#pertuzumab#'),\n",
       " ('#pten#', '#pi3k-beta inhibitor gsk2636771#'),\n",
       " ('#pten#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#pten#', '#recombinant interleukin-6#'),\n",
       " ('#pten#', '#tamoxifen#'),\n",
       " ('#pten#', '#temsirolimus#'),\n",
       " ('#pten#', '#trans sodium crocetinate#'),\n",
       " ('#pten#', '#trastuzumab#'),\n",
       " ('#pten#', '#vinorelbine#'),\n",
       " ('#raf1#', '#cyclophosphamide#'),\n",
       " ('#raf1#', '#docetaxel#'),\n",
       " ('#raf1#', '#doxorubicin#'),\n",
       " ('#raf1#', '#fluorouracil#'),\n",
       " ('#rnase1#', '#binimetinib#'),\n",
       " ('#rnase1#', '#buparlisib#'),\n",
       " ('#rnase1#', '#dalotuzumab#'),\n",
       " ('#rnase1#', '#everolimus#'),\n",
       " ('#rnase1#', '#lapatinib#'),\n",
       " ('#rnase1#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#rnase1#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#rnase1#', '#ridaforolimus#'),\n",
       " ('#rnase1#', '#temsirolimus#'),\n",
       " ('#rnase1#', '#vinorelbine#'),\n",
       " ('#rnase1#', '#voxtalisib#'),\n",
       " ('#rrad#', '#akt inhibitor gsk2141795#'),\n",
       " ('#rrad#', '#akt inhibitor mk2206#'),\n",
       " ('#rrad#', '#buparlisib#'),\n",
       " ('#rrad#', '#carboplatin#'),\n",
       " ('#rrad#', '#cisplatin#'),\n",
       " ('#rrad#', '#everolimus#'),\n",
       " ('#rrad#', '#exemestane#'),\n",
       " ('#rrad#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#rrad#', '#paclitaxel#'),\n",
       " ('#rrad#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#rrad#', '#temsirolimus#'),\n",
       " ('#rrad#', '#trastuzumab#'),\n",
       " ('#rrad#', '#vinorelbine#'),\n",
       " ('#sgcg#', '#afuresertib#'),\n",
       " ('#sgcg#', '#akt inhibitor gsk2141795#'),\n",
       " ('#sgcg#', '#akt inhibitor mk2206#'),\n",
       " ('#sgcg#', '#binimetinib#'),\n",
       " ('#sgcg#', '#buparlisib#'),\n",
       " ('#sgcg#', '#carboplatin#'),\n",
       " ('#sgcg#', '#cisplatin#'),\n",
       " ('#sgcg#', '#cixutumumab#'),\n",
       " ('#sgcg#', '#cyclophosphamide#'),\n",
       " ('#sgcg#', '#doxorubicin#'),\n",
       " ('#sgcg#', '#erlotinib#'),\n",
       " ('#sgcg#', '#everolimus#'),\n",
       " ('#sgcg#', '#exemestane#'),\n",
       " ('#sgcg#', '#gemcitabine#'),\n",
       " ('#sgcg#', '#mtor kinase inhibitor cc-223#'),\n",
       " ('#sgcg#', '#paclitaxel#'),\n",
       " ('#sgcg#', '#pi3k/mtor inhibitor bez235#'),\n",
       " ('#sgcg#', '#tamoxifen citrate#'),\n",
       " ('#sgcg#', '#tamoxifen#'),\n",
       " ('#sgcg#', '#temsirolimus#'),\n",
       " ('#sgcg#', '#trametinib#'),\n",
       " ('#sgcg#', '#trastuzumab#'),\n",
       " ('#sgcg#', '#triciribine phosphate#'),\n",
       " ('#sgcg#', '#vinorelbine#'),\n",
       " ('#sirt1#', '#iproplatin#'),\n",
       " ('#sirt1#', '#niacinamide#'),\n",
       " ('#sirt1#', '#tamoxifen citrate#'),\n",
       " ('#sirt1#', '#tamoxifen#'),\n",
       " ('#slc7a1#', '#iproplatin#'),\n",
       " ('#sltm#', '#trastuzumab#'),\n",
       " ('#smim10l2a#', '#trastuzumab#'),\n",
       " ('#snai2#', '#trastuzumab#'),\n",
       " ('#spen#', '#tamoxifen#'),\n",
       " ('#src#', '#lapatinib#'),\n",
       " ('#t#', '#akt inhibitor mk2206#'),\n",
       " ('#t#', '#docetaxel#'),\n",
       " ('#t#', '#lapatinib#'),\n",
       " ('#t#', '#paclitaxel#'),\n",
       " ('#t#', '#pertuzumab#'),\n",
       " ('#t#', '#trastuzumab emtansine#'),\n",
       " ('#t#', '#trastuzumab#'),\n",
       " ('#th#', '#tamoxifen#'),\n",
       " ('#tnip1#', '#everolimus#'),\n",
       " ('#tnip1#', '#fulvestrant#'),\n",
       " ('#tnip1#', '#trastuzumab#'),\n",
       " ('#tp53#', '#cyclophosphamide#'),\n",
       " ('#tp53#', '#docetaxel#'),\n",
       " ('#tp53#', '#doxorubicin#'),\n",
       " ('#tp53#', '#epirubicin#'),\n",
       " ('#tp53#', '#fluorouracil#'),\n",
       " ('#tp53#', '#hydroxyurea#'),\n",
       " ('#tp53#', '#iproplatin#'),\n",
       " ('#tp53#', '#paclitaxel#')]"
      ]
     },
     "execution_count": 192,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pairs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import networkx as nx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def back_to_original_indeces(pairs):\n",
    "    original_index = []\n",
    "    for item in pairs:\n",
    "        original = (col_idx[item[0]], col_idx[item[1]])\n",
    "        original_index.append(original)\n",
    "    return pairs ,original_index\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def back_to_original_index(pair):\n",
    "    original = (col_idx[pair[0]],col_idx[pair[1]])\n",
    "    return pair,original"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(('#abcg2#', '#mitoxantrone#'), (0, 221))"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "back_to_original_index(('#abcg2#', '#mitoxantrone#'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def back_to_original_indeces(pairs):\n",
    "    original_indeces = {}\n",
    "    for item in pairs:\n",
    "        original = back_to_original_index(item)\n",
    "        original_indeces[original[0]]=original[1]\n",
    "    return original_indeces"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "original_indeces = back_to_original_indeces(pairs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{('#abcg2#', '#mitoxantrone#'): (0, 221),\n",
       " ('#acpp#', '#binimetinib#'): (3, 35),\n",
       " ('#acpp#', '#buparlisib#'): (3, 47),\n",
       " ('#acpp#', '#dalotuzumab#'): (3, 92),\n",
       " ('#acpp#', '#everolimus#'): (3, 127),\n",
       " ('#acpp#', '#pi3k/mtor inhibitor bez235#'): (3, 276),\n",
       " ('#acpp#', '#ridaforolimus#'): (3, 316),\n",
       " ('#acpp#', '#temsirolimus#'): (3, 363),\n",
       " ('#acpp#', '#vinorelbine#'): (3, 395),\n",
       " ('#acpp#', '#voxtalisib#'): (3, 396),\n",
       " ('#ado#', '#trastuzumab emtansine#'): (4, 385),\n",
       " ('#ado#', '#trastuzumab#'): (4, 386),\n",
       " ('#akt1#', '#afuresertib#'): (11, 5),\n",
       " ('#akt1#', '#akt inhibitor gsk2141795#'): (11, 9),\n",
       " ('#akt1#', '#akt inhibitor mk2206#'): (11, 10),\n",
       " ('#akt1#', '#bevacizumab#'): (11, 33),\n",
       " ('#akt1#', '#binimetinib#'): (11, 35),\n",
       " ('#akt1#', '#buparlisib#'): (11, 47),\n",
       " ('#akt1#', '#capecitabine#'): (11, 51),\n",
       " ('#akt1#', '#carboplatin#'): (11, 53),\n",
       " ('#akt1#', '#cisplatin#'): (11, 74),\n",
       " ('#akt1#', '#cyclophosphamide#'): (11, 85),\n",
       " ('#akt1#', '#docetaxel#'): (11, 100),\n",
       " ('#akt1#', '#doxorubicin#'): (11, 102),\n",
       " ('#akt1#', '#everolimus#'): (11, 127),\n",
       " ('#akt1#', '#exemestane#'): (11, 128),\n",
       " ('#akt1#', '#gemcitabine#'): (11, 152),\n",
       " ('#akt1#', '#lapatinib#'): (11, 192),\n",
       " ('#akt1#', '#letrozole#'): (11, 197),\n",
       " ('#akt1#', '#mtor kinase inhibitor cc-223#'): (11, 230),\n",
       " ('#akt1#', '#paclitaxel albumin-stabilized nanoparticle formulation#'): (11,\n",
       "  252),\n",
       " ('#akt1#', '#paclitaxel#'): (11, 253),\n",
       " ('#akt1#', '#perifosine#'): (11, 267),\n",
       " ('#akt1#', '#pertuzumab#'): (11, 268),\n",
       " ('#akt1#', '#pi3k/mtor inhibitor bez235#'): (11, 276),\n",
       " ('#akt1#', '#recombinant interleukin-6#'): (11, 310),\n",
       " ('#akt1#', '#ribociclib#'): (11, 315),\n",
       " ('#akt1#', '#tamoxifen#'): (11, 359),\n",
       " ('#akt1#', '#temsirolimus#'): (11, 363),\n",
       " ('#akt1#', '#trametinib#'): (11, 383),\n",
       " ('#akt1#', '#trans sodium crocetinate#'): (11, 384),\n",
       " ('#akt1#', '#trastuzumab#'): (11, 386),\n",
       " ('#akt1#', '#triciribine phosphate#'): (11, 387),\n",
       " ('#akt1#', '#vinorelbine#'): (11, 395),\n",
       " ('#alb#', '#bevacizumab#'): (12, 33),\n",
       " ('#alb#', '#capecitabine#'): (12, 51),\n",
       " ('#alb#', '#carboplatin#'): (12, 53),\n",
       " ('#alb#', '#cisplatin#'): (12, 74),\n",
       " ('#alb#', '#cyclophosphamide#'): (12, 85),\n",
       " ('#alb#', '#dexamethasone#'): (12, 96),\n",
       " ('#alb#', '#docetaxel#'): (12, 100),\n",
       " ('#alb#', '#epirubicin#'): (12, 118),\n",
       " ('#alb#', '#everolimus#'): (12, 127),\n",
       " ('#alb#', '#exemestane#'): (12, 128),\n",
       " ('#alb#', '#lapatinib#'): (12, 192),\n",
       " ('#alb#', '#paclitaxel albumin-stabilized nanoparticle formulation#'): (12,\n",
       "  252),\n",
       " ('#alb#', '#paclitaxel#'): (12, 253),\n",
       " ('#alb#', '#ribociclib#'): (12, 315),\n",
       " ('#alb#', '#trastuzumab#'): (12, 386),\n",
       " ('#anpep#', '#lapatinib#'): (19, 192),\n",
       " ('#ar#', '#anastrozole#'): (20, 18),\n",
       " ('#ar#', '#exemestane#'): (20, 128),\n",
       " ('#ar#', '#letrozole#'): (20, 197),\n",
       " ('#armc9#', '#akt inhibitor mk2206#'): (24, 10),\n",
       " ('#armc9#', '#bevacizumab#'): (24, 33),\n",
       " ('#armc9#', '#buparlisib#'): (24, 47),\n",
       " ('#armc9#', '#capecitabine#'): (24, 51),\n",
       " ('#armc9#', '#cisplatin#'): (24, 74),\n",
       " ('#armc9#', '#cyclophosphamide#'): (24, 85),\n",
       " ('#armc9#', '#docetaxel#'): (24, 100),\n",
       " ('#armc9#', '#doxorubicin#'): (24, 102),\n",
       " ('#armc9#', '#everolimus#'): (24, 127),\n",
       " ('#armc9#', '#exemestane#'): (24, 128),\n",
       " ('#armc9#', '#gemcitabine#'): (24, 152),\n",
       " ('#armc9#', '#lapatinib#'): (24, 192),\n",
       " ('#armc9#', '#paclitaxel albumin-stabilized nanoparticle formulation#'): (24,\n",
       "  252),\n",
       " ('#armc9#', '#paclitaxel#'): (24, 253),\n",
       " ('#armc9#', '#ribociclib#'): (24, 315),\n",
       " ('#armc9#', '#trastuzumab#'): (24, 386),\n",
       " ('#armc9#', '#triciribine phosphate#'): (24, 387),\n",
       " ('#armc9#', '#vinorelbine#'): (24, 395),\n",
       " ('#birc5#', '#trastuzumab#'): (36, 386),\n",
       " ('#brca1#', '#cisplatin#'): (42, 74),\n",
       " ('#c19orf66#', '#tamoxifen#'): (48, 359),\n",
       " ('#cd24#', '#trastuzumab#'): (58, 386),\n",
       " ('#cdk4#', '#bevacizumab#'): (63, 33),\n",
       " ('#cdk4#', '#capecitabine#'): (63, 51),\n",
       " ('#cdk4#', '#carboplatin#'): (63, 53),\n",
       " ('#cdk4#', '#cisplatin#'): (63, 74),\n",
       " ('#cdk4#', '#cyclophosphamide#'): (63, 85),\n",
       " ('#cdk4#', '#dexamethasone#'): (63, 96),\n",
       " ('#cdk4#', '#docetaxel#'): (63, 100),\n",
       " ('#cdk4#', '#epirubicin#'): (63, 118),\n",
       " ('#cdk4#', '#everolimus#'): (63, 127),\n",
       " ('#cdk4#', '#exemestane#'): (63, 128),\n",
       " ('#cdk4#', '#lapatinib#'): (63, 192),\n",
       " ('#cdk4#', '#paclitaxel albumin-stabilized nanoparticle formulation#'): (63,\n",
       "  252),\n",
       " ('#cdk4#', '#paclitaxel#'): (63, 253),\n",
       " ('#cdk4#', '#ribociclib#'): (63, 315),\n",
       " ('#cdk4#', '#trastuzumab#'): (63, 386),\n",
       " ('#cebpd#', '#folitixorin#'): (67, 142),\n",
       " ('#celp#', '#afuresertib#'): (68, 5),\n",
       " ('#celp#', '#akt inhibitor mk2206#'): (68, 10),\n",
       " ('#celp#', '#binimetinib#'): (68, 35),\n",
       " ('#celp#', '#buparlisib#'): (68, 47),\n",
       " ('#celp#', '#cixutumumab#'): (68, 75),\n",
       " ('#celp#', '#dalotuzumab#'): (68, 92),\n",
       " ('#celp#', '#erlotinib#'): (68, 122),\n",
       " ('#celp#', '#everolimus#'): (68, 127),\n",
       " ('#celp#', '#iproplatin#'): (68, 177),\n",
       " ('#celp#', '#lapatinib#'): (68, 192),\n",
       " ('#celp#', '#letrozole#'): (68, 197),\n",
       " ('#celp#', '#mtor kinase inhibitor cc-223#'): (68, 230),\n",
       " ('#celp#', '#niacinamide#'): (68, 240),\n",
       " ('#celp#', '#pi3k-beta inhibitor gsk2636771#'): (68, 275),\n",
       " ('#celp#', '#pi3k/mtor inhibitor bez235#'): (68, 276),\n",
       " ('#celp#', '#recombinant interleukin-6#'): (68, 310),\n",
       " ('#celp#', '#ridaforolimus#'): (68, 316),\n",
       " ('#celp#', '#tamoxifen citrate#'): (68, 358),\n",
       " ('#celp#', '#tamoxifen#'): (68, 359),\n",
       " ('#celp#', '#temsirolimus#'): (68, 363),\n",
       " ('#celp#', '#trametinib#'): (68, 383),\n",
       " ('#celp#', '#trastuzumab#'): (68, 386),\n",
       " ('#celp#', '#vinorelbine#'): (68, 395),\n",
       " ('#celp#', '#voxtalisib#'): (68, 396),\n",
       " ('#cic#', '#letrozole#'): (72, 197),\n",
       " ('#cic#', '#recombinant interleukin-6#'): (72, 310),\n",
       " ('#cic#', '#trastuzumab#'): (72, 386),\n",
       " ('#cyp19a1#', '#everolimus#'): (87, 127),\n",
       " ('#cyp19a1#', '#fulvestrant#'): (87, 149),\n",
       " ('#cyp19a1#', '#letrozole#'): (87, 197),\n",
       " ('#cyp19a1#', '#pertuzumab#'): (87, 268),\n",
       " ('#cyp19a1#', '#tamoxifen#'): (87, 359),\n",
       " ('#cyp19a1#', '#trastuzumab#'): (87, 386),\n",
       " ('#dcstamp#', '#akt inhibitor mk2206#'): (94, 10),\n",
       " ('#dgcr2#', '#tamoxifen#'): (97, 359),\n",
       " ('#dlx4#', '#trans sodium crocetinate#'): (98, 384),\n",
       " ('#dpm2#', '#recombinant interleukin-6#'): (104, 310),\n",
       " ('#eec1#', '#pi3k-beta inhibitor gsk2636771#'): (109, 275),\n",
       " ('#eec1#', '#temsirolimus#'): (109, 363),\n",
       " ('#egfr#', '#afuresertib#'): (113, 5),\n",
       " ('#egfr#', '#alpelisib#'): (113, 14),\n",
       " ('#egfr#', '#binimetinib#'): (113, 35),\n",
       " ('#egfr#', '#buparlisib#'): (113, 47),\n",
       " ('#egfr#', '#cixutumumab#'): (113, 75),\n",
       " ('#egfr#', '#erlotinib#'): (113, 122),\n",
       " ('#egfr#', '#everolimus#'): (113, 127),\n",
       " ('#egfr#', '#lapatinib#'): (113, 192),\n",
       " ('#egfr#', '#pertuzumab#'): (113, 268),\n",
       " ('#egfr#', '#pi3k/mtor inhibitor bez235#'): (113, 276),\n",
       " ('#egfr#', '#temsirolimus#'): (113, 363),\n",
       " ('#egfr#', '#trametinib#'): (113, 383),\n",
       " ('#egfr#', '#vinorelbine#'): (113, 395),\n",
       " ('#egr3#', '#bevacizumab#'): (114, 33),\n",
       " ('#egr3#', '#capecitabine#'): (114, 51),\n",
       " ('#egr3#', '#carbendazim#'): (114, 52),\n",
       " ('#egr3#', '#cisplatin#'): (114, 74),\n",
       " ('#egr3#', '#dacarbazine#'): (114, 90),\n",
       " ('#egr3#', '#docetaxel#'): (114, 100),\n",
       " ('#egr3#', '#doxorubicin#'): (114, 102),\n",
       " ('#egr3#', '#erlotinib#'): (114, 122),\n",
       " ('#egr3#', '#everolimus#'): (114, 127),\n",
       " ('#egr3#', '#fluorouracil#'): (114, 141),\n",
       " ('#egr3#', '#lenalidomide#'): (114, 196),\n",
       " ('#egr3#', '#leucovorin calcium#'): (114, 198),\n",
       " ('#egr3#', '#oxaliplatin#'): (114, 251),\n",
       " ('#egr3#', '#paclitaxel albumin-stabilized nanoparticle formulation#'): (114,\n",
       "  252),\n",
       " ('#egr3#', '#paclitaxel#'): (114, 253),\n",
       " ('#egr3#', '#pi3k/mtor inhibitor bez235#'): (114, 276),\n",
       " ('#egr3#', '#selumetinib#'): (114, 331),\n",
       " ('#egr3#', '#sorafenib tosylate#'): (114, 347),\n",
       " ('#egr3#', '#temsirolimus#'): (114, 363),\n",
       " ('#egr3#', '#thalidomide#'): (114, 371),\n",
       " ('#erbb2#', '#akt inhibitor gsk2141795#'): (119, 9),\n",
       " ('#erbb2#', '#akt inhibitor mk2206#'): (119, 10),\n",
       " ('#erbb2#', '#anastrozole#'): (119, 18),\n",
       " ('#erbb2#', '#bevacizumab#'): (119, 33),\n",
       " ('#erbb2#', '#buparlisib#'): (119, 47),\n",
       " ('#erbb2#', '#capecitabine#'): (119, 51),\n",
       " ('#erbb2#', '#carbendazim#'): (119, 52),\n",
       " ('#erbb2#', '#carboplatin#'): (119, 53),\n",
       " ('#erbb2#', '#cisplatin#'): (119, 74),\n",
       " ('#erbb2#', '#cyclophosphamide#'): (119, 85),\n",
       " ('#erbb2#', '#dacarbazine#'): (119, 90),\n",
       " ('#erbb2#', '#dexamethasone#'): (119, 96),\n",
       " ('#erbb2#', '#docetaxel#'): (119, 100),\n",
       " ('#erbb2#', '#doxorubicin#'): (119, 102),\n",
       " ('#erbb2#', '#epirubicin#'): (119, 118),\n",
       " ('#erbb2#', '#erlotinib#'): (119, 122),\n",
       " ('#erbb2#', '#everolimus#'): (119, 127),\n",
       " ('#erbb2#', '#exemestane#'): (119, 128),\n",
       " ('#erbb2#', '#fluorouracil#'): (119, 141),\n",
       " ('#erbb2#', '#gemcitabine#'): (119, 152),\n",
       " ('#erbb2#', '#lapatinib#'): (119, 192),\n",
       " ('#erbb2#', '#lenalidomide#'): (119, 196),\n",
       " ('#erbb2#', '#letrozole#'): (119, 197),\n",
       " ('#erbb2#', '#leucovorin calcium#'): (119, 198),\n",
       " ('#erbb2#', '#mtor kinase inhibitor cc-223#'): (119, 230),\n",
       " ('#erbb2#', '#neratinib#'): (119, 237),\n",
       " ('#erbb2#', '#oxaliplatin#'): (119, 251),\n",
       " ('#erbb2#', '#paclitaxel albumin-stabilized nanoparticle formulation#'): (119,\n",
       "  252),\n",
       " ('#erbb2#', '#paclitaxel#'): (119, 253),\n",
       " ('#erbb2#', '#pertuzumab#'): (119, 268),\n",
       " ('#erbb2#', '#pi3k/mtor inhibitor bez235#'): (119, 276),\n",
       " ('#erbb2#', '#recombinant interleukin-6#'): (119, 310),\n",
       " ('#erbb2#', '#ribociclib#'): (119, 315),\n",
       " ('#erbb2#', '#selumetinib#'): (119, 331),\n",
       " ('#erbb2#', '#sorafenib tosylate#'): (119, 347),\n",
       " ('#erbb2#', '#tamoxifen#'): (119, 359),\n",
       " ('#erbb2#', '#temsirolimus#'): (119, 363),\n",
       " ('#erbb2#', '#thalidomide#'): (119, 371),\n",
       " ('#erbb2#', '#trastuzumab#'): (119, 386),\n",
       " ('#erbb2#', '#triciribine phosphate#'): (119, 387),\n",
       " ('#erbb2#', '#vinorelbine#'): (119, 395),\n",
       " ('#esr1#', '#iproplatin#'): (123, 177),\n",
       " ('#esr1#', '#niacinamide#'): (123, 240),\n",
       " ('#esr1#', '#tamoxifen citrate#'): (123, 358),\n",
       " ('#esr1#', '#tamoxifen#'): (123, 359),\n",
       " ('#esr2#', '#iproplatin#'): (124, 177),\n",
       " ('#esrra#', '#iproplatin#'): (125, 177),\n",
       " ('#esrrb#', '#iproplatin#'): (126, 177),\n",
       " ('#fancc#', '#cyclophosphamide#'): (135, 85),\n",
       " ('#fancc#', '#docetaxel#'): (135, 100),\n",
       " ('#fancc#', '#doxorubicin#'): (135, 102),\n",
       " ('#fancc#', '#fluorouracil#'): (135, 141),\n",
       " ('#gaa#', '#bevacizumab#'): (150, 33),\n",
       " ('#gaa#', '#capecitabine#'): (150, 51),\n",
       " ('#gaa#', '#carbendazim#'): (150, 52),\n",
       " ('#gaa#', '#cisplatin#'): (150, 74),\n",
       " ('#gaa#', '#dacarbazine#'): (150, 90),\n",
       " ('#gaa#', '#docetaxel#'): (150, 100),\n",
       " ('#gaa#', '#doxorubicin#'): (150, 102),\n",
       " ('#gaa#', '#erlotinib#'): (150, 122),\n",
       " ('#gaa#', '#everolimus#'): (150, 127),\n",
       " ('#gaa#', '#fluorouracil#'): (150, 141),\n",
       " ('#gaa#', '#lenalidomide#'): (150, 196),\n",
       " ('#gaa#', '#leucovorin calcium#'): (150, 198),\n",
       " ('#gaa#', '#oxaliplatin#'): (150, 251),\n",
       " ('#gaa#', '#paclitaxel albumin-stabilized nanoparticle formulation#'): (150,\n",
       "  252),\n",
       " ('#gaa#', '#paclitaxel#'): (150, 253),\n",
       " ('#gaa#', '#pi3k/mtor inhibitor bez235#'): (150, 276),\n",
       " ('#gaa#', '#selumetinib#'): (150, 331),\n",
       " ('#gaa#', '#sorafenib tosylate#'): (150, 347),\n",
       " ('#gaa#', '#temsirolimus#'): (150, 363),\n",
       " ('#gaa#', '#thalidomide#'): (150, 371),\n",
       " ('#glys1#', '#afuresertib#'): (154, 5),\n",
       " ('#glys1#', '#binimetinib#'): (154, 35),\n",
       " ('#glys1#', '#buparlisib#'): (154, 47),\n",
       " ('#glys1#', '#cixutumumab#'): (154, 75),\n",
       " ('#glys1#', '#dalotuzumab#'): (154, 92),\n",
       " ('#glys1#', '#erlotinib#'): (154, 122),\n",
       " ('#glys1#', '#everolimus#'): (154, 127),\n",
       " ('#glys1#', '#pi3k/mtor inhibitor bez235#'): (154, 276),\n",
       " ('#glys1#', '#ridaforolimus#'): (154, 316),\n",
       " ('#glys1#', '#temsirolimus#'): (154, 363),\n",
       " ('#glys1#', '#trametinib#'): (154, 383),\n",
       " ('#glys1#', '#vinorelbine#'): (154, 395),\n",
       " ('#glys1#', '#voxtalisib#'): (154, 396),\n",
       " ('#hdac9#', '#tamoxifen citrate#'): (161, 358),\n",
       " ('#hpp1#', '#anastrozole#'): (164, 18),\n",
       " ('#hpp1#', '#bevacizumab#'): (164, 33),\n",
       " ('#hpp1#', '#capecitabine#'): (164, 51),\n",
       " ('#hpp1#', '#carbendazim#'): (164, 52),\n",
       " ('#hpp1#', '#carboplatin#'): (164, 53),\n",
       " ('#hpp1#', '#cisplatin#'): (164, 74),\n",
       " ('#hpp1#', '#cyclophosphamide#'): (164, 85),\n",
       " ('#hpp1#', '#epirubicin#'): (164, 118),\n",
       " ('#hpp1#', '#everolimus#'): (164, 127),\n",
       " ('#hpp1#', '#exemestane#'): (164, 128),\n",
       " ('#hpp1#', '#letrozole#'): (164, 197),\n",
       " ('#hpp1#', '#paclitaxel#'): (164, 253),\n",
       " ('#hpp1#', '#trastuzumab#'): (164, 386),\n",
       " ('#hpp1#', '#vinorelbine#'): (164, 395),\n",
       " ('#hr#', '#akt inhibitor gsk2141795#'): (165, 9),\n",
       " ('#hr#', '#akt inhibitor mk2206#'): (165, 10),\n",
       " ('#hr#', '#buparlisib#'): (165, 47),\n",
       " ('#hr#', '#carboplatin#'): (165, 53),\n",
       " ('#hr#', '#cisplatin#'): (165, 74),\n",
       " ('#hr#', '#everolimus#'): (165, 127),\n",
       " ('#hr#', '#exemestane#'): (165, 128),\n",
       " ('#hr#', '#mtor kinase inhibitor cc-223#'): (165, 230),\n",
       " ('#hr#', '#paclitaxel#'): (165, 253),\n",
       " ('#hr#', '#pi3k/mtor inhibitor bez235#'): (165, 276),\n",
       " ('#hr#', '#temsirolimus#'): (165, 363),\n",
       " ('#hr#', '#trastuzumab#'): (165, 386),\n",
       " ('#hr#', '#vinorelbine#'): (165, 395),\n",
       " ('#hrg#', '#lapatinib#'): (167, 192),\n",
       " ('#hrg#', '#pi3k/mtor inhibitor bez235#'): (167, 276),\n",
       " ('#hrg#', '#trastuzumab#'): (167, 386),\n",
       " ('#il36g#', '#binimetinib#'): (173, 35),\n",
       " ('#il36g#', '#buparlisib#'): (173, 47),\n",
       " ('#il36g#', '#dalotuzumab#'): (173, 92),\n",
       " ('#il36g#', '#everolimus#'): (173, 127),\n",
       " ('#il36g#', '#pi3k/mtor inhibitor bez235#'): (173, 276),\n",
       " ('#il36g#', '#ridaforolimus#'): (173, 316),\n",
       " ('#il36g#', '#temsirolimus#'): (173, 363),\n",
       " ('#il36g#', '#vinorelbine#'): (173, 395),\n",
       " ('#il36g#', '#voxtalisib#'): (173, 396),\n",
       " ('#impact#', '#epirubicin#'): (174, 118),\n",
       " ('#impact#', '#paclitaxel#'): (174, 253),\n",
       " ('#impact#', '#trastuzumab#'): (174, 386),\n",
       " ('#iv#', '#afuresertib#'): (180, 5),\n",
       " ('#iv#', '#akt inhibitor gsk2141795#'): (180, 9),\n",
       " ('#iv#', '#akt inhibitor mk2206#'): (180, 10),\n",
       " ('#iv#', '#bevacizumab#'): (180, 33),\n",
       " ('#iv#', '#binimetinib#'): (180, 35),\n",
       " ('#iv#', '#buparlisib#'): (180, 47),\n",
       " ('#iv#', '#capecitabine#'): (180, 51),\n",
       " ('#iv#', '#carboplatin#'): (180, 53),\n",
       " ('#iv#', '#cisplatin#'): (180, 74),\n",
       " ('#iv#', '#cixutumumab#'): (180, 75),\n",
       " ('#iv#', '#cyclophosphamide#'): (180, 85),\n",
       " ('#iv#', '#dalotuzumab#'): (180, 92),\n",
       " ('#iv#', '#dexamethasone#'): (180, 96),\n",
       " ('#iv#', '#docetaxel#'): (180, 100),\n",
       " ('#iv#', '#doxorubicin#'): (180, 102),\n",
       " ('#iv#', '#epirubicin#'): (180, 118),\n",
       " ('#iv#', '#erlotinib#'): (180, 122),\n",
       " ('#iv#', '#everolimus#'): (180, 127),\n",
       " ('#iv#', '#exemestane#'): (180, 128),\n",
       " ('#iv#', '#gemcitabine#'): (180, 152),\n",
       " ('#iv#', '#lapatinib#'): (180, 192),\n",
       " ('#iv#', '#mtor kinase inhibitor cc-223#'): (180, 230),\n",
       " ('#iv#', '#paclitaxel albumin-stabilized nanoparticle formulation#'): (180,\n",
       "  252),\n",
       " ('#iv#', '#paclitaxel#'): (180, 253),\n",
       " ('#iv#', '#pi3k/mtor inhibitor bez235#'): (180, 276),\n",
       " ('#iv#', '#ribociclib#'): (180, 315),\n",
       " ('#iv#', '#ridaforolimus#'): (180, 316),\n",
       " ('#iv#', '#temsirolimus#'): (180, 363),\n",
       " ('#iv#', '#trametinib#'): (180, 383),\n",
       " ('#iv#', '#trastuzumab#'): (180, 386),\n",
       " ('#iv#', '#triciribine phosphate#'): (180, 387),\n",
       " ('#iv#', '#vinorelbine#'): (180, 395),\n",
       " ('#iv#', '#voxtalisib#'): (180, 396),\n",
       " ('#kit#', '#iproplatin#'): (185, 177),\n",
       " ('#large1#', '#everolimus#'): (193, 127),\n",
       " ('#large1#', '#lapatinib#'): (193, 192),\n",
       " ('#large1#', '#mtor kinase inhibitor cc-223#'): (193, 230),\n",
       " ('#lcn2#', '#doxorubicin#'): (194, 102),\n",
       " ('#map2k1#', '#buparlisib#'): (206, 47),\n",
       " ('#map2k1#', '#mek inhibitor ci-1040#'): (206, 213),\n",
       " ('#map2k1#', '#pi3k family kinase inhibitor xl147#'): (206, 274),\n",
       " ('#mapk3#', '#lapatinib#'): (208, 192),\n",
       " ('#mb#', '#mitoxantrone#'): (209, 221),\n",
       " ('#mdm2#', '#epirubicin#'): (212, 118),\n",
       " ('#mdm2#', '#paclitaxel#'): (212, 253),\n",
       " ('#mtor#', '#afuresertib#'): (231, 5),\n",
       " ('#mtor#', '#akt inhibitor gsk2141795#'): (231, 9),\n",
       " ('#mtor#', '#akt inhibitor mk2206#'): (231, 10),\n",
       " ('#mtor#', '#bevacizumab#'): (231, 33),\n",
       " ('#mtor#', '#binimetinib#'): (231, 35),\n",
       " ('#mtor#', '#buparlisib#'): (231, 47),\n",
       " ('#mtor#', '#capecitabine#'): (231, 51),\n",
       " ('#mtor#', '#carbendazim#'): (231, 52),\n",
       " ('#mtor#', '#carboplatin#'): (231, 53),\n",
       " ('#mtor#', '#cisplatin#'): (231, 74),\n",
       " ('#mtor#', '#cixutumumab#'): (231, 75),\n",
       " ('#mtor#', '#clofarabine#'): (231, 76),\n",
       " ('#mtor#', '#cyclophosphamide#'): (231, 85),\n",
       " ('#mtor#', '#dacarbazine#'): (231, 90),\n",
       " ('#mtor#', '#dalotuzumab#'): (231, 92),\n",
       " ('#mtor#', '#docetaxel#'): (231, 100),\n",
       " ('#mtor#', '#doxorubicin#'): (231, 102),\n",
       " ('#mtor#', '#erlotinib#'): (231, 122),\n",
       " ('#mtor#', '#everolimus#'): (231, 127),\n",
       " ('#mtor#', '#exemestane#'): (231, 128),\n",
       " ('#mtor#', '#fluorouracil#'): (231, 141),\n",
       " ('#mtor#', '#lapatinib#'): (231, 192),\n",
       " ('#mtor#', '#lenalidomide#'): (231, 196),\n",
       " ('#mtor#', '#letrozole#'): (231, 197),\n",
       " ('#mtor#', '#leucovorin calcium#'): (231, 198),\n",
       " ('#mtor#', '#mtor kinase inhibitor cc-223#'): (231, 230),\n",
       " ('#mtor#', '#oxaliplatin#'): (231, 251),\n",
       " ('#mtor#', '#paclitaxel albumin-stabilized nanoparticle formulation#'): (231,\n",
       "  252),\n",
       " ('#mtor#', '#paclitaxel#'): (231, 253),\n",
       " ('#mtor#', '#pertuzumab#'): (231, 268),\n",
       " ('#mtor#', '#pi3k/mtor inhibitor bez235#'): (231, 276),\n",
       " ('#mtor#', '#ridaforolimus#'): (231, 316),\n",
       " ('#mtor#', '#selumetinib#'): (231, 331),\n",
       " ('#mtor#', '#sorafenib tosylate#'): (231, 347),\n",
       " ('#mtor#', '#temsirolimus#'): (231, 363),\n",
       " ('#mtor#', '#thalidomide#'): (231, 371),\n",
       " ('#mtor#', '#trametinib#'): (231, 383),\n",
       " ('#mtor#', '#trastuzumab#'): (231, 386),\n",
       " ('#mtor#', '#vinorelbine#'): (231, 395),\n",
       " ('#mtor#', '#voxtalisib#'): (231, 396),\n",
       " ('#nf1#', '#folitixorin#'): (238, 142),\n",
       " ('#nm#', '#resistant starch#'): (241, 314),\n",
       " ('#pdlim5#', '#tamoxifen citrate#'): (264, 358),\n",
       " ('#phospho2#', '#recombinant interferon#'): (272, 309),\n",
       " ('#phospho2#', '#trastuzumab#'): (272, 386),\n",
       " ('#pik3ca#', '#afuresertib#'): (277, 5),\n",
       " ('#pik3ca#', '#akt inhibitor gsk2141795#'): (277, 9),\n",
       " ('#pik3ca#', '#akt inhibitor mk2206#'): (277, 10),\n",
       " ('#pik3ca#', '#alpelisib#'): (277, 14),\n",
       " ('#pik3ca#', '#bevacizumab#'): (277, 33),\n",
       " ('#pik3ca#', '#binimetinib#'): (277, 35),\n",
       " ('#pik3ca#', '#buparlisib#'): (277, 47),\n",
       " ('#pik3ca#', '#capecitabine#'): (277, 51),\n",
       " ('#pik3ca#', '#carbendazim#'): (277, 52),\n",
       " ('#pik3ca#', '#carboplatin#'): (277, 53),\n",
       " ('#pik3ca#', '#cisplatin#'): (277, 74),\n",
       " ('#pik3ca#', '#cixutumumab#'): (277, 75),\n",
       " ('#pik3ca#', '#cyclophosphamide#'): (277, 85),\n",
       " ('#pik3ca#', '#dalotuzumab#'): (277, 92),\n",
       " ('#pik3ca#', '#docetaxel#'): (277, 100),\n",
       " ('#pik3ca#', '#doxorubicin#'): (277, 102),\n",
       " ('#pik3ca#', '#erlotinib#'): (277, 122),\n",
       " ('#pik3ca#', '#everolimus#'): (277, 127),\n",
       " ('#pik3ca#', '#exemestane#'): (277, 128),\n",
       " ('#pik3ca#', '#gemcitabine#'): (277, 152),\n",
       " ('#pik3ca#', '#lapatinib#'): (277, 192),\n",
       " ('#pik3ca#', '#letrozole#'): (277, 197),\n",
       " ('#pik3ca#', '#mek inhibitor ci-1040#'): (277, 213),\n",
       " ('#pik3ca#', '#mtor kinase inhibitor cc-223#'): (277, 230),\n",
       " ('#pik3ca#',\n",
       "  '#paclitaxel albumin-stabilized nanoparticle formulation#'): (277, 252),\n",
       " ('#pik3ca#', '#paclitaxel#'): (277, 253),\n",
       " ('#pik3ca#', '#pertuzumab#'): (277, 268),\n",
       " ('#pik3ca#', '#pi3k family kinase inhibitor xl147#'): (277, 274),\n",
       " ('#pik3ca#', '#pi3k-beta inhibitor gsk2636771#'): (277, 275),\n",
       " ('#pik3ca#', '#pi3k/mtor inhibitor bez235#'): (277, 276),\n",
       " ('#pik3ca#', '#recombinant interleukin-6#'): (277, 310),\n",
       " ('#pik3ca#', '#ribociclib#'): (277, 315),\n",
       " ('#pik3ca#', '#ridaforolimus#'): (277, 316),\n",
       " ('#pik3ca#', '#tamoxifen#'): (277, 359),\n",
       " ('#pik3ca#', '#temsirolimus#'): (277, 363),\n",
       " ('#pik3ca#', '#trametinib#'): (277, 383),\n",
       " ('#pik3ca#', '#trans sodium crocetinate#'): (277, 384),\n",
       " ('#pik3ca#', '#trastuzumab#'): (277, 386),\n",
       " ('#pik3ca#', '#triciribine phosphate#'): (277, 387),\n",
       " ('#pik3ca#', '#vinorelbine#'): (277, 395),\n",
       " ('#pik3ca#', '#voxtalisib#'): (277, 396),\n",
       " ('#prdx2#', '#tamoxifen citrate#'): (290, 358),\n",
       " ('#prkaa2#', '#trastuzumab#'): (291, 386),\n",
       " ('#pten#', '#akt inhibitor gsk2141795#'): (298, 9),\n",
       " ('#pten#', '#akt inhibitor mk2206#'): (298, 10),\n",
       " ('#pten#', '#buparlisib#'): (298, 47),\n",
       " ('#pten#', '#carboplatin#'): (298, 53),\n",
       " ('#pten#', '#cisplatin#'): (298, 74),\n",
       " ('#pten#', '#everolimus#'): (298, 127),\n",
       " ('#pten#', '#exemestane#'): (298, 128),\n",
       " ('#pten#', '#fulvestrant#'): (298, 149),\n",
       " ('#pten#', '#lapatinib#'): (298, 192),\n",
       " ('#pten#', '#letrozole#'): (298, 197),\n",
       " ('#pten#', '#mtor kinase inhibitor cc-223#'): (298, 230),\n",
       " ('#pten#', '#paclitaxel#'): (298, 253),\n",
       " ('#pten#', '#pertuzumab#'): (298, 268),\n",
       " ('#pten#', '#pi3k-beta inhibitor gsk2636771#'): (298, 275),\n",
       " ('#pten#', '#pi3k/mtor inhibitor bez235#'): (298, 276),\n",
       " ('#pten#', '#recombinant interleukin-6#'): (298, 310),\n",
       " ('#pten#', '#tamoxifen#'): (298, 359),\n",
       " ('#pten#', '#temsirolimus#'): (298, 363),\n",
       " ('#pten#', '#trans sodium crocetinate#'): (298, 384),\n",
       " ('#pten#', '#trastuzumab#'): (298, 386),\n",
       " ('#pten#', '#vinorelbine#'): (298, 395),\n",
       " ('#raf1#', '#cyclophosphamide#'): (303, 85),\n",
       " ('#raf1#', '#docetaxel#'): (303, 100),\n",
       " ('#raf1#', '#doxorubicin#'): (303, 102),\n",
       " ('#raf1#', '#fluorouracil#'): (303, 141),\n",
       " ('#rnase1#', '#binimetinib#'): (318, 35),\n",
       " ('#rnase1#', '#buparlisib#'): (318, 47),\n",
       " ('#rnase1#', '#dalotuzumab#'): (318, 92),\n",
       " ('#rnase1#', '#everolimus#'): (318, 127),\n",
       " ('#rnase1#', '#lapatinib#'): (318, 192),\n",
       " ('#rnase1#', '#mtor kinase inhibitor cc-223#'): (318, 230),\n",
       " ('#rnase1#', '#pi3k/mtor inhibitor bez235#'): (318, 276),\n",
       " ('#rnase1#', '#ridaforolimus#'): (318, 316),\n",
       " ('#rnase1#', '#temsirolimus#'): (318, 363),\n",
       " ('#rnase1#', '#vinorelbine#'): (318, 395),\n",
       " ('#rnase1#', '#voxtalisib#'): (318, 396),\n",
       " ('#rrad#', '#akt inhibitor gsk2141795#'): (324, 9),\n",
       " ('#rrad#', '#akt inhibitor mk2206#'): (324, 10),\n",
       " ('#rrad#', '#buparlisib#'): (324, 47),\n",
       " ('#rrad#', '#carboplatin#'): (324, 53),\n",
       " ('#rrad#', '#cisplatin#'): (324, 74),\n",
       " ('#rrad#', '#everolimus#'): (324, 127),\n",
       " ('#rrad#', '#exemestane#'): (324, 128),\n",
       " ('#rrad#', '#mtor kinase inhibitor cc-223#'): (324, 230),\n",
       " ('#rrad#', '#paclitaxel#'): (324, 253),\n",
       " ('#rrad#', '#pi3k/mtor inhibitor bez235#'): (324, 276),\n",
       " ('#rrad#', '#temsirolimus#'): (324, 363),\n",
       " ('#rrad#', '#trastuzumab#'): (324, 386),\n",
       " ('#rrad#', '#vinorelbine#'): (324, 395),\n",
       " ('#sgcg#', '#afuresertib#'): (334, 5),\n",
       " ('#sgcg#', '#akt inhibitor gsk2141795#'): (334, 9),\n",
       " ('#sgcg#', '#akt inhibitor mk2206#'): (334, 10),\n",
       " ('#sgcg#', '#binimetinib#'): (334, 35),\n",
       " ('#sgcg#', '#buparlisib#'): (334, 47),\n",
       " ('#sgcg#', '#carboplatin#'): (334, 53),\n",
       " ('#sgcg#', '#cisplatin#'): (334, 74),\n",
       " ('#sgcg#', '#cixutumumab#'): (334, 75),\n",
       " ('#sgcg#', '#cyclophosphamide#'): (334, 85),\n",
       " ('#sgcg#', '#doxorubicin#'): (334, 102),\n",
       " ('#sgcg#', '#erlotinib#'): (334, 122),\n",
       " ('#sgcg#', '#everolimus#'): (334, 127),\n",
       " ('#sgcg#', '#exemestane#'): (334, 128),\n",
       " ('#sgcg#', '#gemcitabine#'): (334, 152),\n",
       " ('#sgcg#', '#mtor kinase inhibitor cc-223#'): (334, 230),\n",
       " ('#sgcg#', '#paclitaxel#'): (334, 253),\n",
       " ('#sgcg#', '#pi3k/mtor inhibitor bez235#'): (334, 276),\n",
       " ('#sgcg#', '#tamoxifen citrate#'): (334, 358),\n",
       " ('#sgcg#', '#tamoxifen#'): (334, 359),\n",
       " ('#sgcg#', '#temsirolimus#'): (334, 363),\n",
       " ('#sgcg#', '#trametinib#'): (334, 383),\n",
       " ('#sgcg#', '#trastuzumab#'): (334, 386),\n",
       " ('#sgcg#', '#triciribine phosphate#'): (334, 387),\n",
       " ('#sgcg#', '#vinorelbine#'): (334, 395),\n",
       " ('#sirt1#', '#iproplatin#'): (337, 177),\n",
       " ('#sirt1#', '#niacinamide#'): (337, 240),\n",
       " ('#sirt1#', '#tamoxifen citrate#'): (337, 358),\n",
       " ('#sirt1#', '#tamoxifen#'): (337, 359),\n",
       " ('#slc7a1#', '#iproplatin#'): (339, 177),\n",
       " ('#sltm#', '#trastuzumab#'): (340, 386),\n",
       " ('#smim10l2a#', '#trastuzumab#'): (343, 386),\n",
       " ('#snai2#', '#trastuzumab#'): (345, 386),\n",
       " ('#spen#', '#tamoxifen#'): (350, 359),\n",
       " ('#src#', '#lapatinib#'): (351, 192),\n",
       " ('#t#', '#akt inhibitor mk2206#'): (357, 10),\n",
       " ('#t#', '#docetaxel#'): (357, 100),\n",
       " ('#t#', '#lapatinib#'): (357, 192),\n",
       " ('#t#', '#paclitaxel#'): (357, 253),\n",
       " ('#t#', '#pertuzumab#'): (357, 268),\n",
       " ('#t#', '#trastuzumab emtansine#'): (357, 385),\n",
       " ('#t#', '#trastuzumab#'): (357, 386),\n",
       " ('#th#', '#tamoxifen#'): (370, 359),\n",
       " ('#tnip1#', '#everolimus#'): (377, 127),\n",
       " ('#tnip1#', '#fulvestrant#'): (377, 149),\n",
       " ('#tnip1#', '#trastuzumab#'): (377, 386),\n",
       " ('#tp53#', '#cyclophosphamide#'): (381, 85),\n",
       " ('#tp53#', '#docetaxel#'): (381, 100),\n",
       " ('#tp53#', '#doxorubicin#'): (381, 102),\n",
       " ('#tp53#', '#epirubicin#'): (381, 118),\n",
       " ('#tp53#', '#fluorouracil#'): (381, 141),\n",
       " ('#tp53#', '#hydroxyurea#'): (381, 169),\n",
       " ('#tp53#', '#iproplatin#'): (381, 177),\n",
       " ('#tp53#', '#paclitaxel#'): (381, 253)}"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "original_indeces"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "a =np.zeros((vocab_matrix.shape[1],1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "a[0]=1\n",
    "a[221]=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(406, 1)"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "b = scs.lil_matrix((vocab_matrix.shape[1],1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<406x1 sparse matrix of type '<class 'numpy.float64'>'\n",
       "\twith 0 stored elements in LInked List format>"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "b[list((0,221))]=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  (0, 0)\t1.0\n",
      "  (221, 0)\t1.0\n"
     ]
    }
   ],
   "source": [
    "print(b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<2893x1 sparse matrix of type '<class 'numpy.float64'>'\n",
       "\twith 4 stored elements in Compressed Sparse Row format>"
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vocab_matrix*b"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1, 2, 1, ..., 1, 1, 1])"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vocab_matrix.data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "vocab_test =vocab_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "vocab_test[vocab_test>1]=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<2893x406 sparse matrix of type '<class 'numpy.bool_'>'\n",
       "\twith 0 stored elements in Compressed Sparse Row format>"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vocab_test>1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([448], dtype=int32)"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(vocab_test*b>1).nonzero()[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'In the studies with the side populations from MCF-7 and MDA-MB-231 cells, increased ABCG2 was observed in the side populations as well as elevated resistance to the chemotherapeutic drug mitoxantrone.'"
      ]
     },
     "execution_count": 212,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orig_sentences[448]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 528,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_evidence_sentences(gene, drug, max_number):\n",
    "    gen = '#'+gene+'#'\n",
    "    drg = '#'+drug+'#'\n",
    "    gene = gene.lower()\n",
    "    drug = drug.lower()\n",
    "    if gen not in gene_values:\n",
    "        try:\n",
    "            gen=gene_dict[gene]\n",
    "        except KeyError:\n",
    "            return (\"Gene: {} not in list\".format(gene))\n",
    "    if drg not in drug_values:\n",
    "        try:\n",
    "            drg=drug_dict[drug]\n",
    "        except KeyError:\n",
    "            return (\"Drug: {} not in list\".format(drug))\n",
    "    try:\n",
    "        indices = original_indeces[(gen, drg)]\n",
    "    except KeyError:\n",
    "        return('There is no evidence of interaction'\n",
    "        ' between the gene {} and the drug {}'. format(gen[1:-1],drg[1:-1]))\n",
    "    test_col = scs.lil_matrix((vocab_matrix.shape[1],1))\n",
    "    test_col[list(indices)]=1\n",
    "    index_evidence = (vocab_matrix*test_col>1).nonzero()[0]\n",
    "    evidence_list = []\n",
    "    #gene_key = [k for k, v in gene_dict.items() if v == gen]\n",
    "    #drug_key = [k for k, v in drug_dict.items() if v == drg]\n",
    "    while len(evidence_list) < max_number and len(evidence_list) < len(index_evidence) :\n",
    "        for index in index_evidence:\n",
    "            sent = orig_sentences[index]#.lower()\n",
    "            #gene_evd = [word for word in sent.replace(',','').replace('.','').replace(':','').split(' ') if word in gene_key][0]\n",
    "            #drug_evd = [word for word in sent.replace(',','').replace('.','').replace(':','').split(' ') if word in drug_key][0]\n",
    "            if len(sent)<500:\n",
    "                evidence_list.append('{}: {}'.format(doc_names[index] ,sent))\n",
    "            else:\n",
    "                evidence_list.append('{}: sentence too long to display'.format(doc_names[index]))  \n",
    "    return evidence_list\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 529,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'test.txt'"
      ]
     },
     "execution_count": 529,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "doc_names[448]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 530,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['test.txt: In the studies with the side populations from MCF-7 and MDA-MB-231 cells, increased ABCG2 was observed in the side populations as well as elevated resistance to the chemotherapeutic drug mitoxantrone.']"
      ]
     },
     "execution_count": 530,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_evidence_sentences('abcg2','mitoxantrone',10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 531,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['test.txt: In BT474 and MCF10A breast cancer cells transfected with the construct encoding HER2-T798M mutation, elevated HER2 kinase activity was detected and lapatinib did not block phosphorylation of HER2, EGFR3 or downstream Akt and ERK1 2.',\n",
       " 'test.txt: Treatment of the lapatinib-resistant HER2+ cells with Src inhibitors suppressed the PI3K PTEN Akt mTOR pathway as well as  Figure 5: Sites of Targeting the EGFR PI3K PTEN Akt mTORC Pathway with Small Molecule Membrane-Permeable Inhibitors and Monoclonal Antibodies (MoAbs).',\n",
       " 'test.txt: sentence too long to display']"
      ]
     },
     "execution_count": 531,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_evidence_sentences('akt1','lapatinib',2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 535,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "almost_vocab = data_frame_creator.open_pickle('almost_vocab_columns.pickle')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 537,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open('almost_vocab.txt','w') as f:\n",
    "    f.write(\"\\n\".join(almost_vocab))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 541,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "almost_vocab_drugs_index =[almost_vocab.index(i) for i in almost_vocab if i in drug_values] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 588,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "almost_vocab_drug_name =[i for i in almost_vocab if i in drug_values] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 589,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open('almost_vocab_drugs.txt','w') as f:\n",
    "    f.write(\"\\n\".join(almost_vocab_drug_name))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 590,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "repeat_names =[]\n",
    "for drug in almost_vocab_drug_name:\n",
    "    for drug2 in almost_vocab_drug_name:\n",
    "        if drug2[1:-1] in drug[1:-1] and len(drug2)<len(drug):\n",
    "            repeat_names.append((drug2,drug))\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 591,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "124"
      ]
     },
     "execution_count": 591,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(repeat_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 592,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'#7-ethyl-10-hydroxycamptothecin#'"
      ]
     },
     "execution_count": 592,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "repeat_names[0][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 593,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "drug_key = [k for k, v in drug_dict.items() if v == repeat_names[0][1]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 594,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['7-ethyl-10-hydroxycamptothecin',\n",
       " '7-ethyl-10-hydroxy-20(s)-camptothecin',\n",
       " 'sn 38',\n",
       " 'sn-38']"
      ]
     },
     "execution_count": 594,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 607,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for repeat in repeat_names:\n",
    "    drug_dict[repeat[1][1:-1]] = repeat[0]\n",
    "    drug_key = [k for k, v in drug_dict.items() if v == repeat[1]]\n",
    "    for drug in drug_key:\n",
    "        #print(drug,repeat[0])\n",
    "        drug_dict[drug]=repeat[0]\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 608,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "repeat_names_2 =[]\n",
    "for drug in almost_vocab_drug_name_2:\n",
    "    for drug2 in almost_vocab_drug_name_2:\n",
    "        if drug2[1:-1] in drug[1:-1] and len(drug2)<len(drug):\n",
    "            repeat_names_2.append((drug2,drug))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 609,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 609,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "repeat_names_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 616,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'#acridine#'"
      ]
     },
     "execution_count": 616,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_dict['acridine carboxamide']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 627,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open('drug_dictionary_final.pickle', 'wb') as handle:\n",
    "    pickle.dump(drug_dict, handle, protocol=pickle.HIGHEST_PROTOCOL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 629,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test_dict = data_frame_creator.open_pickle('drug_dictionary_final.pickle')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 630,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'#acridine#'"
      ]
     },
     "execution_count": 630,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_dict['acridine carboxamide']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 631,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open('drug_dictionary_final.json', 'w') as fp:\n",
    "    json.dump(drug_dict, fp, sort_keys=True, indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
